// @codecademy/gamut-labs/dist/AppHeaderMobile/AppHeaderMainMenuMobile/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import { FlexBox } from '@codecademy/gamut';
import React, { useState } from 'react';
import { AppHeaderFillButton, AppHeaderTextButton } from '../..';
import { AppHeaderLinkMobile } from '../AppHeaderLinkMobile';
import { AppHeaderSubMenuMobile } from '../AppHeaderSubMenuMobile';
import { AppHeaderSubMenuTarget } from '../AppHeaderSubMenuTarget';
export var AppHeaderMainMenuMobile = function AppHeaderMainMenuMobile(_ref) {
  var items = _ref.items,
      action = _ref.action,
      renderSearch = _ref.renderSearch;

  var _useState = useState(),
      _useState2 = _slicedToArray(_useState, 2),
      subMenuItem = _useState2[0],
      setSubMenuItem = _useState2[1];

  var openSubMenu = function openSubMenu(event, item) {
    action(event, item);
    setSubMenuItem(item);
  };

  var closeSubMenu = function closeSubMenu() {
    setSubMenuItem(undefined);
  };

  var mapItemToElement = function mapItemToElement(item, action) {
    switch (item.type) {
      case 'link':
        return /*#__PURE__*/React.createElement(AppHeaderLinkMobile, {
          key: item.id,
          item: item,
          action: action
        });

      case 'dropdown':
      case 'profile-dropdown':
        return /*#__PURE__*/React.createElement(AppHeaderSubMenuTarget, {
          key: item.id,
          item: item,
          openSubMenu: openSubMenu
        });

      case 'fill-button':
        return /*#__PURE__*/React.createElement(FlexBox, {
          justifyContent: "center",
          mt: 32,
          key: item.id
        }, /*#__PURE__*/React.createElement(AppHeaderFillButton, {
          "data-testid": item.dataTestId,
          href: item.href,
          onClick: function onClick(event) {
            return action(event, item);
          }
        }, item.text));

      case 'text-button':
        return /*#__PURE__*/React.createElement(FlexBox, {
          justifyContent: "center",
          mt: 16,
          key: item.id
        }, /*#__PURE__*/React.createElement(AppHeaderTextButton, {
          "data-testid": item.dataTestId,
          href: item.href,
          onClick: function onClick(event) {
            return action(event, item);
          }
        }, item.text));
    }
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, subMenuItem ? /*#__PURE__*/React.createElement(AppHeaderSubMenuMobile, {
    handleClose: closeSubMenu,
    action: action,
    item: subMenuItem
  }) : /*#__PURE__*/React.createElement(React.Fragment, null, renderSearch === null || renderSearch === void 0 ? void 0 : renderSearch(), items.map(function (item) {
    return mapItemToElement(item, action);
  })));
};
