// @codecademy/gamut/dist/Pattern/defs.js
import React from 'react';
export var defs = function defs(name, idSuffix) {
  return /*#__PURE__*/React.createElement("defs", null, name === 'diagonalStripesLoose' && /*#__PURE__*/React.createElement("pattern", {
    id: "diagonalStripesLoose".concat(idSuffix),
    x: "0",
    y: "0",
    width: "16",
    height: "16",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("rect", {
    y: "15",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "1",
    y: "14",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "2",
    y: "13",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "12",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "4",
    y: "11",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "5",
    y: "10",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "6",
    y: "9",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "7",
    y: "8",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "8",
    y: "7",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "9",
    y: "6",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "10",
    y: "5",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "11",
    y: "4",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "12",
    y: "3",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "13",
    y: "2",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "14",
    y: "1",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "15",
    width: "1",
    height: "1",
    fill: "currentColor"
  })), name === 'diagonalStripesRegular' && /*#__PURE__*/React.createElement("pattern", {
    id: "diagonalStripesRegular".concat(idSuffix),
    x: "0",
    y: "0",
    width: "8",
    height: "8",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("g", {
    clipPath: "url(#clipDiagonalStripesRegular".concat(idSuffix, ")")
  }, /*#__PURE__*/React.createElement("rect", {
    width: "8",
    height: "8",
    fill: "white"
  }), /*#__PURE__*/React.createElement("rect", {
    y: "7",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "1",
    y: "6",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "2",
    y: "5",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "4",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "4",
    y: "3",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "5",
    y: "2",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "6",
    y: "1",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "7",
    width: "1",
    height: "1",
    fill: "currentColor"
  })), /*#__PURE__*/React.createElement("defs", null, /*#__PURE__*/React.createElement("clipPath", {
    id: "clipDiagonalStripesRegular".concat(idSuffix)
  }, /*#__PURE__*/React.createElement("rect", {
    width: "8",
    height: "8",
    fill: "white"
  })))), name === 'diagonalStripesDense' && /*#__PURE__*/React.createElement("pattern", {
    id: "diagonalStripesDense".concat(idSuffix),
    x: "0",
    y: "0",
    width: "4",
    height: "4",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("g", {
    clipPath: "url(#clipDiagonalStripesDense".concat(idSuffix, ")")
  }, /*#__PURE__*/React.createElement("rect", {
    width: "4",
    height: "4",
    fill: "white"
  }), /*#__PURE__*/React.createElement("rect", {
    y: "3",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "1",
    y: "2",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "2",
    y: "1",
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    width: "1",
    height: "1",
    fill: "currentColor"
  })), /*#__PURE__*/React.createElement("defs", null, /*#__PURE__*/React.createElement("clipPath", {
    id: "clipDiagonalStripesDense".concat(idSuffix)
  }, /*#__PURE__*/React.createElement("rect", {
    width: "4",
    height: "4",
    fill: "white"
  })))), name === 'checkerLoose' && /*#__PURE__*/React.createElement("pattern", {
    id: "checkerLoose".concat(idSuffix),
    x: "0",
    y: "0",
    width: "16",
    height: "16",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("rect", {
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "8",
    y: "8",
    width: "1",
    height: "1",
    fill: "currentColor"
  })), name === 'checkerRegular' && /*#__PURE__*/React.createElement("pattern", {
    id: "checkerRegular".concat(idSuffix),
    x: "0",
    y: "0",
    width: "8",
    height: "8",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("rect", {
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "4",
    y: "4",
    width: "1",
    height: "1",
    fill: "currentColor"
  })), name === 'checkerDense' && /*#__PURE__*/React.createElement("pattern", {
    id: "checkerDense".concat(idSuffix),
    x: "0",
    y: "0",
    width: "4",
    height: "4",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("rect", {
    width: "1",
    height: "1",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "2",
    y: "2",
    width: "1",
    height: "1",
    fill: "currentColor"
  })), name === 'dotLoose' && /*#__PURE__*/React.createElement("pattern", {
    id: "dotLoose".concat(idSuffix),
    x: "0",
    y: "0",
    width: "16",
    height: "16",
    patternUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("rect", {
    width: "0.5",
    height: "0.5",
    fill: "currentColor",
    fillOpacity: "0.5"
  }), /*#__PURE__*/React.createElement("rect", {
    y: "1",
    width: "0.5",
    height: "0.5",
    fill: "currentColor",
    fillOpacity: "0.5"
  }), /*#__PURE__*/React.createElement("rect", {
    y: "0.5",
    width: "0.5",
    height: "0.5",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "1",
    width: "0.5",
    height: "0.5",
    fill: "currentColor",
    fillOpacity: "0.5"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "1",
    y: "1",
    width: "0.5",
    height: "0.5",
    fill: "currentColor",
    fillOpacity: "0.5"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "1",
    y: "0.5",
    width: "0.5",
    height: "0.5",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0.5",
    width: "0.5",
    height: "0.5",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0.5",
    y: "1",
    width: "0.5",
    height: "0.5",
    fill: "currentColor"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0.5",
    y: "0.5",
    width: "0.5",
    height: "0.5",
    fill: "currentColor"
  })));
};
