// @pluralsight/ps-design-system-verticaltabs/dist/cjs/react/group.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CollapsibleGroup = exports.Group = void 0;
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_collapsible_1 = __importDefault(require("@pluralsight/ps-design-system-collapsible"));
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var list_1 = require("./list");
var glamor = glamor_1.default || glamorExports;
var styles = {
    group: function () { return glamor.css({ label: 'verticaltabs__group' }); },
    groupHeader: function (themeName) {
        var label = 'verticaltabs__group__header';
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]), glamor.css(index_1.default[".psds-" + label + ".psds-theme--" + themeName]));
    },
    groupButton: function () {
        return glamor.css(index_1.default['.psds-verticaltabs__group__button']);
    },
    groupCollapsibleList: function () {
        return glamor.compose(glamor.css(index_1.default['.psds-verticaltabs__list']), glamor.css(index_1.default['.psds-verticaltabs__group__collapsible-list']));
    },
    rotatable: function (open) {
        return glamor.css(index_1.default['.psds-verticaltabs__rotatable'], open && index_1.default['.psds-verticaltabs__rotatable--isOpen']);
    },
    headerLabel: function () { return glamor.css(index_1.default['.psds-verticaltabs__header__label']); }
};
var Group = react_1.default.forwardRef(function (props, ref) {
    var rest = ps_design_system_util_1.omit(props, ['header', 'startOpen']);
    return (react_1.default.createElement("li", __assign({ ref: ref }, styles.group(), rest),
        props.header,
        react_1.default.createElement(list_1.List, null, props.children)));
});
exports.Group = Group;
var GroupHeader = react_1.default.forwardRef(function (props, ref) {
    var children = props.children, _a = props.tagName, Tag = _a === void 0 ? 'h2' : _a, rest = __rest(props, ["children", "tagName"]);
    var themeName = ps_design_system_theme_1.useTheme();
    return (react_1.default.createElement(Tag, __assign({}, rest, styles.groupHeader(themeName), { ref: ref }),
        react_1.default.createElement("div", null,
            react_1.default.createElement("span", __assign({}, styles.headerLabel()), children))));
});
Group.Header = GroupHeader;
Group.displayName = 'VerticalTabs.Group';
Group.Header.displayName = 'VerticalTabs.Group.Header';
var CollapsibleGroup = react_1.default.forwardRef(function (props, ref) {
    var children = props.children, header = props.header, startOpen = props.startOpen, rest = __rest(props, ["children", "header", "startOpen"]);
    var _a = react_1.default.useState(startOpen), open = _a[0], setOpenState = _a[1];
    var toggle = function () {
        setOpenState(!open);
    };
    return (react_1.default.createElement("div", __assign({ ref: ref }, rest),
        react_1.default.isValidElement(header) &&
            react_1.default.cloneElement(header, {
                toggle: toggle,
                open: open
            }),
        react_1.default.createElement(ps_design_system_collapsible_1.default, __assign({ isOpen: open, tagName: "ul" }, styles.groupCollapsibleList()), children)));
});
exports.CollapsibleGroup = CollapsibleGroup;
var CollapsibleGroupHeader = react_1.default.forwardRef(function (props, ref) {
    var children = props.children, open = props.open, _a = props.tagName, Tag = _a === void 0 ? 'h2' : _a, toggle = props.toggle, rest = __rest(props, ["children", "open", "tagName", "toggle"]);
    var themeName = ps_design_system_theme_1.useTheme();
    return (react_1.default.createElement(Tag, __assign({}, rest, styles.groupHeader(themeName), { ref: ref }),
        react_1.default.createElement("button", __assign({}, styles.groupButton(), { onClick: toggle, "aria-expanded": open ? 'true' : 'false' }),
            react_1.default.createElement("span", __assign({}, styles.headerLabel()), children),
            react_1.default.createElement(ps_design_system_icon_1.CaretDownIcon, __assign({ "aria-hidden": "true", size: ps_design_system_icon_1.CaretDownIcon.sizes.small }, styles.rotatable(Boolean(open)))))));
});
CollapsibleGroup.Header = CollapsibleGroupHeader;
CollapsibleGroup.displayName = 'VerticalTabs.CollapsibleGroup';
CollapsibleGroup.Header.displayName = 'VerticalTabs.CollapsibleGroup.Header';
