// @pluralsight/ps-design-system-verticaltabs/dist/cjs/react/item.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tier2 = exports.Tier1 = void 0;
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importStar(require("react"));
var context_1 = require("./context");
var index_1 = __importDefault(require("../css/index"));
var list_1 = require("./list");
var glamor = glamor_1.default || glamorExports;
var styles = {
    item: function (themeName) {
        var label = 'verticaltabs__item';
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]), glamor.css(index_1.default[".psds-" + label + ".psds-theme--" + themeName]));
    },
    itemIcon: function () { return glamor.css(index_1.default['.psds-verticaltabs__item__icon']); },
    itemIconActive: function () {
        return glamor.css(index_1.default['.psds-verticaltabs__item__icon--active']);
    },
    itemTier: function (themeName, props) {
        var label = "verticaltabs__" + props.type;
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]), glamor.css(index_1.default[".psds-" + label + ".psds-theme--" + themeName]));
    },
    tier1Header: function () {
        var label = "verticaltabs__tier1__header";
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]));
    },
    tier1HeaderInner: function () {
        var label = "verticaltabs__tier1__header__inner";
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]));
    },
    tier2Header: function () {
        var label = "verticaltabs__tier2__header";
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]));
    },
    tierHeaderLabel: function (hideLabels) {
        var label = "verticaltabs__header__label";
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]), hideLabels && glamor.css(index_1.default[".psds-" + label + "--hide-labels"]));
    },
    tierHeaderLabelIcon: function (collapsed) {
        var label = "verticaltabs__header__label__icon";
        return glamor.compose(glamor.css({ label: label }), glamor.css(index_1.default[".psds-" + label]), collapsed && glamor.css(index_1.default[".psds-" + label + "--collapsed"]));
    }
};
var Item = react_1.default.forwardRef(function (props, ref) {
    var active = props.active, _a = props.collapsed, initialCollapsed = _a === void 0 ? false : _a, _b = props.collapsible, collapsible = _b === void 0 ? false : _b, children = props.children, header = props.header, onClick = props.onClick, type = props.itemType, rest = __rest(props, ["active", "collapsed", "collapsible", "children", "header", "onClick", "itemType"]);
    var _c = react_1.default.useState(initialCollapsed), collapsed = _c[0], setCollapsed = _c[1];
    react_1.useEffect(function () {
        setCollapsed(initialCollapsed);
    }, [initialCollapsed]);
    var themeName = ps_design_system_theme_1.useTheme();
    var handleHeaderClick = ps_design_system_util_1.combineFns(function () {
        setCollapsed(!collapsed);
    }, onClick);
    var ListComp = collapsible ? list_1.CollapsibleList : list_1.List;
    return (react_1.default.createElement("li", __assign({ ref: ref }, rest),
        react_1.default.createElement("div", __assign({}, styles.item(themeName), styles.itemTier(themeName, { type: type }), (active && { 'data-active': true })), header &&
            react_1.default.cloneElement(header, __assign({ active: active,
                collapsed: collapsed,
                collapsible: collapsible }, (collapsible && { onClick: handleHeaderClick })))),
        children && react_1.default.createElement(ListComp, { collapsed: collapsed }, children)));
});
Item.displayName = 'VerticalTabs.Item';
var Tier1 = function (props) { return (react_1.default.createElement(Item, __assign({}, props, { itemType: "tier1" }))); };
exports.Tier1 = Tier1;
var Tier1Header = react_1.default.forwardRef(function (props, ref) {
    var active = props.active, collapsed = props.collapsed, collapsible = props.collapsible, children = props.children, icon = props.icon, rest = __rest(props, ["active", "collapsed", "collapsible", "children", "icon"]);
    var hideLabels = context_1.useHideLabels();
    var Tag = function (wrapperProps) {
        return rest.href ? (react_1.default.createElement("a", __assign({}, wrapperProps, { ref: ref }))) : rest.onClick ? (react_1.default.createElement("button", __assign({}, wrapperProps, { ref: ref }))) : (react_1.default.createElement("span", __assign({}, wrapperProps, { ref: ref })));
    };
    return (react_1.default.createElement(Tag, __assign({}, styles.tier1Header(), rest, { "aria-expanded": !collapsed }),
        icon &&
            react_1.default.cloneElement(icon, __assign(__assign({ size: ps_design_system_icon_1.CaretDownIcon.sizes.medium }, styles.itemIcon()), (active ? { 'data-active': true } : {}))),
        react_1.default.createElement("span", __assign({}, styles.tierHeaderLabel(hideLabels)), children),
        collapsible && (react_1.default.createElement(ps_design_system_icon_1.CaretDownIcon, __assign({ "aria-hidden": "true", size: ps_design_system_icon_1.CaretDownIcon.sizes.small }, styles.tierHeaderLabelIcon(Boolean(collapsed)))))));
});
Tier1.Header = Tier1Header;
Tier1.displayName = 'VerticalTabs.Tier1';
Tier1.Header.displayName = 'VerticalTabs.Tier1.Header';
var Tier2 = function (props) {
    return react_1.default.createElement(Item, __assign({}, props, { itemType: "tier2" }));
};
exports.Tier2 = Tier2;
var Tier2Header = react_1.default.forwardRef(function (props, ref) {
    var hideLabels = context_1.useHideLabels();
    // NOTE: some props are given during clone that are not used as should not be
    //       passed to the underlying dom node
    var rest = ps_design_system_util_1.omit(props, ['active', 'collapsed', 'collapsible']);
    var Tag = function (wrapperProps) {
        return rest.href ? (react_1.default.createElement("a", __assign({}, wrapperProps, { ref: ref }))) : rest.onClick ? (react_1.default.createElement("button", __assign({}, wrapperProps, { ref: ref }))) : (react_1.default.createElement("span", __assign({}, wrapperProps, { ref: ref })));
    };
    var hideItemProps = props.collapsed || hideLabels
        ? {
            'aria-hidden': 'true',
            tabIndex: -1
        }
        : {};
    return (react_1.default.createElement(Tag, __assign({}, styles.tier2Header, rest, hideItemProps),
        react_1.default.createElement("span", __assign({}, styles.tierHeaderLabel(hideLabels)), props.children)));
});
Tier2.Header = Tier2Header;
Tier2.displayName = 'VerticalTabs.Tier2';
Tier2.Header.displayName = 'VerticalTabs.Tier2.Header';
exports.default = Item;
