// @codecademy/gamut-labs/dist/CurriculumCard/Footer/index.js
import { FlexBox, Text } from '@codecademy/gamut';
import React from 'react';
import { BottomTag } from '../BottomTag/index';
export var Footer = function Footer(_ref) {
  var progressState = _ref.progressState,
      tag = _ref.tag,
      tagColor = _ref.tagColor;

  if (!progressState && tag && tagColor) {
    return /*#__PURE__*/React.createElement(BottomTag, {
      text: tag,
      color: tagColor
    });
  }

  return /*#__PURE__*/React.createElement("div", null, progressState === 'completed' && /*#__PURE__*/React.createElement(Text, {
    fontSize: 16,
    fontFamily: "accent",
    textColor: "yellow"
  }, "Completed"), progressState === 'inProgress' && /*#__PURE__*/React.createElement(FlexBox, {
    flexWrap: "wrap",
    justifyContent: "space-between",
    alignItems: "flex-end",
    width: "100%"
  }, /*#__PURE__*/React.createElement(Text, {
    fontFamily: "accent",
    fontSize: 14
  }, "Enrolled..."), /*#__PURE__*/React.createElement(Text, {
    textColor: "hyper"
  }, "Keep Going")));
};
