// @pluralsight/ps-design-system-appframe/dist/esm/css/index.js
var _a, _b, _c;
import { breakpoints, colorsBackgroundDark, colorsBackgroundLight, colorsBorder, layout, motion } from '@pluralsight/ps-design-system-core';
import { names as themeNames } from '@pluralsight/ps-design-system-theme';
import { sidenavVariants } from '../vars/index';
var layers = {
    content: 0,
    sidenav: 930,
    topnav: 950,
    skipBanner: 1600
};
var sidenavMinimizedWidth = '72px';
var sidenavWidth = '240px';
var topnavHeight = '56px';
export default (_a = {
        '.psds-appframe': {
            label: 'appframe'
        }
    },
    _a[".psds-appframe.psds-theme--" + themeNames.dark] = {
        background: colorsBackgroundDark[1]
    },
    _a[".psds-appframe.psds-theme--" + themeNames.light] = {
        background: colorsBackgroundLight[1]
    },
    _a['.psds-appframe__skip-banner'] = {
        label: 'appframe__skip-banner',
        top: 0,
        left: '50%',
        padding: " " + layout.spacingSmall,
        position: 'fixed',
        transform: 'translate(-50%, -300px)',
        transition: 'transform 1s ease-out',
        zIndex: layers.skipBanner,
        '&:focus-within, &[focus-within]': {
            opacity: 1,
            transform: "translate(-50%, 0)",
            transition: 'top 0.1s ease-in'
        },
        '@media print ': { display: 'none' }
    },
    _a['.psds-appframe__container'] = {
        label: 'appframe__container',
        minHeight: '100vh',
        transition: "all " + motion.speedFast + " ease-in-out",
        transitionProperty: 'padding-left'
    },
    _a[".psds-appframe__container--" + sidenavVariants.open] = {
        paddingLeft: sidenavWidth
    },
    _a[".psds-appframe__container--" + sidenavVariants.overlay] = (_b = {},
        _b["@media (min-width: " + breakpoints.small + ")"] = {
            paddingLeft: sidenavMinimizedWidth
        },
        _b),
    _a[".psds-appframe__container--" + sidenavVariants.minimized] = {
        paddingLeft: sidenavMinimizedWidth
    },
    _a['.psds-appframe__content'] = {
        label: 'appframe__content',
        paddingBottom: 1,
        paddingTop: topnavHeight,
        zIndex: layers.content
    },
    _a['.psds-appframe__sidenav'] = {
        label: 'appframe__sidenav',
        backgroundColor: colorsBackgroundDark[2],
        borderRight: "1px solid " + colorsBorder.lowOnDark,
        flexDirection: 'row',
        height: "calc(100vh - " + topnavHeight + ")",
        left: 0,
        position: 'fixed',
        top: topnavHeight,
        transition: "all " + motion.speedFast + " ease-in-out",
        transitionProperty: 'transform, width',
        width: sidenavWidth,
        zIndex: layers.sidenav,
        '&::after': {
            content: ' ',
            display: 'block',
            height: '100%',
            left: 0,
            pointerEvents: 'none',
            position: 'fixed',
            top: 0,
            transition: "box-shadow " + motion.speedFast + " ease-in-out",
            width: '100%'
        }
    },
    _a[".psds-appframe__sidenav--" + sidenavVariants.overlay] = {
        transform: 'translate(0, 0)',
        '&::after': { boxShadow: '50px 0 100px rgba(0, 0, 0, .8)' }
    },
    _a[".psds-appframe__sidenav--" + sidenavVariants.closed] = (_c = {
            transform: 'translate(-100%, 0)',
            '&::after': { boxShadow: '0 0 0 rgba(0, 0, 0, .8)' }
        },
        _c["@media (min-width: " + breakpoints.small + ")"] = {
            transform: 'translate(0, 0)',
            width: 0
        },
        _c),
    _a[".psds-appframe__sidenav--" + sidenavVariants.minimized] = {
        width: sidenavMinimizedWidth
    },
    _a['.psds-appframe__sidenav__overflow-mask'] = {
        label: 'appframe__sidenav__overflow-mask',
        height: '100%',
        overflow: 'hidden'
    },
    _a['.psds-appframe__sidenav__inner'] = {
        label: 'appframe__sidenav__inner',
        height: '100%',
        minWidth: sidenavWidth,
        '&:before, &:after': {
            display: 'block',
            content: ' ',
            height: 1
        }
    },
    _a['.psds-appframe__topnav'] = {
        label: 'appframe__topnav',
        height: topnavHeight,
        left: 0,
        position: 'fixed',
        right: 0,
        top: 0,
        width: '100vw',
        zIndex: layers.topnav
    },
    _a);
