// @pluralsight/ps-design-system-verticaltabs/dist/cjs/css/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var resetButton = {
    background: 'transparent',
    border: 'none',
    color: 'inherit',
    font: 'inherit',
    lineHeight: 'normal',
    margin: 0,
    overflow: 'visible',
    padding: 0,
    width: 'auto',
    MozOsxFontSmoothing: 'inherit',
    WebkitAppearance: 'none',
    WebkitFontSmoothing: 'inherit',
    '&::-moz-focus-inner': {
        border: 0,
        padding: 0
    }
};
var truncate = {
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
};
exports.default = (_a = {
        '.psds-verticaltabs': {
            height: '100%',
            overflowY: 'auto',
            padding: ps_design_system_core_1.layout.spacingMedium + " 0",
            listStyle: 'none',
            margin: 0
        },
        '.psds-verticaltabs__divider': {
            margin: ps_design_system_core_1.layout.spacingMedium + " 0",
            borderWidth: '1px 0 0 0',
            borderStyle: 'solid'
        }
    },
    _a[".psds-verticaltabs__divider.psds-theme--" + ps_design_system_theme_1.names.light] = {
        borderColor: ps_design_system_core_1.colorsBorder.lowOnLight
    },
    _a[".psds-verticaltabs__divider.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        borderColor: ps_design_system_core_1.colorsBorder.lowOnDark
    },
    _a['.psds-verticaltabs__list'] = {
        listStyle: 'none',
        margin: 0,
        padding: 0
    },
    _a['.psds-verticaltabs__list--collapsible'] = {
        overflow: 'hidden',
        transition: "height " + ps_design_system_core_1.motion.speedNormal,
        visibility: 'hidden'
    },
    _a['.psds-verticaltabs__item'] = {
        alignItems: 'center',
        display: 'flex',
        position: 'relative',
        '& > *': __assign(__assign({}, resetButton), { color: 'inherit', textDecoration: 'none', height: '100%', width: '100%' })
    },
    _a[".psds-verticaltabs__item.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        '& > a, & > button': {
            cursor: 'pointer',
            '&:focus, &:hover': {
                backgroundColor: ps_design_system_core_1.colorsBackgroundUtility[25],
                color: ps_design_system_core_1.colorsTextIcon.highOnLight
            }
        }
    },
    _a[".psds-verticaltabs__item.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        '& > a, & > button': {
            cursor: 'pointer',
            '&:focus, &:hover': {
                backgroundColor: ps_design_system_core_1.colorsBackgroundUtility[25],
                color: ps_design_system_core_1.colorsTextIcon.highOnDark
            }
        }
    },
    _a['.psds-verticaltabs__item__icon'] = {
        flexShrink: 0,
        marginRight: ps_design_system_core_1.layout.spacingSmall
    },
    _a['.psds-verticaltabs__tier1'] = {
        fontSize: ps_design_system_core_1.type.fontSize200,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        height: '40px',
        '&:before': {
            content: ' ',
            display: 'block',
            height: 0,
            left: 0,
            position: 'absolute',
            top: '50%',
            transform: 'translateY(-50%)',
            width: '4px'
        },
        '& > *': {
            padding: ps_design_system_core_1.layout.spacingXSmall + " " + ps_design_system_core_1.layout.spacingLarge
        },
        '&[data-active]': {
            '&:before': {
                height: "calc(100% - " + ps_design_system_core_1.layout.spacingXSmall + ")",
                transition: "height " + ps_design_system_core_1.motion.speedNormal + " ease-out"
            }
        }
    },
    _a[".psds-verticaltabs__tier1.psds-theme--" + ps_design_system_theme_1.names.light] = {
        '&:before': { background: ps_design_system_core_1.colorsPrimaryAction.background },
        '&[data-active]': { color: ps_design_system_core_1.colorsTextIcon.highOnLight }
    },
    _a[".psds-verticaltabs__tier1.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        '&:before': { background: ps_design_system_core_1.colorsPrimaryAction.background },
        '&[data-active]': { color: ps_design_system_core_1.colorsTextIcon.highOnDark }
    },
    _a['.psds-verticaltabs__tier1__header'] = {
        alignItems: 'center',
        display: 'flex',
        transition: "color " + ps_design_system_core_1.motion.speedXFast + " linear",
        width: '100%',
        '& > span': __assign({}, truncate)
    },
    _a['.psds-verticaltabs__tier2'] = {
        position: 'relative',
        '&:before': {
            content: ' ',
            display: 'block',
            height: '100%',
            left: '36px',
            position: 'absolute',
            top: '50%',
            transform: 'translateY(-50%)',
            transition: "height " + ps_design_system_core_1.motion.speedXFast + " linear",
            width: '1px'
        },
        '& > *': __assign(__assign({}, truncate), { fontSize: ps_design_system_core_1.type.fontSize100, fontWeight: ps_design_system_core_1.type.fontWeightDefault, marginLeft: '37px', minHeight: '32px', padding: ps_design_system_core_1.layout.spacingXSmall + " " + ps_design_system_core_1.layout.spacingLarge, textAlign: 'left' })
    },
    _a[".psds-verticaltabs__tier2.psds-theme--" + ps_design_system_theme_1.names.light] = {
        '&:before': { background: ps_design_system_core_1.colorsBorder.lowOnLight },
        '&[data-active]': {
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            '&:before': { background: ps_design_system_core_1.colorsPrimaryAction.background }
        }
    },
    _a[".psds-verticaltabs__tier2.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        '&:before': { background: ps_design_system_core_1.colorsBorder.lowOnDark },
        '&[data-active]': {
            color: ps_design_system_core_1.colorsTextIcon.highOnDark,
            '&:before': { background: ps_design_system_core_1.colorsPrimaryAction.background }
        }
    },
    _a['.psds-verticaltabs__tier2__header'] = {
        transition: "color " + ps_design_system_core_1.motion.speedXFast + " ease-out"
    },
    _a['.psds-verticaltabs__group__header'] = {
        alignItems: 'center',
        display: 'flex',
        fontSize: '11px',
        fontWeight: ps_design_system_core_1.type.fontWeightStrong,
        textTransform: 'uppercase',
        '& > a, & > button, & > div': {
            alignItems: 'center',
            display: 'flex',
            height: '100%',
            justifyContent: 'space-between',
            padding: ps_design_system_core_1.layout.spacingSmall + " " + ps_design_system_core_1.layout.spacingLarge,
            textAlign: 'left',
            textTransform: 'inherit',
            width: '100%'
        }
    },
    _a['.psds-verticaltabs__group__header__inner'] = {},
    _a[".psds-verticaltabs__group__header.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-verticaltabs__group__header.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a['.psds-verticaltabs__header__label'] = __assign(__assign({}, truncate), { flex: 1, textAlign: 'left' }),
    _a['.psds-verticaltabs__header__label--hide-labels'] = {
        color: 'transparent'
    },
    _a['.psds-verticaltabs__header__label__icon'] = {
        flexShrink: 0,
        flexGrow: 0,
        transform: 'translateX(10px) rotateZ(180deg)',
        transition: "transform " + ps_design_system_core_1.motion.speedNormal + " linear"
    },
    _a['.psds-verticaltabs__header__label__icon--collapsed'] = {
        transform: 'translateX(10px)'
    },
    _a['.psds-verticaltabs__group__button'] = __assign(__assign({}, resetButton), { cursor: 'pointer' }),
    _a['.psds-verticaltabs__group__collapsible-list'] = {
        overflow: 'hidden',
        transition: "height " + ps_design_system_core_1.motion.speedNormal,
        visibility: 'hidden'
    },
    _a[".psds-verticaltabs__rotatable"] = {
        flexShrink: 0,
        flexGrow: 0,
        transform: 'translateX(10px)',
        transition: "transform " + ps_design_system_core_1.motion.speedNormal
    },
    _a[".psds-verticaltabs__rotatable--isOpen"] = {
        transform: 'translateX(10px) rotateZ(180deg)'
    },
    _a);
