// @pluralsight/ps-design-system-circularprogress/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sizes = void 0;
var ps_design_system_screenreaderonly_1 = __importDefault(require("@pluralsight/ps-design-system-screenreaderonly"));
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
var radius = vars.style.width / 2 - vars.style.strokeWidth / 2;
var circumference = 2 * Math.PI * radius;
var spin = glamor.keyframes({
    to: {
        transform: 'rotate(270deg)'
    }
});
var styles = {
    circularprogress: function (themeName, _a) {
        var _b = _a.size, size = _b === void 0 ? vars.sizes.medium : _b;
        return glamor.css(index_1.default[".psds-circularprogress--size-" + size]);
    },
    svg: function (themeName, _a) {
        var value = _a.value;
        var noValue = typeof value === 'undefined';
        return glamor.compose(glamor.css(index_1.default['.psds-circularprogress__svg']), noValue &&
            glamor.css(index_1.default['.psds-circularprogress__svg--no-value']({ spin: spin })));
    },
    bg: function (themeName, _props) {
        return glamor.compose(glamor.css(index_1.default['.psds-circularprogress__bg']), glamor.css(index_1.default[".psds-circularprogress__bg.psds-theme--" + themeName]));
    },
    fg: function (themeName, _props) {
        return glamor.compose(glamor.css(index_1.default['.psds-circularprogress__fg']), glamor.css(index_1.default[".psds-circularprogress__fg.psds-theme--" + themeName]));
    }
};
var CircularProgress = react_1.default.forwardRef(function (props, ref) {
    var defaultIndeterminateValue = 25;
    var size = props.size, value = props.value, ariaLabel = props["aria-label"], rest = __rest(props, ["size", "value", 'aria-label']);
    var themeName = ps_design_system_theme_1.useTheme();
    var dashOffset = ((100 - (value || defaultIndeterminateValue)) / 100) * circumference;
    var busy = value
        ? value === 100
            ? 'false'
            : value > 0
                ? 'true'
                : 'false'
        : 'true';
    var ariaAttributes = value
        ? {
            'aria-label': ariaLabel,
            'aria-valuemin': 0,
            'aria-valuemax': 100,
            'aria-valuenow': Math.round(value)
        }
        : {
            'aria-label': 'Loading'
        };
    return (react_1.default.createElement("div", __assign({ ref: ref }, styles.circularprogress(themeName, props), rest, { role: "progressbar" }, ariaAttributes, { "aria-busy": busy }),
        value ? (react_1.default.createElement(ps_design_system_screenreaderonly_1.default, { "aria-live": 'polite' }, value < 100 ? Math.round(value) + "%" : 'complete')) : null,
        react_1.default.createElement("svg", __assign({}, styles.svg(themeName, props), { viewBox: "0 0 " + vars.style.width + " " + vars.style.width, version: "1.1", xmlns: "http://www.w3.org/2000/svg" }),
            react_1.default.createElement("circle", __assign({ r: radius, cx: "24", cy: "24" }, styles.bg(themeName, props))),
            react_1.default.createElement("circle", __assign({ r: radius, cx: "24", cy: "24" }, styles.fg(themeName, props), { strokeDasharray: circumference + " " + circumference, strokeDashoffset: dashOffset })))));
});
CircularProgress.sizes = vars.sizes;
exports.sizes = vars.sizes;
exports.default = CircularProgress;
