// @pluralsight/ps-design-system-navitem/dist/cjs/react/context.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.initialValue = void 0;
var react_1 = __importDefault(require("react"));
var index_1 = require("../vars/index");
var defaultRenderContainer = react_1.default.forwardRef(function (props, ref) { return react_1.default.createElement("button", __assign({ ref: ref }, props)); });
exports.initialValue = {
    alignment: index_1.alignments.horizontal,
    menu: false,
    ref: null,
    selected: false,
    renderContainer: defaultRenderContainer
};
var Context = react_1.default.createContext(exports.initialValue);
exports.default = Context;
