// @pluralsight/ps-design-system-navitem/dist/cjs/react/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var context_1 = __importStar(require("./context"));
var horz_1 = require("./horz");
var index_1 = require("../vars/index");
var vert_1 = require("./vert");
var NavItem = react_1.default.forwardRef(function (props, forwardedRef) {
    var _a = props.alignment, alignment = _a === void 0 ? context_1.initialValue.alignment : _a, bar = props.bar, children = props.children, icon = props.icon, _b = props.menu, menu = _b === void 0 ? context_1.initialValue.menu : _b, _c = props.renderContainer, renderContainer = _c === void 0 ? context_1.initialValue.renderContainer : _c, _d = props.selected, selected = _d === void 0 ? context_1.initialValue.selected : _d, 
    // eslint-disable-next-line camelcase
    UNSAFE_stylesFor = props.UNSAFE_stylesFor, rest = __rest(props, ["alignment", "bar", "children", "icon", "menu", "renderContainer", "selected", "UNSAFE_stylesFor"]);
    var Layout = alignment === index_1.alignments.horizontal ? horz_1.HorzLayout : vert_1.VertLayout;
    var ctx = {
        alignment: alignment,
        bar: bar,
        icon: icon,
        menu: menu,
        ref: forwardedRef,
        selected: selected,
        renderContainer: renderContainer,
        UNSAFE_stylesFor: UNSAFE_stylesFor,
        rest: rest
    };
    return (react_1.default.createElement(context_1.default.Provider, { value: ctx },
        react_1.default.createElement(Layout, null, children)));
});
NavItem.displayName = 'NavItem';
NavItem.alignments = index_1.alignments;
exports.default = NavItem;
