// @pluralsight/ps-design-system-switch/dist/esm/css/index.js
var _a;
import { accessibility, colorsBackgroundUtility, colorsBlue, colorsBorder, colorsGreen, colorsTextIcon, colorsWhite, layout, motion, type } from '@pluralsight/ps-design-system-core';
import { defaultName as themeDefaultName, names as themeNames } from '@pluralsight/ps-design-system-theme';
import * as vars from '../vars/index';
export default (_a = {
        '.psds-switch': {
            display: 'inline-flex',
            alignItems: 'center',
            cursor: 'pointer',
            opacity: '1',
            background: 'none',
            border: 'none',
            padding: 0,
            '&:focus': { outline: 'none' }
        },
        '.psds-switch--disabled': {
            opacity: '0.4',
            cursor: 'not-allowed'
        }
    },
    _a[".psds-switch--labelAlign-" + vars.labelAligns.right] = {
        flexDirection: 'row'
    },
    _a[".psds-switch--labelAlign-" + vars.labelAligns.left] = {
        flexDirection: 'row-reverse'
    },
    _a['.psds-switch__track'] = {
        position: 'relative',
        backgroundColor: colorsBackgroundUtility[25],
        border: "1px solid " + colorsBorder.lowOnDark,
        transition: "background-color " + motion.speedFast + " ease-in-out"
    },
    _a[".psds-switch__track.psds-theme--" + themeNames.light] = {
        border: "1px solid " + colorsBorder.lowOnLight
    },
    _a[".psds-switch__track.psds-switch__track--size-" + vars.sizes.small] = {
        height: '12px',
        width: '24px',
        borderRadius: '6px'
    },
    _a[".psds-switch__track.psds-switch__track--size-" + vars.sizes.large] = {
        height: '24px',
        width: '48px',
        borderRadius: '12px'
    },
    _a[".psds-switch__track--checked.psds-switch__track--color-" + vars.colors.blue] = {
        backgroundColor: colorsBlue[6],
        borderColor: 'transparent'
    },
    _a[".psds-switch__track--checked.psds-switch__track--color-" + vars.colors.green] = {
        backgroundColor: colorsGreen[6],
        borderColor: 'transparent'
    },
    _a[".psds-switch__track--checked.psds-switch__track--color-" + vars.colors.orange] = {
        // NOTE: temp: make orange blue until we do a breaking change to remove orange
        backgroundColor: colorsBlue[6],
        borderColor: 'transparent'
    },
    _a[".psds-switch__thumb"] = {
        backgroundColor: colorsWhite,
        borderRadius: '50%',
        transition: "transform " + motion.speedFast + " ease-in-out"
    },
    _a[".psds-switch__thumb--size-" + vars.sizes.small] = {
        height: '10px',
        width: '10px'
    },
    _a[".psds-switch__thumb--size-" + vars.sizes.large] = {
        height: '22px',
        width: '22px'
    },
    _a[".psds-switch__thumb--checked.psds-switch__thumb--size-" + vars.sizes.small] = {
        transform: 'translateX(12px)'
    },
    _a[".psds-switch__thumb--checked.psds-switch__thumb--size-" + vars.sizes.large] = {
        transform: 'translateX(24px)'
    },
    _a[".psds-switch__label"] = {
        fontWeight: type.fontWeightDefault,
        lineHeight: '1em'
    },
    _a[".psds-switch__label--size-" + vars.sizes.small] = {
        fontSize: type.fontSize100
    },
    _a[".psds-switch__label--size-" + vars.sizes.large] = {
        fontSize: type.fontSize200
    },
    _a[".psds-switch__label--size-" + vars.sizes.small + ".psds-switch__label--labelAlign-" + vars.labelAligns.left] = {
        marginRight: layout.spacingXSmall
    },
    _a[".psds-switch__label--size-" + vars.sizes.large + ".psds-switch__label--labelAlign-" + vars.labelAligns.left] = {
        marginRight: layout.spacingMedium
    },
    _a[".psds-switch__label--size-" + vars.sizes.small + ".psds-switch__label--labelAlign-" + vars.labelAligns.right] = {
        marginLeft: layout.spacingXSmall
    },
    _a[".psds-switch__label--size-" + vars.sizes.large + ".psds-switch__label--labelAlign-" + vars.labelAligns.right] = {
        marginLeft: layout.spacingMedium
    },
    _a[".psds-switch__label.psds-theme--" + themeNames.light] = {
        color: colorsTextIcon.highOnLight
    },
    _a[".psds-switch__label.psds-theme--" + themeDefaultName] = {
        color: colorsTextIcon.highOnDark
    },
    _a['.psds-switch__checkbox'] = accessibility.screenReaderOnly,
    _a);
