// @pluralsight/ps-design-system-steps/dist/esm/css/index.js
var _a;
import { colorsBackgroundUtility, colorsPrimaryAction, colorsGreen, colorsTextIcon, layout, motion, type } from '@pluralsight/ps-design-system-core';
import { names as themeNames } from '@pluralsight/ps-design-system-theme';
import * as vars from '../vars/index';
var dark = ".psds-theme--" + themeNames.dark;
var light = ".psds-theme--" + themeNames.light;
export default (_a = {
        '.psds-steps': {
            label: 'steps',
            counterReset: 'line-number',
            display: 'flex'
        }
    },
    _a[".psds-steps--" + vars.orientations.horizontal] = {
        flexDirection: 'row',
        marginBottom: layout.spacingMedium
    },
    _a[".psds-steps--" + vars.orientations.vertical] = {
        flexDirection: 'column'
    },
    _a['.psds-steps__step'] = {
        label: 'steps__step',
        alignItems: 'flex-start',
        counterIncrement: 'line-number',
        display: 'flex',
        flex: 1,
        padding: layout.spacingXSmall
    },
    _a['.psds-steps__step--interactive'] = {
        borderRadius: 4,
        cursor: 'pointer',
        '&:hover': { backgroundColor: colorsBackgroundUtility[25] }
    },
    _a[".psds-steps__step" + dark] = {
        color: colorsTextIcon.lowOnDark
    },
    _a[".psds-steps_step" + light] = {
        color: colorsTextIcon.lowOnLight
    },
    _a[".psds-steps__step--" + vars.orientations.horizontal] = {
        marginRight: layout.spacingXSmall,
        '&:last-child': { marginRight: 0 }
    },
    _a[".psds-steps__step--" + vars.orientations.vertical] = {
        marginBottom: layout.spacingXSmall,
        '&:last-child': { marginBottom: 0 }
    },
    _a[".psds-steps__step--" + vars.sizes.medium] = {
        fontSize: type.fontSize200
    },
    _a[".psds-steps__step--" + vars.sizes.large] = {
        fontSize: type.fontSize400
    },
    _a[".psds-steps__step--" + vars.statuses.current] = {
        fontWeight: type.fontWeight700
    },
    _a['.psds-steps__title'] = {
        '& p:last-child': { marginBottom: 0 }
    },
    _a[".psds-steps__title--" + vars.sizes.large] = {
        paddingTop: 5 // NOTE: align text vertically with marker
    },
    _a[".psds-steps__title--" + vars.sizes.medium] = {
        paddingTop: 1 // NOTE: align text vertically with marker
    },
    _a[".psds-steps__title--" + vars.statuses.current + dark] = {
        color: colorsTextIcon.highOnDark
    },
    _a[".psds-steps__title--" + vars.statuses.current + light] = {
        color: colorsTextIcon.highOnLight
    },
    _a['.psds-steps__description'] = {
        '& p:last-child': { marginBottom: 0 }
    },
    _a[".psds-steps__description--" + vars.sizes.medium] = {
        fontSize: type.fontSize100
    },
    _a[".psds-steps__description--" + vars.sizes.large] = {
        fontSize: type.fontSize200
    },
    _a[".psds-steps__description--" + vars.statuses.current] = {
        fontWeight: type.fontWeightDefault
    },
    _a['.psds-steps__marker-container'] = {
        label: 'steps__marker-container',
        flexShrink: 0
    },
    _a[".psds-steps__marker-container--" + vars.sizes.medium] = {
        marginRight: layout.spacingXSmall
    },
    _a[".psds-steps__marker-container--" + vars.sizes.large] = {
        marginRight: layout.spacingSmall
    },
    _a['.psds-steps__marker'] = {
        label: 'steps__marker',
        fontSize: type.fontSize100,
        position: 'relative',
        textAlign: 'center',
        '&:after': {
            content: 'counter(line-number)',
            left: '50%',
            margin: 'auto',
            opacity: 1,
            position: 'absolute',
            top: '50%',
            transform: 'translate(-50%, -50%)',
            transition: "all " + motion.speedFast,
            width: '50%'
        }
    },
    _a['.psds-steps__marker--hide-counter'] = {
        '&:after': { content: '' }
    },
    _a[".psds-steps__marker" + dark] = {
        color: colorsTextIcon.lowOnDark
    },
    _a[".psds-steps__marker" + light] = {
        color: colorsTextIcon.lowOnLight
    },
    _a[".psds-steps__marker--" + vars.sizes.medium] = {
        fontSize: type.fontSize100
    },
    _a[".psds-steps__marker--" + vars.sizes.large] = {
        fontSize: type.fontSize400
    },
    _a[".psds-steps__marker--" + vars.statuses.completed] = {
        color: colorsGreen[6],
        '&:after': { opacity: 0 }
    },
    _a[".psds-steps__marker--" + vars.statuses.current + dark] = {
        color: colorsPrimaryAction.textDarkTheme
    },
    _a[".psds-steps__marker--" + vars.statuses.current + light] = {
        color: colorsPrimaryAction.textLightTheme
    },
    _a['.psds-steps__marker__circle'] = {
        label: 'steps__marker__circle',
        fill: 'transparent',
        stroke: 'currentColor'
    },
    _a['.psds-steps__marker__check'] = {
        label: 'steps__marker__check',
        fill: 'transparent',
        stroke: 'currentColor',
        strokeDasharray: 1000,
        strokeDashoffset: 1000,
        strokeLinecap: 'round',
        strokeLinejoin: 'round',
        transition: "stroke-dashoffset 0s linear 0s"
    },
    _a['.psds-steps__marker__check--completed'] = {
        strokeDashoffset: 0,
        transition: "stroke-dashoffset 3s linear " + motion.speedNormal
    },
    _a);
