// @codecademy/gamut/dist/GridForm/GridFormSections/__fixtures__/renderers.js
import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { GridFormContent } from '../GridFormContent';
import { GridFormSection } from '../GridFormSection.tsx';
export var GridFormContentTestComponent = function GridFormContentTestComponent(_ref) {
  var field = _ref.field,
      showRequired = _ref.showRequired,
      _ref$mode = _ref.mode,
      mode = _ref$mode === void 0 ? 'onSubmit' : _ref$mode;
  var methods = useForm({
    mode: mode
  });
  return /*#__PURE__*/React.createElement(FormProvider, methods, /*#__PURE__*/React.createElement(GridFormContent, {
    field: field,
    showRequired: showRequired,
    key: field.name
  }));
};
export var GridFormSectionTestComponent = function GridFormSectionTestComponent(_ref2) {
  var fields = _ref2.fields,
      showRequired = _ref2.showRequired,
      _ref2$mode = _ref2.mode,
      mode = _ref2$mode === void 0 ? 'onSubmit' : _ref2$mode;
  var methods = useForm({
    mode: mode
  });
  return /*#__PURE__*/React.createElement(FormProvider, methods, /*#__PURE__*/React.createElement(GridFormSection, {
    fields: fields,
    showRequired: showRequired
  }));
};
