// @pluralsight/ps-design-system-dropdown/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '@keyframes psds-dropdown__menu__keyframes__slide': {
            '100%': {
                transform: 'translate(0,0)',
                opacity: 1
            }
        },
        '.psds-dropdown': {
            display: 'inline-flex',
            flexDirection: 'column',
            position: 'relative'
        },
        '.psds-dropdown--disabled': {
            opacity: 0.5
        },
        // __field
        '.psds-dropdown__field': {
            position: 'relative',
            height: '40px',
            width: '100%',
            border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnDark,
            borderRadius: '2px',
            background: ps_design_system_core_1.colorsBackgroundLight[3],
            fontSize: ps_design_system_core_1.type.fontSize200,
            lineHeight: ps_design_system_core_1.type.lineHeightStandard,
            fontWeight: ps_design_system_core_1.type.fontWeightDefault,
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            textAlign: 'left',
            cursor: 'pointer',
            padding: "0 " + ps_design_system_core_1.layout.spacingSmall,
            '&:focus': {
                outline: 'none'
            }
        },
        '.psds-dropdown__field.psds-dropdown--small': {
            height: '32px',
            padding: "0 " + ps_design_system_core_1.layout.spacingXSmall
        }
    },
    _a[".psds-dropdown__field.psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnLight,
        '&:focus': {
            border: '1px solid transparent'
        }
    },
    _a[".psds-dropdown__field--error.psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: '1px solid transparent'
    },
    _a[".psds-dropdown__field--appearance-" + vars.appearances.subtle] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        backgroundColor: ps_design_system_core_1.colorsBackgroundUtility[30],
        border: "none"
    },
    _a['.psds-dropdown__field-inner'] = {
        position: 'relative',
        height: '100%',
        display: 'inline-flex',
        alignItems: 'center',
        textAlign: 'left',
        minWidth: 0,
        width: '100%'
    },
    // __field-container
    _a['.psds-dropdown__field-container'] = {
        position: 'relative',
        display: 'inline-block',
        alignItems: 'center',
        width: '100%',
        minWidth: "calc(" + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + ")",
        maxWidth: '100%'
    },
    _a['.psds-dropdown__field-aligner'] = {
        display: 'inline-flex',
        alignItems: 'center',
        width: '100%',
        maxWidth: '100%'
    },
    _a['.psds-dropdown__divider'] = {
        height: '1px',
        width: '100%',
        backgroundColor: ps_design_system_core_1.colorsBorder.lowOnLight,
        margin: ps_design_system_core_1.layout.spacingXXSmall + " 0"
    },
    // __icon
    _a['.psds-dropdown__icon'] = {
        position: 'absolute',
        left: 'auto',
        top: '50%',
        transform: 'translateY(-50%)',
        right: ps_design_system_core_1.layout.spacingXSmall,
        display: 'flex',
        alignItems: 'center',
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        cursor: 'pointer'
    },
    _a[".psds-dropdown__icon--appearance-" + vars.appearances.subtle] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    // __label
    _a['.psds-dropdown__label'] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        marginBottom: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-dropdown__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    // __sub-label
    _a['.psds-dropdown__sub-label'] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeightDefault,
        marginTop: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-dropdown__sub-label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    // __error
    _a['.psds-dropdown__error'] = {
        position: 'absolute',
        right: 0,
        top: '50%',
        transform: "translate(calc(100% + " + ps_design_system_core_1.layout.spacingXSmall + "), -50%)",
        display: 'flex',
        alignItems: 'center',
        color: ps_design_system_core_1.colorsStatus.error,
        marginLeft: ps_design_system_core_1.layout.spacingXSmall
    },
    // __placeholder
    _a['.psds-dropdown__placeholder'] = {
        padding: "0 calc(" + ps_design_system_core_1.layout.spacingXSmall + " + " + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + ") 0 0",
        position: 'absolute',
        top: ps_design_system_core_1.layout.spacingXSmall,
        left: 0,
        width: '100%',
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis'
    },
    _a['.psds-dropdown__placeholder--color'] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-dropdown__field--appearance-" + vars.appearances.subtle + ".psds-dropdown__placeholder--color"] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a['.psds-dropdown__placeholder.psds-dropdown--small'] = {
        top: ps_design_system_core_1.layout.spacingXXSmall
    },
    // __button-sizer
    _a['.psds-dropdown__button-sizer'] = {
        visibility: 'hidden',
        display: 'inline-block',
        whiteSpace: 'nowrap',
        padding: "0 calc(" + ps_design_system_core_1.layout.spacingXSmall + " + " + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + ") 0 0"
    },
    // __menu
    _a['.psds-dropdown__menu-wrapper'] = {
        position: 'fixed',
        zIndex: ps_design_system_core_1.layers.globalDropdown,
        marginTop: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a['.psds-dropdown__menu'] = {
        maxHeight: 400,
        overflowY: 'auto',
        display: 'block',
        background: ps_design_system_core_1.colorsBackgroundLight[3],
        borderRadius: '2px',
        padding: ps_design_system_core_1.layout.spacingXXSmall + " 0",
        boxShadow: "0 2px 4px rgba(0, 0, 0, 0.5)",
        fontSize: ps_design_system_core_1.type.fontSize200,
        opacity: 0,
        transform: "translate(0, calc(-1 * " + ps_design_system_core_1.layout.spacingXSmall + "))",
        right: 0,
        top: 0
    },
    _a['.psds-dropdown__menu__animation'] = function (_a) {
        var slide = _a.slide;
        return ({
            animation: (slide || 'psds-dropdown__menu__keyframes__slide') + " " + ps_design_system_core_1.motion.speedNormal + " forwards"
        });
    },
    _a['.psds-dropdown__field-halo'] = {
        maxWidth: '100%',
        width: '100%',
        display: 'block'
    },
    _a['.psds-dropdown__item'] = {
        label: 'dropdown__item',
        display: 'flex',
        alignItems: 'center',
        textAlign: 'left',
        width: '100%',
        transition: "background " + ps_design_system_core_1.motion.speedXFast,
        color: ps_design_system_core_1.colorsTextIcon.highOnLight,
        lineHeight: ps_design_system_core_1.type.lineHeightExtra,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        cursor: 'pointer',
        border: 'none',
        paddingTop: '0',
        paddingBottom: '0',
        paddingLeft: ps_design_system_core_1.layout.spacingMedium,
        paddingRight: ps_design_system_core_1.layout.spacingMedium,
        background: 'none',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            outline: 'none'
        },
        '&:hover:not(:disabled)': {
            background: ps_design_system_core_1.colorsBackgroundUtility[25],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        }
    },
    _a['.psds-dropdown__item--disabled'] = {
        opacity: '50%',
        cursor: 'not-allowed',
        outline: 'none',
        pointerEvents: 'none'
    },
    _a['.psds-dropdown__item--active'] = {
        '&:not(:disabled)': {
            background: ps_design_system_core_1.colorsBackgroundUtility[25],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        }
    },
    _a['.psds-dropdown__item-icon'] = {
        display: 'inline-flex',
        alignItems: 'center',
        marginRight: ps_design_system_core_1.layout.spacingXSmall
    },
    _a['.psds-dropdown__item-text'] = {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        '&:not(:last-child)': {
            marginRight: ps_design_system_core_1.layout.spacingXSmall
        }
    },
    _a['.psds-dropdown__item-selected-icon'] = {
        marginLeft: 'auto',
        color: ps_design_system_core_1.colorsBlue[7]
    },
    _a);
