// @pluralsight/ps-design-system-dropdown/dist/cjs/react/button.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Button = void 0;
var ps_design_system_halo_1 = __importDefault(require("@pluralsight/ps-design-system-halo"));
var ps_design_system_icon_1 = __importDefault(require("@pluralsight/ps-design-system-icon"));
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var caret_down_1 = require("./caret-down");
var error_icon_1 = require("./error-icon");
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    field: function (_a, themeName) {
        var appearance = _a.appearance, error = _a.error, size = _a.size;
        var label = 'psds-dropdown__field';
        var isSmall = size === vars.sizes.small;
        return glamor.compose(glamor.css(index_1.default["." + label]), isSmall && glamor.css(index_1.default["." + label + ".psds-dropdown--small"]), glamor.css(index_1.default["." + label + "--appearance-" + appearance]), glamor.css(index_1.default["." + label + ".psds-theme--" + themeName]), error &&
            glamor.css(index_1.default["." + label + "-error.psds-theme--" + themeName]));
    },
    fieldAligner: function () { return glamor.css(index_1.default['.psds-dropdown__field-aligner']); },
    fieldContainer: function () {
        return glamor.css(index_1.default['.psds-dropdown__field-container']);
    },
    halo: function () { return glamor.css(index_1.default['.psds-dropdown__field-halo']); },
    icon: function (appearance, themeName) {
        var label = 'psds-dropdown__icon';
        return glamor.compose(glamor.css(index_1.default["." + label]), glamor.css(index_1.default["." + label + "--appearance-" + appearance]), glamor.css(index_1.default["." + label + ".psds-theme--" + themeName]));
    },
    inner: function () { return glamor.css(index_1.default['.psds-dropdown__field-inner']); }
};
exports.Button = react_1.default.forwardRef(function (_a, ref) {
    var appearance = _a.appearance, children = _a.children, disabled = _a.disabled, error = _a.error, isOpen = _a.isOpen, onClick = _a.onClick, setMenuPosition = _a.setMenuPosition, size = _a.size, rest = __rest(_a, ["appearance", "children", "disabled", "error", "isOpen", "onClick", "setMenuPosition", "size"]);
    var themeName = ps_design_system_theme_1.useTheme();
    var fieldContainerRef = react_1.default.useRef(null);
    react_1.default.useLayoutEffect(function () {
        if (!isOpen || !fieldContainerRef.current)
            return;
        var _a = fieldContainerRef.current.getBoundingClientRect(), left = _a.left, bottom = _a.bottom, width = _a.width;
        var requestId = requestAnimationFrame(function () {
            return setMenuPosition({ left: left, top: bottom, width: width });
        });
        return function () { return cancelAnimationFrame(requestId); };
    }, [fieldContainerRef, isOpen, setMenuPosition]);
    return (react_1.default.createElement("div", __assign({}, styles.fieldContainer(), { ref: fieldContainerRef }),
        react_1.default.createElement(ps_design_system_halo_1.default, __assign({ error: error, gapSize: ps_design_system_halo_1.default.gapSizes.small }, styles.halo()),
            react_1.default.createElement("div", __assign({}, styles.fieldAligner()),
                react_1.default.createElement("button", __assign({}, rest, styles.field({ appearance: appearance, error: error, size: size }, themeName), { disabled: disabled, onClick: disabled ? undefined : onClick, ref: ref }),
                    react_1.default.createElement("span", __assign({}, styles.inner()), children),
                    react_1.default.createElement("div", __assign({}, styles.icon(appearance, themeName)),
                        react_1.default.createElement(ps_design_system_icon_1.default, null,
                            react_1.default.createElement(caret_down_1.CaretDown, null)))))),
        react_1.default.createElement(error_icon_1.ErrorIcon, { error: !!error })));
});
exports.Button.displayName = 'Dropdown.Button';
