// @pluralsight/ps-design-system-core/dist/cjs/js/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
var accessibility_1 = require("./accessibility");
Object.defineProperty(exports, "accessibility", { enumerable: true, get: function () { return accessibility_1.default; } });
var breakpoints_1 = require("./breakpoints");
Object.defineProperty(exports, "breakpoints", { enumerable: true, get: function () { return breakpoints_1.default; } });
var layers_1 = require("./layers");
Object.defineProperty(exports, "layers", { enumerable: true, get: function () { return layers_1.default; } });
var layout_1 = require("./layout");
Object.defineProperty(exports, "layout", { enumerable: true, get: function () { return layout_1.default; } });
var type_1 = require("./type");
Object.defineProperty(exports, "type", { enumerable: true, get: function () { return type_1.default; } });
var motion_1 = require("./motion");
Object.defineProperty(exports, "motion", { enumerable: true, get: function () { return motion_1.default; } });
__exportStar(require("./colors"), exports);
