// @pluralsight/ps-design-system-checkbox/dist/cjs/css/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
exports.default = (_a = {
        '.psds-checkbox': {
            display: 'flex',
            alignItems: 'center',
            padding: ps_design_system_core_1.layout.spacingXXSmall + " 0",
            cursor: 'pointer'
        },
        '.psds-checkbox--disabled': {
            opacity: '0.5',
            cursor: 'not-allowed'
        },
        // __square
        '.psds-checkbox__square': {
            position: 'relative',
            display: 'inline-flex',
            alignItems: 'center',
            height: '16px',
            width: '16px',
            borderRadius: '2px',
            border: "2px solid " + ps_design_system_core_1.colorsTextIcon.lowOnDark,
            background: ps_design_system_core_1.colorsBackgroundDark[1],
            color: ps_design_system_core_1.colorsWhite
        }
    },
    _a[".psds-checkbox__square.psds-theme--" + ps_design_system_theme_1.names.light] = {
        background: ps_design_system_core_1.colorsBackgroundLight[3],
        borderColor: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        '&:focus': { outline: 'none' }
    },
    _a['.psds-checkbox__square:focus:before .psds-checkbox__square--error:before'] = {
        content: ' ',
        position: 'absolute',
        top: '-3px',
        left: '-3px',
        right: '-3px',
        bottom: '-3px',
        zIndex: '-1',
        borderRadius: '2px'
    },
    _a[".psds-checkbox__square.psds-theme--" + ps_design_system_theme_1.names.light + ":focus:before .psds-checkbox__square--error.psds-theme--" + ps_design_system_theme_1.names.light + ":before"] = {
        top: '-4px',
        left: '-4px',
        right: '-4px',
        bottom: '-4px'
    },
    _a['.psds-checkbox__square:focus:after .psds-checkbox__square--error:after'] = {
        content: ' ',
        position: 'absolute',
        top: '-6px',
        left: '-6px',
        right: '-6px',
        bottom: '-6px',
        zIndex: '-2',
        borderRadius: '4px'
    },
    _a[".psds-checkbox__square.psds-theme--" + ps_design_system_theme_1.names.light + ":focus:after .psds-checkbox__square--error.psds-theme--" + ps_design_system_theme_1.names.light + ":after"] = {
        top: '-7px',
        left: '-7px',
        right: '-7px',
        bottom: '-7px'
    },
    _a['.psds-checkbox__square:focus:after'] = {
        background: ps_design_system_core_1.colorsBlue[6]
    },
    _a['.psds-checkbox__square--error:after'] = {
        background: ps_design_system_core_1.colorsStatus.error
    },
    _a['.psds-checkbox__square--active'] = {
        background: ps_design_system_core_1.colorsBlue[6],
        border: 'none'
    },
    _a['.psds-checkbox__square--active:focus:before, .psds-checkbox__square--active--error:before'] = {
        top: '-1px',
        left: '-1px',
        right: '-1px',
        bottom: '-1px'
    },
    _a[".psds-checkbox__square--active.psds-theme--" + ps_design_system_theme_1.names.light + ":focus:before, .psds-checkbox__square--active--error.psds-theme--" + ps_design_system_theme_1.names.light + ":before"] = {
        top: '-2px',
        left: '-2px',
        right: '-2px',
        bottom: '-2px'
    },
    _a['.psds-checkbox__square--active:focus:after, .psds-checkbox__square--active--error:after'] = {
        top: '-4px',
        left: '-4px',
        right: '-4px',
        bottom: '-4px'
    },
    _a[".psds-checkbox__square--active.psds-theme--" + ps_design_system_theme_1.names.light + ":focus:after, .psds-checkbox__square--active--error.psds-theme--" + ps_design_system_theme_1.names.light + ":after"] = {
        top: '-5px',
        left: '-5px',
        right: '-5px',
        bottom: '-5px'
    },
    // __icon
    _a['.psds-checkbox__icon'] = {
        height: '100%',
        width: '100%',
        fill: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    // __label
    _a['.psds-checkbox__label'] = {
        marginLeft: ps_design_system_core_1.layout.spacingSmall,
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        fontSize: ps_design_system_core_1.type.fontSize200,
        lineHeight: ps_design_system_core_1.type.lineHeightStandard,
        fontWeight: ps_design_system_core_1.type.fontWeightDefault
    },
    _a[".psds-checkbox__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a);
