// @pluralsight/ps-design-system-halo/dist/cjs/react/__stories__/focusable.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var core = __importStar(require("@pluralsight/ps-design-system-core"));
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var glamor = glamor_1.default || glamorExports;
var stylesheet = {
    '.focusable': {
        background: core.colorsBackgroundDark[3],
        borderRadius: '2px',
        color: core.colorsTextIcon.highOnDark,
        cursor: 'pointer',
        fontSize: core.type.fontSize200,
        fontWeight: core.type.fontWeightDefault,
        lineHeight: core.type.lineHeightStandard,
        outline: 'none',
        padding: core.layout.spacingXSmall + " " + core.layout.spacingMedium,
        position: 'relative',
        textAlign: 'center',
        width: '100%'
    },
    '.focusable--theme-light': {
        background: core.colorsBackgroundLight[1],
        borderColor: core.colorsBorder.highOnLight,
        color: core.colorsTextIcon.highOnLight
    },
    '.focusable--shape-pill': {
        borderRadius: '1000px'
    }
};
var styles = {
    focusable: function (themeName, _a) {
        var shape = _a.shape;
        return glamor.css(stylesheet['.focusable'], stylesheet[".focusable--theme-" + themeName], stylesheet[".focusable--shape-" + shape]);
    }
};
var Focusable = function (props) {
    var themeName = ps_design_system_theme_1.useTheme();
    var shape = props.shape, _a = props.children, children = _a === void 0 ? 'focusable area' : _a;
    return (react_1.default.createElement("div", __assign({}, styles.focusable(themeName, { shape: shape }), { tabIndex: 0 }), children));
};
exports.default = Focusable;
