// @codecademy/gamut/dist/GridForm/GridFormInputGroup/GridFormRadioGroupInput/index.js
import { isString } from 'lodash';
import React from 'react';
import { Radio, RadioGroup } from '../../../Form';
export var GridFormRadioGroupInput = function GridFormRadioGroupInput(_ref) {
  var _field$ariaLabel;

  var className = _ref.className,
      field = _ref.field,
      register = _ref.register,
      setValue = _ref.setValue,
      showRequired = _ref.showRequired;
  var ariaLabel = (_field$ariaLabel = field.ariaLabel) !== null && _field$ariaLabel !== void 0 ? _field$ariaLabel : isString(field.label) ? field.label : undefined;
  return /*#__PURE__*/React.createElement(RadioGroup, {
    className: className,
    htmlForPrefix: field.name,
    name: field.name,
    role: "radiogroup",
    "aria-label": ariaLabel,
    "aria-required": showRequired,
    onChange: function onChange(event) {
      var _field$onUpdate;

      var value = event.target.value;
      setValue(field.name, value);
      (_field$onUpdate = field.onUpdate) === null || _field$onUpdate === void 0 ? void 0 : _field$onUpdate.call(field, value);
    }
  }, field.options.map(function (_ref2) {
    var label = _ref2.label,
        value = _ref2.value;
    return /*#__PURE__*/React.createElement(Radio, {
      disabled: field.disabled,
      key: value,
      label: label,
      ref: register(field.validation),
      value: value,
      id: field.id
    });
  }));
};
