// @codecademy/gamut-styles/dist/variables/typography.js
import { pxRem } from '../styles/pxRem';
export var fontAccent = "\"Suisse\", \"Apercu\", -apple-system, BlinkMacSystemFont,\n\"Segoe UI\", \"Roboto\", \"Ubuntu\", \"Cantarell\", \"Fira Sans\", \"Droid Sans\",\n\"Helvetica Neue\", sans-serif";
export var fontBase = "\"Apercu\", -apple-system, BlinkMacSystemFont, \"Segoe UI\",\n\"Roboto\", \"Ubuntu\", \"Cantarell\", \"Fira Sans\", \"Droid Sans\", \"Helvetica Neue\",\nsans-serif";
export var fontMonospace = "Monaco, Menlo, \"Ubuntu Mono\", \"Droid Sans Mono\", Consolas,\nmonospace";
export var fontSystem = "-apple-system, BlinkMacSystemFont, \"Segoe UI\", \"Roboto\", \"Ubuntu\",\n\"Cantarell\", \"Fira Sans\", \"Droid Sans\", \"Helvetica Neue\", sans-serif";
export var fontFamily = {
  accent: fontAccent,
  base: fontBase,
  monospace: fontMonospace,
  system: fontSystem
};
export var fontSize = {
  64: pxRem(64),
  44: pxRem(44),
  34: pxRem(34),
  26: pxRem(26),
  22: pxRem(22),
  20: pxRem(20),
  18: pxRem(18),
  16: pxRem(16),
  14: pxRem(14)
};
export var lineHeight = {
  base: 1.5,
  title: 1.1
};
export var fontWeight = {
  base: 400,
  title: 700,
  700: 700,
  400: 400
};
