// @pluralsight/ps-design-system-navcookiebanner/dist/cjs/css/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
exports.default = {
    '.psds-navcookiebanner': {
        alignItems: 'center',
        background: ps_design_system_core_1.colorsBackgroundDark[3],
        borderRadius: 2,
        bottom: ps_design_system_core_1.layout.spacingSmall,
        boxShadow: '0px 0px 5px 0px rgba(0, 0, 0, 0.65)',
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        display: 'flex',
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        justifyContent: 'center',
        left: ps_design_system_core_1.layout.spacingSmall,
        lineHeight: ps_design_system_core_1.type.lineHeightTight,
        paddingBottom: ps_design_system_core_1.layout.spacingXSmall,
        paddingLeft: ps_design_system_core_1.layout.spacingMedium,
        paddingRight: ps_design_system_core_1.layout.spacingXSmall,
        paddingTop: ps_design_system_core_1.layout.spacingXSmall,
        position: 'fixed',
        right: ps_design_system_core_1.layout.spacingSmall,
        zIndex: 1002,
        '& a': {
            color: 'inherit',
            textDecoration: 'underline',
            '&:visited': { color: 'inherit' }
        }
    },
    '.psds-navcookiebanner__message': {
        flexGrow: 1,
        marginBottom: 0,
        marginRight: ps_design_system_core_1.layout.spacingSmall
    },
    '.psds-navcookiebanner__dismiss': {
        cursor: 'pointer',
        background: 'transparent',
        border: 'none',
        color: 'inherit',
        font: 'inherit',
        lineHeight: 'normal',
        margin: 0,
        overflow: 'visible',
        padding: 0,
        width: 'auto',
        MozOsxFontSmoothing: 'inherit',
        WebkitAppearance: 'none',
        WebkitFontSmoothing: 'inherit',
        '&::-moz-focus-inner': {
            border: 0,
            padding: 0
        }
    },
    '@media (min-width: 769px)': {
        '.psds-navcookiebanner': {
            right: 'auto'
        }
    },
    '@media print': {
        '.psds-navcookiebanner': {
            display: 'none'
        }
    }
};
