// @pluralsight/ps-design-system-datepicker/dist/cjs/react/calendar.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calendar = void 0;
/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
var ps_design_system_button_1 = __importDefault(require("@pluralsight/ps-design-system-button"));
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_theme_1 = __importDefault(require("@pluralsight/ps-design-system-theme"));
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var context_1 = require("./context");
var index_1 = __importDefault(require("../css/index"));
var index_2 = require("../vars/index");
var glamor = glamor_1.default || glamorExports;
var forward = glamor.keyframes(index_1.default['@keyframes psds-calendar__keyframes__forward']);
var backward = glamor.keyframes(index_1.default['@keyframes psds-calendar__keyframes__backward']);
var styles = {
    calendar: function () { return glamor.css(index_1.default['.psds-calendar']); },
    headerWrapper: function () { return glamor.css(index_1.default[".psds-calendar__header-wrapper"]); },
    gridWrapper: function () { return glamor.css(index_1.default[".psds-calendar__grid-wrapper"]); },
    gridSlide: function (slide) {
        return glamor.css(index_1.default[".psds-calendar__grid-slide"], slide === 'forward' &&
            index_1.default[".psds-calendar__grid-slide--forward"](forward), slide === 'backward' &&
            index_1.default[".psds-calendar__grid-slide--backward"](backward));
    },
    month: function () { return glamor.css(index_1.default['.psds-calendar__month']); },
    header: function () { return glamor.css(index_1.default['.psds-calendar__header']); },
    headerButton: function () { return glamor.css(index_1.default['.psds-calendar__header-button']); },
    headerMonth: function () { return glamor.css(index_1.default['.psds-calendar__header-month']); },
    weekdayHeader: function () { return glamor.css(index_1.default['.psds-calendar__weekday-header']); },
    dateFiller: function () { return glamor.css(index_1.default['.psds-calendar__filler']); },
    dateGrid: function () { return glamor.css(index_1.default['.psds-calendar__date-grid']); }
};
var monthNamesShort = [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
];
var weekdayNamesShort = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
exports.Calendar = react_1.default.forwardRef(function (_a, ref) {
    var calendars = _a.calendars, getBackProps = _a.getBackProps, getForwardProps = _a.getForwardProps, children = _a.children, slide = _a.slide, rest = __rest(_a, ["calendars", "getBackProps", "getForwardProps", "children", "slide"]);
    var _b = react_1.default.useState(slide), _slide = _b[0], setSlide = _b[1];
    var _c = react_1.default.useState(), height = _c[0], setHeight = _c[1];
    var _d = getBackProps({ calendars: calendars }), onBackClick = _d.onClick, backRest = __rest(_d, ["onClick"]);
    var _e = getForwardProps({
        calendars: calendars
    }), onForwardClick = _e.onClick, forwardRest = __rest(_e, ["onClick"]);
    var handleForwardClick = function (e) {
        onForwardClick(e);
        setSlide(index_2.slides.forward);
    };
    var handleBackClick = function (e) {
        onBackClick(e);
        setSlide(index_2.slides.backward);
    };
    var animationRef = react_1.default.useRef();
    react_1.default.useEffect(function () {
        var el = animationRef.current;
        if (el) {
            var height_1 = el.getBoundingClientRect().height;
            setHeight(height_1);
        }
    }, [_slide, slide]);
    react_1.default.useEffect(function () {
        var el = animationRef.current;
        if (el) {
            var height_2 = el.getBoundingClientRect().height;
            setHeight(height_2);
            var updateOffset_1 = function () { return setSlide(undefined); };
            el.addEventListener('animationend', updateOffset_1);
            return function () { return el.removeEventListener('animationend', updateOffset_1); };
        }
    }, []);
    if (calendars.length) {
        return (react_1.default.createElement("div", __assign({}, styles.calendar(), rest, { ref: ref }),
            react_1.default.createElement("div", __assign({}, styles.headerWrapper()), calendars.map(function (calendar, i) { return (react_1.default.createElement("div", __assign({ key: "" + calendar.month + calendar.year }, styles.header()),
                react_1.default.createElement(ps_design_system_theme_1.default, { name: ps_design_system_theme_1.default.names.light },
                    react_1.default.createElement("div", __assign({}, styles.month()),
                        i === 0 ? (react_1.default.createElement(ps_design_system_button_1.default, __assign({}, backRest, styles.headerButton(), { onClick: handleBackClick, icon: react_1.default.createElement(ps_design_system_icon_1.CaretLeftIcon, null), appearance: ps_design_system_button_1.default.appearances.flat }))) : (react_1.default.createElement("div", __assign({}, styles.headerButton()))),
                        react_1.default.createElement("div", __assign({ key: "" + calendar.month + calendar.year }, styles.headerMonth()),
                            monthNamesShort[calendar.month],
                            " ",
                            calendar.year),
                        calendars.length - 1 === i ? (react_1.default.createElement(ps_design_system_button_1.default, __assign({}, forwardRest, styles.headerButton(), { onClick: handleForwardClick, icon: react_1.default.createElement(ps_design_system_icon_1.CaretRightIcon, null), appearance: ps_design_system_button_1.default.appearances.flat }))) : (react_1.default.createElement("div", __assign({}, styles.headerButton()))))),
                weekdayNamesShort.map(function (weekday) { return (react_1.default.createElement("div", __assign({ key: "" + calendar.month + calendar.year + weekday }, styles.weekdayHeader()), weekday)); }))); })),
            react_1.default.createElement("div", __assign({ style: { height: height } }, styles.gridWrapper()),
                react_1.default.createElement("div", __assign({}, styles.gridSlide(_slide), { ref: animationRef }), calendars.map(function (calendar, i) { return (react_1.default.createElement("div", __assign({}, styles.dateGrid(), { key: "" + calendar.month + calendar.year }),
                    react_1.default.createElement(context_1.DateContext.Provider, { value: calendar }, children))); })))));
    }
    return null;
});
