// @pluralsight/ps-design-system-datepicker/dist/cjs/react/datepicker.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatePicker = void 0;
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var dayzed_1 = require("dayzed");
var react_1 = __importDefault(require("react"));
var calendar_1 = require("./calendar");
var calendar_dates_1 = require("./calendar-dates");
var text_input_field_1 = require("./text-input-field");
var utils_1 = require("./utils");
exports.DatePicker = function (_a) {
    var disabled = _a.disabled, error = _a.error, label = _a.label, prefix = _a.prefix, renderContainer = _a.renderContainer, renderTag = _a.renderTag, size = _a.size, subLabel = _a.subLabel, suffix = _a.suffix, onSelect = _a.onSelect, valueFromProps = _a.value, _uniqueId = _a._uniqueId, props = __rest(_a, ["disabled", "error", "label", "prefix", "renderContainer", "renderTag", "size", "subLabel", "suffix", "onSelect", "value", "_uniqueId"]);
    var _b = react_1.default.useState(valueFromProps), selected = _b[0], setSelected = _b[1];
    react_1.default.useEffect(function () { return setSelected(valueFromProps); }, [valueFromProps]);
    var _c = react_1.default.useState(false), open = _c[0], setOpen = _c[1];
    var onDateSelected = function (dateObj, evt) {
        onSelect && onSelect(evt, dateObj);
        setSelected(dateObj.date);
        setOpen(false);
    };
    var _d = dayzed_1.useDayzed({
        date: selected,
        selected: selected,
        onDateSelected: onDateSelected
    }), getDateProps = _d.getDateProps, dayzedData = __rest(_d, ["getDateProps"]);
    var handleIconClick = function (evt) {
        setOpen(!open);
    };
    var _e = react_1.default.useState(), slide = _e[0], setSlide = _e[1];
    var _f = utils_1.useDateSelectChange({
        selected: selected,
        setSlide: setSlide,
        setSelected: setSelected
    }), value = _f[0], onChange = _f[1];
    var ref = react_1.default.useRef();
    react_1.default.useEffect(function () {
        if (!ps_design_system_util_1.canUseDOM())
            return function () { };
        var handleClickOutsideMenu = function (evt) {
            if (evt.target instanceof HTMLElement) {
                // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
                if (ref.current.contains(evt.target))
                    return;
                setOpen(false);
            }
        };
        document.addEventListener('click', handleClickOutsideMenu, {
            capture: true
        });
        return function () {
            return document.removeEventListener('click', handleClickOutsideMenu, {
                capture: true
            });
        };
    }, [setOpen]);
    return (react_1.default.createElement("div", __assign({ style: { display: 'inline-block', position: 'relative' } }, props, { ref: ref }),
        react_1.default.createElement(text_input_field_1.TextInputField, { disabled: disabled, error: error, label: label, onChange: onChange, onClick: handleIconClick, placeholder: "mm/dd/yyyy", prefix: prefix, renderContainer: renderContainer, renderTag: renderTag, size: size, subLabel: subLabel, suffix: suffix, value: value, _uniqueId: _uniqueId }),
        react_1.default.createElement("br", null),
        open && (react_1.default.createElement(calendar_1.Calendar, __assign({}, dayzedData, { style: { position: 'absolute', zIndex: 1, marginTop: 4 }, slide: slide }),
            react_1.default.createElement(calendar_dates_1.CalendarDates, { getDateProps: getDateProps }, function (renderProps) {
                return react_1.default.createElement("button", __assign({}, renderProps));
            })))));
};
