// @pluralsight/ps-design-system-datepicker/dist/cjs/react/utils.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useRangeSelectChange = exports.useDateSelectChange = exports.onMultiDateSelected = exports.onRangeDateSelected = exports.useIsInRange = void 0;
var differenceInCalendarMonths_1 = __importDefault(require("date-fns/differenceInCalendarMonths"));
var format_1 = __importDefault(require("date-fns/format"));
var parse_1 = __importDefault(require("date-fns/parse"));
var isMatch_1 = __importDefault(require("date-fns/isMatch"));
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var index_2 = require("../vars/index");
var glamor = glamor_1.default || glamorExports;
var style = {
    start: function () { return glamor.css(index_1.default['.psds-calendar__date--selected-start']); },
    end: function () { return glamor.css(index_1.default['.psds-calendar__date--selected-end']); },
    inRange: function (arg) {
        return glamor.css(arg && index_1.default['.psds-calendar__date--in-range']);
    }
};
exports.useIsInRange = function (selected) {
    if (selected === void 0) { selected = []; }
    var _a = react_1.default.useState(), hoveredDate = _a[0], setHoveredDate = _a[1];
    // Calendar level
    var onMouseLeave = function () {
        setHoveredDate(undefined);
    };
    // Date level
    var onMouseEnter = function (date) {
        if (!selected.length) {
            return;
        }
        setHoveredDate(date);
    };
    var isInRange = function (date) {
        if (selected.length) {
            var firstSelected = selected[0].getTime();
            if (selected.length === 2) {
                var secondSelected = selected[1].getTime();
                return __assign(__assign(__assign({}, style.inRange(firstSelected < date.getTime() && secondSelected > date.getTime())), (firstSelected === date.getTime() && style.start())), (secondSelected === date.getTime() && style.end()));
            }
            else {
                return __assign(__assign({}, style.inRange(hoveredDate &&
                    ((firstSelected < date.getTime() &&
                        hoveredDate.getTime() >= date.getTime()) ||
                        (date.getTime() < firstSelected &&
                            date.getTime() >= hoveredDate.getTime())))), (firstSelected === date.getTime() && style.start()));
            }
        }
        return {};
    };
    return { onMouseLeave: onMouseLeave, onMouseEnter: onMouseEnter, isInRange: isInRange };
};
exports.onRangeDateSelected = function (_a) {
    var _b = _a.selected, selected = _b === void 0 ? [] : _b, setSelected = _a.setSelected, onSelect = _a.onSelect;
    return function (dateObj, evt) {
        var selectable = dateObj.selectable, date = dateObj.date;
        if (!selectable) {
            return;
        }
        var dateTime = date.getTime();
        var newDates = __spreadArrays(selected);
        if (selected.length) {
            if (selected.length === 1) {
                var firstTime = selected[0].getTime();
                if (firstTime < dateTime) {
                    newDates.push(date);
                }
                else {
                    newDates.unshift(date);
                }
                setSelected(newDates);
            }
            else if (newDates.length === 2) {
                setSelected([date]);
            }
        }
        else {
            newDates.push(date);
            setSelected(newDates);
        }
        onSelect && onSelect(evt, dateObj);
    };
};
var getDateIndex = function (selected, condition) {
    var index;
    selected.some(function (date, i) {
        index = i;
        if (condition(date.getTime())) {
            return true;
        }
        // If we loop through all the selected dates and still didn't find
        // one, make sure to add it to the end of the array.
        index++;
        return false;
    });
    return index;
};
exports.onMultiDateSelected = function (_a) {
    var _b = _a.selected, selected = _b === void 0 ? [] : _b, setSelected = _a.setSelected, onSelect = _a.onSelect;
    return function (dateObj, evt) {
        var isSelected = dateObj.selected, selectable = dateObj.selectable, date = dateObj.date;
        if (!selectable) {
            return;
        }
        var newSelectedDates = selected.slice();
        var selectedTime = date.getTime();
        if (isSelected) {
            var index = getDateIndex(selected, function (time) { return selectedTime === time; });
            newSelectedDates.splice(index, 1);
        }
        else {
            // Add
            var index = getDateIndex(selected, function (time) { return selectedTime < time; });
            newSelectedDates.splice(index, 0, date);
        }
        setSelected(newSelectedDates);
        onSelect && onSelect(evt, dateObj);
    };
};
exports.useDateSelectChange = function (_a) {
    var selected = _a.selected, setSlide = _a.setSlide, setSelected = _a.setSelected, _b = _a.dateFormat, dateFormat = _b === void 0 ? 'MM/dd/yyyy' : _b;
    var _c = react_1.default.useState(selected ? format_1.default(selected, dateFormat) : ''), value = _c[0], setValue = _c[1];
    react_1.default.useEffect(function () {
        selected ? setValue(format_1.default(selected, dateFormat)) : '';
    }, [selected]);
    var onChange = function (evt) {
        var nextValue = evt.target.value;
        setValue(nextValue);
        var fulldate = false;
        try {
            fulldate = isMatch_1.default(nextValue, dateFormat);
        }
        catch (err) {
            if (!(err instanceof RangeError)) {
                throw err;
            }
        }
        var nextSelected = fulldate
            ? parse_1.default(nextValue, dateFormat, new Date())
            : selected;
        if (nextSelected instanceof Date &&
            !isNaN(nextSelected)) {
            var nextSlide = void 0;
            if (selected) {
                nextSlide =
                    differenceInCalendarMonths_1.default(nextSelected, selected) > 0
                        ? 'forward'
                        : differenceInCalendarMonths_1.default(nextSelected, selected) < 0
                            ? 'backward'
                            : undefined;
            }
            setSelected(nextSelected);
            setSlide(nextSlide);
        }
    };
    return [value, onChange];
};
exports.useRangeSelectChange = function (_a) {
    var start = _a.start, _b = _a.selected, selected = _b === void 0 ? [] : _b, setSlide = _a.setSlide, setSelected = _a.setSelected, _c = _a.dateFormat, dateFormat = _c === void 0 ? 'MM/dd/yyyy' : _c;
    var _selected = start ? selected[0] : selected[1];
    var _d = react_1.default.useState(_selected ? format_1.default(_selected, dateFormat) : ''), value = _d[0], setValue = _d[1];
    react_1.default.useEffect(function () {
        _selected ? setValue(format_1.default(_selected, dateFormat)) : '';
    }, [_selected]);
    var onChange = function (evt) {
        var nextValue = evt.target.value;
        setValue(nextValue);
        var fulldate = false;
        try {
            fulldate = isMatch_1.default(nextValue, dateFormat);
        }
        catch (err) {
            if (!(err instanceof RangeError)) {
                throw err;
            }
        }
        var nextSelected = fulldate
            ? parse_1.default(nextValue, dateFormat, new Date())
            : selected;
        if (nextSelected instanceof Date &&
            !isNaN(nextSelected)) {
            var nextSlide = void 0;
            if (selected) {
                !selected[0] && setSelected([]);
                start && setSelected([nextSelected, selected[1]].filter(Boolean));
                selected[0] && !start && setSelected([selected[0], nextSelected]);
                nextSlide =
                    differenceInCalendarMonths_1.default(nextSelected, _selected) > 0
                        ? index_2.slides.forward
                        : differenceInCalendarMonths_1.default(nextSelected, _selected) < 0
                            ? index_2.slides.backward
                            : index_2.slides.undefined;
            }
            setSlide(nextSlide);
        }
    };
    return [value, onChange];
};
