// @pluralsight/ps-design-system-switch/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '.psds-switch': {
            display: 'inline-flex',
            alignItems: 'center',
            cursor: 'pointer',
            opacity: '1',
            background: 'none',
            border: 'none',
            padding: 0,
            '&:focus': { outline: 'none' }
        },
        '.psds-switch--disabled': {
            opacity: '0.4',
            cursor: 'not-allowed'
        }
    },
    _a[".psds-switch--labelAlign-" + vars.labelAligns.right] = {
        flexDirection: 'row'
    },
    _a[".psds-switch--labelAlign-" + vars.labelAligns.left] = {
        flexDirection: 'row-reverse'
    },
    _a['.psds-switch__track'] = {
        position: 'relative',
        backgroundColor: ps_design_system_core_1.colorsBackgroundUtility[25],
        border: "1px solid " + ps_design_system_core_1.colorsBorder.lowOnDark,
        transition: "background-color " + ps_design_system_core_1.motion.speedFast + " ease-in-out"
    },
    _a[".psds-switch__track.psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: "1px solid " + ps_design_system_core_1.colorsBorder.lowOnLight
    },
    _a[".psds-switch__track.psds-switch__track--size-" + vars.sizes.small] = {
        height: '12px',
        width: '24px',
        borderRadius: '6px'
    },
    _a[".psds-switch__track.psds-switch__track--size-" + vars.sizes.large] = {
        height: '24px',
        width: '48px',
        borderRadius: '12px'
    },
    _a[".psds-switch__track--checked.psds-switch__track--color-" + vars.colors.blue] = {
        backgroundColor: ps_design_system_core_1.colorsBlue[6],
        borderColor: 'transparent'
    },
    _a[".psds-switch__track--checked.psds-switch__track--color-" + vars.colors.green] = {
        backgroundColor: ps_design_system_core_1.colorsGreen[6],
        borderColor: 'transparent'
    },
    _a[".psds-switch__track--checked.psds-switch__track--color-" + vars.colors.orange] = {
        // NOTE: temp: make orange blue until we do a breaking change to remove orange
        backgroundColor: ps_design_system_core_1.colorsBlue[6],
        borderColor: 'transparent'
    },
    _a[".psds-switch__thumb"] = {
        backgroundColor: ps_design_system_core_1.colorsWhite,
        borderRadius: '50%',
        transition: "transform " + ps_design_system_core_1.motion.speedFast + " ease-in-out"
    },
    _a[".psds-switch__thumb--size-" + vars.sizes.small] = {
        height: '10px',
        width: '10px'
    },
    _a[".psds-switch__thumb--size-" + vars.sizes.large] = {
        height: '22px',
        width: '22px'
    },
    _a[".psds-switch__thumb--checked.psds-switch__thumb--size-" + vars.sizes.small] = {
        transform: 'translateX(12px)'
    },
    _a[".psds-switch__thumb--checked.psds-switch__thumb--size-" + vars.sizes.large] = {
        transform: 'translateX(24px)'
    },
    _a[".psds-switch__label"] = {
        fontWeight: ps_design_system_core_1.type.fontWeightDefault,
        lineHeight: '1em'
    },
    _a[".psds-switch__label--size-" + vars.sizes.small] = {
        fontSize: ps_design_system_core_1.type.fontSize100
    },
    _a[".psds-switch__label--size-" + vars.sizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize200
    },
    _a[".psds-switch__label--size-" + vars.sizes.small + ".psds-switch__label--labelAlign-" + vars.labelAligns.left] = {
        marginRight: ps_design_system_core_1.layout.spacingXSmall
    },
    _a[".psds-switch__label--size-" + vars.sizes.large + ".psds-switch__label--labelAlign-" + vars.labelAligns.left] = {
        marginRight: ps_design_system_core_1.layout.spacingMedium
    },
    _a[".psds-switch__label--size-" + vars.sizes.small + ".psds-switch__label--labelAlign-" + vars.labelAligns.right] = {
        marginLeft: ps_design_system_core_1.layout.spacingXSmall
    },
    _a[".psds-switch__label--size-" + vars.sizes.large + ".psds-switch__label--labelAlign-" + vars.labelAligns.right] = {
        marginLeft: ps_design_system_core_1.layout.spacingMedium
    },
    _a[".psds-switch__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-switch__label.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    _a['.psds-switch__checkbox'] = ps_design_system_core_1.accessibility.screenReaderOnly,
    _a);
