// @pluralsight/ps-design-system-navbar/dist/cjs/css/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
exports.default = {
    '.psds-navbar': {
        display: 'flex',
        alignItems: 'center',
        width: '100%',
        height: '56px',
        padding: ps_design_system_core_1.layout.spacingXSmall,
        background: ps_design_system_core_1.colorsBackgroundDark[2],
        borderBottom: "1px solid " + ps_design_system_core_1.colorsBackgroundDark[3],
        boxShadow: 'rgba(0, 0, 0, 0.5) 0px 0 8px 0px'
    },
    '.psds-navbar__brand': {
        marginRight: ps_design_system_core_1.layout.spacingMedium
    },
    '.psds-navbar__items': {
        flex: 1,
        marginRight: ps_design_system_core_1.layout.spacingLarge
    },
    '.psds-navbar__mobile-menu': {
        marginRight: ps_design_system_core_1.layout.spacingXXSmall
    },
    '.psds-navbar__utility': {
        marginLeft: 'auto',
        marginRight: ps_design_system_core_1.layout.spacingXXSmall
    },
    '.psds-navbar__user': {}
};
