// @pluralsight/ps-design-system-textinput/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_halo_1 = __importDefault(require("@pluralsight/ps-design-system-halo"));
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    error: function () { return glamor.css(index_1.default['.psds-text-input__error']); },
    field: function (_a) {
        var appearance = _a.appearance, error = _a.error, fieldAfter = _a.fieldAfter, icon = _a.icon, iconAlign = _a.iconAlign, themeName = _a.themeName, size = _a.size;
        var small = size === vars.sizes.small;
        return glamor.css(index_1.default['.psds-text-input__field'], index_1.default[".psds-text-input__field--appearance-" + appearance], index_1.default[".psds-text-input__field.psds-theme--" + themeName], index_1.default[".psds-text-input__field--appearance-" + appearance + ".psds-theme--" + themeName], Boolean(fieldAfter) && index_1.default[".psds-text-input__field--w-after"], small && index_1.default['.psds-text-input__field.psds-text-input--small'], Boolean(icon) &&
            small &&
            index_1.default[".psds-text-input__field--icon-align-" + iconAlign + ".psds-text-input--small"], Boolean(icon) &&
            !small &&
            index_1.default[".psds-text-input__field--icon-align-" + iconAlign], error &&
            index_1.default[".psds-text-input__field--error.psds-theme--" + themeName]);
    },
    fieldContainer: function () {
        return glamor.css(index_1.default['.psds-text-input__field-container']);
    },
    fieldInput: function (appearance, themeName) {
        return glamor.css(index_1.default['.psds-text-input__field-input'], index_1.default[".psds-text-input__field-input--appearance-" + appearance], index_1.default[".psds-text-input__field-input.psds-theme--" + themeName]);
    },
    icon: function (_a) {
        var appearance = _a.appearance, icon = _a.icon, iconAlign = _a.iconAlign, themeName = _a.themeName;
        return glamor.css(index_1.default['.psds-text-input__icon'], Boolean(icon) &&
            index_1.default[".psds-text-input__icon--icon-align-" + iconAlign], index_1.default[".psds-text-input__icon--appearance-" + appearance], index_1.default[".psds-text-input__icon.psds-theme--" + themeName]);
    },
    textInput: function (disabled) {
        return glamor.css(index_1.default['.psds-text-input'], disabled && index_1.default['.psds-text-input--disabled']);
    },
    label: function (themeName) {
        return glamor.css(index_1.default['.psds-text-input__label'], index_1.default[".psds-text-input__label.psds-theme--" + themeName]);
    },
    subLabel: function (themeName) {
        return glamor.css(index_1.default['.psds-text-input__sub-label'], index_1.default[".psds-text-input__sub-label.psds-theme--" + themeName]);
    }
};
var TextInput = ps_design_system_util_1.forwardRefWithStatics(function (_a, forwardedRef) {
    var _b = _a.appearance, appearance = _b === void 0 ? vars.appearances.default : _b, _c = _a.disabled, disabled = _c === void 0 ? false : _c, _d = _a.error, error = _d === void 0 ? false : _d, _e = _a.iconAlign, iconAlign = _e === void 0 ? vars.iconAligns.left : _e, _f = _a.size, size = _f === void 0 ? vars.sizes.medium : _f, fieldAfter = _a.fieldAfter, icon = _a.icon, label = _a.label, subLabel = _a.subLabel, props = __rest(_a, ["appearance", "disabled", "error", "iconAlign", "size", "fieldAfter", "icon", "label", "subLabel"]);
    var themeName = ps_design_system_theme_1.useTheme();
    return (react_1.default.createElement("label", __assign({}, styles.textInput(disabled), { style: props.style, className: props.className }),
        label && react_1.default.createElement("div", __assign({}, styles.label(themeName)), label),
        react_1.default.createElement("div", __assign({}, styles.fieldContainer()),
            react_1.default.createElement(ps_design_system_halo_1.default, { error: error, gapSize: ps_design_system_halo_1.default.gapSizes.small },
                react_1.default.createElement("div", __assign({}, styles.field({
                    appearance: appearance,
                    error: error,
                    fieldAfter: fieldAfter,
                    icon: icon,
                    iconAlign: iconAlign,
                    themeName: themeName,
                    size: size
                })),
                    react_1.default.createElement("input", __assign({}, props, styles.fieldInput(appearance, themeName), { disabled: disabled, placeholder: props.placeholder, ref: forwardedRef })),
                    fieldAfter)),
            icon && (react_1.default.createElement("div", __assign({}, styles.icon({ appearance: appearance, icon: icon, iconAlign: iconAlign, themeName: themeName })), icon)),
            error && (react_1.default.createElement("div", __assign({}, styles.error()),
                react_1.default.createElement(ps_design_system_icon_1.WarningIcon, null)))),
        subLabel && react_1.default.createElement("div", __assign({}, styles.subLabel(themeName)), subLabel)));
});
TextInput.appearances = vars.appearances;
TextInput.iconAligns = vars.iconAligns;
TextInput.sizes = vars.sizes;
exports.default = TextInput;
