// @pluralsight/ps-design-system-form/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '.psds-form-button-row': {
            display: 'flex'
        }
    },
    _a[".psds-form-button-row--align-" + vars.aligns.right] = {
        justifyContent: 'flex-end'
    },
    _a['.psds-form-button-row__button'] = {
        marginRight: ps_design_system_core_1.layout.spacingLarge,
        '&:last-child': { marginRight: 0 }
    },
    _a['.psds-form-divider'] = {
        height: '1px',
        width: '100%',
        backgroundColor: ps_design_system_core_1.colorsBorder.lowOnDark
    },
    _a[".psds-form-divider.psds-theme--" + ps_design_system_theme_1.names.light] = {
        backgroundColor: ps_design_system_core_1.colorsBorder.lowOnLight
    },
    _a['.psds-form-vertical-layout'] = {
        width: '100%'
    },
    _a['.psds-form-vertical-layout__child'] = {
        marginBottom: ps_design_system_core_1.layout.spacingLarge,
        '&:last-child': { marginBottom: 0 }
    },
    _a);
