// @pluralsight/ps-design-system-steps/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
var dark = ".psds-theme--" + ps_design_system_theme_1.names.dark;
var light = ".psds-theme--" + ps_design_system_theme_1.names.light;
exports.default = (_a = {
        '.psds-steps': {
            label: 'steps',
            counterReset: 'line-number',
            display: 'flex'
        }
    },
    _a[".psds-steps--" + vars.orientations.horizontal] = {
        flexDirection: 'row',
        marginBottom: ps_design_system_core_1.layout.spacingMedium
    },
    _a[".psds-steps--" + vars.orientations.vertical] = {
        flexDirection: 'column'
    },
    _a['.psds-steps__step'] = {
        label: 'steps__step',
        alignItems: 'flex-start',
        counterIncrement: 'line-number',
        display: 'flex',
        flex: 1,
        padding: ps_design_system_core_1.layout.spacingXSmall
    },
    _a['.psds-steps__step--interactive'] = {
        borderRadius: 4,
        cursor: 'pointer',
        '&:hover': { backgroundColor: ps_design_system_core_1.colorsBackgroundUtility[25] }
    },
    _a[".psds-steps__step" + dark] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a[".psds-steps_step" + light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-steps__step--" + vars.orientations.horizontal] = {
        marginRight: ps_design_system_core_1.layout.spacingXSmall,
        '&:last-child': { marginRight: 0 }
    },
    _a[".psds-steps__step--" + vars.orientations.vertical] = {
        marginBottom: ps_design_system_core_1.layout.spacingXSmall,
        '&:last-child': { marginBottom: 0 }
    },
    _a[".psds-steps__step--" + vars.sizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize200
    },
    _a[".psds-steps__step--" + vars.sizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize400
    },
    _a[".psds-steps__step--" + vars.statuses.current] = {
        fontWeight: ps_design_system_core_1.type.fontWeight700
    },
    _a['.psds-steps__title'] = {
        '& p:last-child': { marginBottom: 0 }
    },
    _a[".psds-steps__title--" + vars.sizes.large] = {
        paddingTop: 5 // NOTE: align text vertically with marker
    },
    _a[".psds-steps__title--" + vars.sizes.medium] = {
        paddingTop: 1 // NOTE: align text vertically with marker
    },
    _a[".psds-steps__title--" + vars.statuses.current + dark] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    _a[".psds-steps__title--" + vars.statuses.current + light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a['.psds-steps__description'] = {
        '& p:last-child': { marginBottom: 0 }
    },
    _a[".psds-steps__description--" + vars.sizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize100
    },
    _a[".psds-steps__description--" + vars.sizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize200
    },
    _a[".psds-steps__description--" + vars.statuses.current] = {
        fontWeight: ps_design_system_core_1.type.fontWeightDefault
    },
    _a['.psds-steps__marker-container'] = {
        label: 'steps__marker-container',
        flexShrink: 0
    },
    _a[".psds-steps__marker-container--" + vars.sizes.medium] = {
        marginRight: ps_design_system_core_1.layout.spacingXSmall
    },
    _a[".psds-steps__marker-container--" + vars.sizes.large] = {
        marginRight: ps_design_system_core_1.layout.spacingSmall
    },
    _a['.psds-steps__marker'] = {
        label: 'steps__marker',
        fontSize: ps_design_system_core_1.type.fontSize100,
        position: 'relative',
        textAlign: 'center',
        '&:after': {
            content: 'counter(line-number)',
            left: '50%',
            margin: 'auto',
            opacity: 1,
            position: 'absolute',
            top: '50%',
            transform: 'translate(-50%, -50%)',
            transition: "all " + ps_design_system_core_1.motion.speedFast,
            width: '50%'
        }
    },
    _a['.psds-steps__marker--hide-counter'] = {
        '&:after': { content: '' }
    },
    _a[".psds-steps__marker" + dark] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a[".psds-steps__marker" + light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-steps__marker--" + vars.sizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize100
    },
    _a[".psds-steps__marker--" + vars.sizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize400
    },
    _a[".psds-steps__marker--" + vars.statuses.completed] = {
        color: ps_design_system_core_1.colorsGreen[6],
        '&:after': { opacity: 0 }
    },
    _a[".psds-steps__marker--" + vars.statuses.current + dark] = {
        color: ps_design_system_core_1.colorsPrimaryAction.textDarkTheme
    },
    _a[".psds-steps__marker--" + vars.statuses.current + light] = {
        color: ps_design_system_core_1.colorsPrimaryAction.textLightTheme
    },
    _a['.psds-steps__marker__circle'] = {
        label: 'steps__marker__circle',
        fill: 'transparent',
        stroke: 'currentColor'
    },
    _a['.psds-steps__marker__check'] = {
        label: 'steps__marker__check',
        fill: 'transparent',
        stroke: 'currentColor',
        strokeDasharray: 1000,
        strokeDashoffset: 1000,
        strokeLinecap: 'round',
        strokeLinejoin: 'round',
        transition: "stroke-dashoffset 0s linear 0s"
    },
    _a['.psds-steps__marker__check--completed'] = {
        strokeDashoffset: 0,
        transition: "stroke-dashoffset 3s linear " + ps_design_system_core_1.motion.speedNormal
    },
    _a);
