// @pluralsight/ps-design-system-radio/dist/esm/css/index.js
var _a;
import { accessibility, colorsBlue, colorsWhite, colorsBackgroundDark, colorsBackgroundLight, colorsTextIcon, layout, type } from '@pluralsight/ps-design-system-core';
import { names as themeNames } from '@pluralsight/ps-design-system-theme';
export default (_a = {
        '.psds-radio-group': {
            '&:focus': { outline: 'none' }
        },
        '.psds-radio-group--disabled': {
            opacity: '0.5'
        },
        '.psds-radio-group__button-container': {
            padding: layout.spacingXXSmall + " 0"
        },
        '.psds-radio-group__label': {
            color: colorsTextIcon.highOnDark,
            fontSize: type.fontSize100,
            fontWeight: type.fontWeight500,
            marginBottom: layout.spacingXXSmall
        }
    },
    _a[".psds-radio-group__label.psds-theme--" + themeNames.light] = {
        color: colorsTextIcon.highOnLight
    },
    _a['.psds-radio-group__sub-label'] = {
        color: colorsTextIcon.lowOnDark,
        fontSize: type.fontSize100,
        fontWeight: type.fontWeightDefault,
        marginTop: layout.spacingXXSmall
    },
    _a[".psds-radio-group__sub-label.psds-theme--" + themeNames.light] = {
        color: colorsTextIcon.lowOnLight
    },
    _a['.psds-radio-button'] = {
        display: 'flex',
        alignItems: 'center',
        padding: layout.spacingXXSmall + " 0",
        cursor: 'pointer'
    },
    _a['.psds-radio-button--disabled'] = {
        cursor: 'not-allowed'
    },
    _a['.psds-radio-button__halo'] = {
        transform: 'translateY(3px)'
    },
    _a['.psds-radio-button__circle-outer'] = {
        width: '16px',
        marginRight: layout.spacingSmall,
        pointerEvents: 'none'
    },
    _a['.psds-radio-button__circle'] = {
        position: 'relative',
        display: 'inline-block',
        height: '16px',
        width: '16px',
        borderRadius: '50%',
        border: "2px solid " + colorsTextIcon.lowOnDark,
        background: colorsBackgroundDark[1],
        '&:focus': { outline: 'none' }
    },
    _a[".psds-radio-button__circle.psds-theme--" + themeNames.light] = {
        background: colorsBackgroundLight[3],
        borderColor: colorsTextIcon.lowOnLight
    },
    _a['.psds-radio-button__circle--checked'] = {
        borderColor: colorsBlue[6],
        background: colorsBlue[6]
    },
    _a['.psds-radio-button__circle-inner'] = {
        display: 'block',
        height: 'calc(100% - 6px)',
        width: 'calc(100% - 6px)',
        margin: '3px',
        borderRadius: '50%',
        background: colorsWhite
    },
    _a['.psds-radio-button__input'] = accessibility.screenReaderOnly,
    _a['.psds-radio-button__label'] = {
        color: colorsTextIcon.highOnDark,
        fontSize: type.fontSize200,
        lineHeight: type.lineHeightStandard,
        fontWeight: type.fontWeightDefault
    },
    _a[".psds-radio-button__label.psds-theme--" + themeNames.light] = {
        color: colorsTextIcon.highOnLight
    },
    _a);
