// @pluralsight/ps-design-system-steps/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var context_1 = __importStar(require("./context"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    steps: function (themeName, opts) {
        return glamor.compose(glamor.css(index_1.default['.psds-steps']), glamor.css(index_1.default[".psds-steps.psds-theme--" + themeName]), glamor.css(index_1.default[".psds-steps--" + opts.orientation]));
    },
    step: function (themeName, opts) {
        return glamor.compose(glamor.css(index_1.default['.psds-steps__step']), glamor.css(index_1.default[".psds-steps__step.psds-theme--" + themeName]), glamor.css(index_1.default[".psds-steps__step--" + opts.orientation]), glamor.css(index_1.default[".psds-steps__step--" + opts.size]), glamor.css(index_1.default[".psds-steps__step--" + opts.status]), opts.interactive &&
            glamor.css(index_1.default['.psds-steps__step--interactive']));
    },
    title: function (themeName, opts) {
        var theme = ".psds-theme--" + themeName;
        var status = ".psds-steps__title--" + opts.status;
        return glamor.compose(glamor.css(index_1.default['.psds-steps__title']), glamor.css(index_1.default['.psds-steps__title' + theme]), glamor.css(index_1.default[".psds-steps__title--" + opts.size]), glamor.css(index_1.default[status]), glamor.css(index_1.default[status + theme]));
    },
    description: function (opts) {
        return glamor.compose(glamor.css(index_1.default['.psds-steps__description']), glamor.css(index_1.default[".psds-steps__description--" + opts.size]), glamor.css(index_1.default[".psds-steps__description--" + opts.status]));
    },
    markerContainer: function (opts) {
        return glamor.compose(glamor.css(index_1.default['.psds-steps__marker-container']), glamor.css(index_1.default[".psds-steps__marker-container--" + opts.size]));
    },
    marker: function (themeName, opts) {
        var theme = ".psds-theme--" + themeName;
        var status = ".psds-steps__marker--" + opts.status;
        return glamor.compose(glamor.css(index_1.default['.psds-steps__marker']), glamor.css(index_1.default['.psds-steps__marker' + theme]), glamor.css(index_1.default[".psds-steps__marker--" + opts.size]), glamor.css(index_1.default[status]), glamor.css(index_1.default[status + theme]), !opts.counter &&
            glamor.css(index_1.default['.psds-steps__marker--hide-counter']));
    },
    markerCircle: function () { return glamor.css(index_1.default['.psds-steps__marker__circle']); },
    markerCheck: function (opts) {
        return glamor.compose(glamor.css(index_1.default['.psds-steps__marker__check']), glamor.css(index_1.default[".psds-steps__marker__check--" + opts.status]));
    }
};
var Steps = react_1.default.forwardRef(function (props, ref) {
    var _a = props.counter, counter = _a === void 0 ? context_1.initialContext.counter : _a, _b = props.orientation, orientation = _b === void 0 ? context_1.initialContext.orientation : _b, _c = props.size, size = _c === void 0 ? context_1.initialContext.size : _c, rest = __rest(props, ["counter", "orientation", "size"]);
    var themeName = ps_design_system_theme_1.useTheme();
    var contextValue = { counter: counter, orientation: orientation, size: size };
    return (react_1.default.createElement(context_1.default.Provider, { value: contextValue },
        react_1.default.createElement("div", __assign({ ref: ref }, styles.steps(themeName, { orientation: orientation }), rest))));
});
Steps.displayName = 'Steps';
var defaultMarkerContainer = function (props) { return react_1.default.createElement("div", __assign({}, props)); };
var Step = react_1.default.forwardRef(function (props, ref) {
    var children = props.children, description = props.description, _a = props.marker, StepMarker = _a === void 0 ? Marker : _a, _b = props.renderMarkerContainer, renderMarkerContainer = _b === void 0 ? defaultMarkerContainer : _b, status = props.status, rest = __rest(props, ["children", "description", "marker", "renderMarkerContainer", "status"]);
    var MarkerContainer = react_1.default.useMemo(function () { return renderMarkerContainer; }, [
        renderMarkerContainer
    ]);
    var interactive = 'onClick' in props;
    var themeName = ps_design_system_theme_1.useTheme();
    var _c = react_1.default.useContext(context_1.default), orientation = _c.orientation, size = _c.size;
    return (react_1.default.createElement("div", __assign({ ref: ref }, styles.step(themeName, { interactive: interactive, orientation: orientation, size: size, status: status }), rest),
        react_1.default.createElement(MarkerContainer, __assign({}, styles.markerContainer({ size: size })),
            react_1.default.createElement(StepMarker, { status: status })),
        react_1.default.createElement("div", null,
            react_1.default.createElement("div", __assign({}, styles.title(themeName, { size: size, status: status })), ps_design_system_util_1.isString(children) ? react_1.default.createElement("p", null, children) : children),
            description && (react_1.default.createElement("div", __assign({}, styles.description({ size: size, status: status })), ps_design_system_util_1.isString(description) ? react_1.default.createElement("p", null, description) : description)))));
});
Step.displayName = 'Steps.Step';
var Marker = react_1.default.forwardRef(function (props, ref) {
    var status = props.status, rest = __rest(props, ["status"]);
    var themeName = ps_design_system_theme_1.useTheme();
    var _a = react_1.default.useContext(context_1.default), counter = _a.counter, size = _a.size;
    var _b = react_1.default.useMemo(function () {
        var large = size === 'large';
        var _diameter = large ? 36 : 24;
        var _stroke = large ? 2 : 2;
        var _radius = _diameter / 2 - _stroke / 2;
        return [_diameter, _stroke, _radius];
    }, [size]), diameter = _b[0], stroke = _b[1], radius = _b[2];
    return (react_1.default.createElement("div", __assign({ ref: ref }, styles.marker(themeName, { counter: counter, size: size, status: status }), glamor.css({ width: diameter, height: diameter }), rest),
        react_1.default.createElement("svg", { viewBox: "0 0 " + diameter + " " + diameter },
            react_1.default.createElement("circle", __assign({ cx: diameter / 2, cy: diameter / 2, r: radius, strokeWidth: stroke }, styles.markerCircle())),
            react_1.default.createElement("polyline", __assign({ strokeWidth: stroke, points: "\n            " + diameter / 3.8 + " " + diameter / 2 + ",\n            " + diameter / 2.3 + " " + diameter / 1.5 + ",\n            " + diameter / 1.4 + " " + diameter / 3 + "\n          " }, styles.markerCheck({ status: status }))))));
});
Marker.displayName = 'Steps.Marker';
Steps.Marker = Marker;
Steps.Step = Step;
Steps.orientations = vars.orientations;
Steps.sizes = vars.sizes;
Steps.statuses = vars.statuses;
exports.default = Steps;
