// @codecademy/gamut/dist/GridForm/__tests__/stubs.js
export var stubCheckboxField = {
  description: 'Check me!',
  label: 'Stub Checkbox',
  name: 'stub-checkbox',
  size: 6,
  type: 'checkbox'
};
export var stubRadioGroupOptions = [{
  label: 'Blue Team',
  value: 'blue'
}, {
  label: 'Red Team',
  value: 'red'
}];
export var stubRadioGroupField = {
  label: 'Stub Radio Group',
  options: stubRadioGroupOptions,
  name: 'stub-radio-group',
  size: 6,
  rowspan: 2,
  type: 'radio-group'
};
export var stubSelectOptions = ['aaa', 'bbb'];
export var stubSelectField = {
  label: 'Stub Select',
  options: stubSelectOptions,
  name: 'stub-select',
  size: 6,
  type: 'select'
};
export var stubTextField = {
  label: 'Stub Text',
  name: 'stub-text',
  size: 6,
  type: 'text'
};
export var stubFileField = {
  label: 'Stub File',
  name: 'stub-file',
  size: 6,
  type: 'file'
};
export var stubTextareaField = {
  label: 'Stub Textarea',
  name: 'stub-textarea',
  size: 6,
  type: 'textarea'
};
export var stubHiddenField = {
  name: 'secret-field',
  type: 'hidden',
  defaultValue: "you can't see me!"
};
export var stubSweetContainerField = {
  name: 'sweet_container',
  type: 'sweet-container',
  label: 'Check this input if you are an automated machine, and not a human'
};
