// @pluralsight/ps-design-system-datepicker/dist/esm/react/datepicker.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { canUseDOM } from '@pluralsight/ps-design-system-util';
import { useDayzed } from 'dayzed';
import React from 'react';
import { Calendar } from './calendar';
import { CalendarDates } from './calendar-dates';
import { TextInputField } from './text-input-field';
import { useDateSelectChange } from './utils';
export var DatePicker = function (_a) {
    var disabled = _a.disabled, error = _a.error, label = _a.label, prefix = _a.prefix, renderContainer = _a.renderContainer, renderTag = _a.renderTag, size = _a.size, subLabel = _a.subLabel, suffix = _a.suffix, onSelect = _a.onSelect, valueFromProps = _a.value, _uniqueId = _a._uniqueId, props = __rest(_a, ["disabled", "error", "label", "prefix", "renderContainer", "renderTag", "size", "subLabel", "suffix", "onSelect", "value", "_uniqueId"]);
    var _b = React.useState(valueFromProps), selected = _b[0], setSelected = _b[1];
    React.useEffect(function () { return setSelected(valueFromProps); }, [valueFromProps]);
    var _c = React.useState(false), open = _c[0], setOpen = _c[1];
    var onDateSelected = function (dateObj, evt) {
        onSelect && onSelect(evt, dateObj);
        setSelected(dateObj.date);
        setOpen(false);
    };
    var _d = useDayzed({
        date: selected,
        selected: selected,
        onDateSelected: onDateSelected
    }), getDateProps = _d.getDateProps, dayzedData = __rest(_d, ["getDateProps"]);
    var handleIconClick = function (evt) {
        setOpen(!open);
    };
    var _e = React.useState(), slide = _e[0], setSlide = _e[1];
    var _f = useDateSelectChange({
        selected: selected,
        setSlide: setSlide,
        setSelected: setSelected
    }), value = _f[0], onChange = _f[1];
    var ref = React.useRef();
    React.useEffect(function () {
        if (!canUseDOM())
            return function () { };
        var handleClickOutsideMenu = function (evt) {
            if (evt.target instanceof HTMLElement) {
                // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
                if (ref.current.contains(evt.target))
                    return;
                setOpen(false);
            }
        };
        document.addEventListener('click', handleClickOutsideMenu, {
            capture: true
        });
        return function () {
            return document.removeEventListener('click', handleClickOutsideMenu, {
                capture: true
            });
        };
    }, [setOpen]);
    return (React.createElement("div", __assign({ style: { display: 'inline-block', position: 'relative' } }, props, { ref: ref }),
        React.createElement(TextInputField, { disabled: disabled, error: error, label: label, onChange: onChange, onClick: handleIconClick, placeholder: "mm/dd/yyyy", prefix: prefix, renderContainer: renderContainer, renderTag: renderTag, size: size, subLabel: subLabel, suffix: suffix, value: value, _uniqueId: _uniqueId }),
        React.createElement("br", null),
        open && (React.createElement(Calendar, __assign({}, dayzedData, { style: { position: 'absolute', zIndex: 1, marginTop: 4 }, slide: slide }),
            React.createElement(CalendarDates, { getDateProps: getDateProps }, function (renderProps) {
                return React.createElement("button", __assign({}, renderProps));
            })))));
};
