// @pluralsight/ps-design-system-form/dist/esm/css/index.js
var _a;
import { colorsBorder, layout } from '@pluralsight/ps-design-system-core';
import { names as themeNames } from '@pluralsight/ps-design-system-theme';
import * as vars from '../vars/index';
export default (_a = {
        '.psds-form-button-row': {
            display: 'flex'
        }
    },
    _a[".psds-form-button-row--align-" + vars.aligns.right] = {
        justifyContent: 'flex-end'
    },
    _a['.psds-form-button-row__button'] = {
        marginRight: layout.spacingLarge,
        '&:last-child': { marginRight: 0 }
    },
    _a['.psds-form-divider'] = {
        height: '1px',
        width: '100%',
        backgroundColor: colorsBorder.lowOnDark
    },
    _a[".psds-form-divider.psds-theme--" + themeNames.light] = {
        backgroundColor: colorsBorder.lowOnLight
    },
    _a['.psds-form-vertical-layout'] = {
        width: '100%'
    },
    _a['.psds-form-vertical-layout__child'] = {
        marginBottom: layout.spacingLarge,
        '&:last-child': { marginBottom: 0 }
    },
    _a);
