// @pluralsight/ps-design-system-layout/dist/esm/css/aside-layout.js
var _a, _b;
import { layout } from '@pluralsight/ps-design-system-core';
import { asideLayout as vars } from '../vars/index';
export default (_a = {
        '.psds-aside-layout': {
            display: 'flex'
        }
    },
    _a[".psds-aside-layout--aside-position-" + vars.asidePositions.first] = {
        flexDirection: 'column'
    },
    _a[".psds-aside-layout--aside-position-" + vars.asidePositions.last] = {
        flexDirection: 'column-reverse'
    },
    _a[".psds-aside-layout__aside--aside-position-" + vars.asidePositions.first] = {
        marginBottom: layout.spacingLarge
    },
    _a[".psds-aside-layout__aside--aside-position-" + vars.asidePositions.last] = {
        marginTop: layout.spacingLarge
    },
    _a['@media (min-width: 769px)'] = (_b = {},
        _b[".psds-aside-layout--aside-position-" + vars.asidePositions.first] = {
            flexDirection: 'row'
        },
        _b[".psds-aside-layout--aside-position-" + vars.asidePositions.last] = {
            flexDirection: 'row-reverse'
        },
        _b['.psds-aside-layout__aside'] = {
            width: 'calc(25%)'
        },
        _b[".psds-aside-layout__aside--aside-position-" + vars.asidePositions.first] = {
            marginBottom: 0,
            marginRight: layout.spacingLarge
        },
        _b[".psds-aside-layout__aside--aside-position-" + vars.asidePositions.last] = {
            marginTop: 0,
            marginLeft: layout.spacingLarge
        },
        _b['.psds-aside-layout__main'] = {
            width: 'calc(75%)'
        },
        _b),
    _a);
