// @pluralsight/ps-design-system-textarea/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '.psds-text-area': {
            display: 'inline-block',
            width: '100%',
            maxWidth: '496px'
        },
        '.psds-text-area--disabled': {
            opacity: 0.5
        },
        '.psds-text-area__field': {
            position: 'relative',
            minHeight: '104px',
            width: '100%',
            borderRadius: '2px',
            background: ps_design_system_core_1.colorsBackgroundLight[3],
            fontSize: ps_design_system_core_1.type.fontSize200,
            lineHeight: ps_design_system_core_1.type.lineHeightStandard,
            fontWeight: ps_design_system_core_1.type.fontWeightDefault,
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            padding: ps_design_system_core_1.layout.spacingXSmall + " " + ps_design_system_core_1.layout.spacingMedium,
            border: "1px solid " + ps_design_system_core_1.colorsBorder.lowOnDark,
            '&:focus': { outline: 'none' }
        }
    },
    _a[".psds-text-area__field--appearance--" + vars.appearances.subtle] = {
        '&::placeholder': { color: ps_design_system_core_1.colorsTextIcon.lowOnDark }
    },
    _a[".psds-text-area__field--appearance-" + vars.appearances.subtle + ".psds-theme--" + ps_design_system_theme_1.names.dark] = {
        background: ps_design_system_core_1.colorsBackgroundDark[1],
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnDark
    },
    _a[".psds-text-area__field--appearance-" + vars.appearances.subtle + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-text-area__field.psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnLight,
        '&::placeholder': { color: ps_design_system_core_1.colorsTextIcon.lowOnLight },
        '&:focus': { border: '1px solid transparent' }
    },
    _a[".psds-text-area__field--error.psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: '1px solid transparent'
    },
    _a['.psds-text-area__field-container'] = {
        position: 'relative',
        display: 'flex',
        alignItems: 'center'
    },
    _a['.psds-text-area__label'] = {
        width: '100%',
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        marginBottom: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-text-area__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a['.psds-text-area__sub-label'] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        marginTop: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-text-area__sub-label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a['.psds-text-area__error'] = {
        position: 'absolute',
        right: "calc(-1 * (" + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + "))",
        display: 'flex',
        alignItems: 'flex-start',
        height: '100%',
        paddingTop: ps_design_system_core_1.layout.spacingXSmall,
        marginLeft: ps_design_system_core_1.layout.spacingXSmall,
        color: ps_design_system_core_1.colorsStatus.error
    },
    _a);
