// @pluralsight/ps-design-system-textarea/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.appearances = void 0;
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_halo_1 = __importDefault(require("@pluralsight/ps-design-system-halo"));
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
var calcRowsPxHeight = function (rows) {
    var int = function (varVal) { return parseInt(varVal.replace('px', ''), 10); };
    return (int(ps_design_system_core_1.layout.spacingXSmall) * 2 +
        parseInt(rows, 10) * int(ps_design_system_core_1.type.lineHeightStandard));
};
var styles = {
    error: function () { return glamor.css(index_1.default['.psds-text-area__error']); },
    field: function (themeName, appearance, error) {
        var label = 'psds-text-area__field';
        return glamor.compose(glamor.css(index_1.default["." + label]), glamor.css(index_1.default["." + label + "--appearance--" + appearance]), glamor.css(index_1.default["." + label + "--appearance-" + appearance + ".psds-theme--" + themeName]), glamor.css(index_1.default["." + label + ".psds-theme--" + themeName]), error &&
            glamor.css(index_1.default["." + label + "--error.psds-theme--" + themeName]));
    },
    fieldContainer: function () {
        return glamor.css(index_1.default['.psds-text-area__field-container']);
    },
    textarea: function (disabled) {
        var label = 'psds-text-area';
        return glamor.compose(glamor.css(index_1.default["." + label]), disabled && glamor.css(index_1.default["." + label + "--disabled"]));
    },
    label: function (themeName) {
        var label = 'psds-text-area__label';
        return glamor.compose(glamor.css(index_1.default["." + label]), glamor.css(index_1.default["." + label + ".psds-theme--" + themeName]));
    },
    subLabel: function (themeName) {
        var label = 'psds-text-area__sub-label';
        return glamor.compose(glamor.css(index_1.default["." + label]), glamor.css(index_1.default["." + label + ".psds-theme--" + themeName]));
    }
};
var TextArea = react_1.default.forwardRef(function (_a, ref) {
    var _b = _a.appearance, appearance = _b === void 0 ? vars.appearances.default : _b, className = _a.className, _c = _a.disabled, disabled = _c === void 0 ? false : _c, _d = _a.error, error = _d === void 0 ? false : _d, label = _a.label, placeholder = _a.placeholder, _e = _a.rows, rows = _e === void 0 ? 4 : _e, style = _a.style, subLabel = _a.subLabel, rest = __rest(_a, ["appearance", "className", "disabled", "error", "label", "placeholder", "rows", "style", "subLabel"]);
    var themeName = ps_design_system_theme_1.useTheme();
    return (react_1.default.createElement("label", __assign({}, styles.textarea(disabled), { style: style, className: className }),
        label && react_1.default.createElement("div", __assign({}, styles.label(themeName)), label),
        react_1.default.createElement("div", __assign({}, styles.fieldContainer()),
            react_1.default.createElement(ps_design_system_halo_1.default, { error: error, gapSize: ps_design_system_halo_1.default.gapSizes.small },
                react_1.default.createElement("textarea", __assign({}, rest, styles.field(themeName, appearance, error), { disabled: disabled, placeholder: placeholder, ref: ref, style: { height: calcRowsPxHeight(rows) } }))),
            error && (react_1.default.createElement("div", __assign({}, styles.error()),
                react_1.default.createElement(ps_design_system_icon_1.WarningIcon, null)))),
        subLabel && react_1.default.createElement("div", __assign({}, styles.subLabel(themeName)), subLabel)));
});
TextArea.appearances = vars.appearances;
exports.appearances = vars.appearances;
exports.default = TextArea;
