// @pluralsight/ps-design-system-halo/dist/esm/css/index.js
var _a, _b;
import { colorsBlue, colorsStatus } from '@pluralsight/ps-design-system-core';
import { names as themeNames } from '@pluralsight/ps-design-system-theme';
export var BASE_CLASSNAME = '.psds-halo';
export var themeClasses = (_a = {},
    _a[themeNames.dark] = ".psds-theme--dark",
    _a[themeNames.light] = ".psds-theme--light",
    _a);
export default (_b = {},
    _b[BASE_CLASSNAME] = {
        position: 'relative',
        display: 'inline-block',
        lineHeight: 0,
        flex: '1 1 100%',
        '&:after': {
            content: ' ',
            position: 'absolute',
            borderWidth: '3px',
            borderStyle: 'solid',
            borderColor: colorsBlue[6],
            pointerEvents: 'none',
            visibility: 'hidden'
        }
    },
    _b[BASE_CLASSNAME + "--inline"] = {
        flexBasis: 0,
        flexGrow: 0
    },
    _b[BASE_CLASSNAME + "--error"] = {
        '&:after': {
            visibility: 'visible',
            borderColor: colorsStatus.error
        }
    },
    _b[BASE_CLASSNAME + "--visible"] = {
        '&:after': {
            visibility: 'visible',
            borderColor: colorsBlue[6]
        }
    },
    _b[BASE_CLASSNAME + "--visible-on-focus"] = {
        '&:focus-within:after, &[focus-within]:after': {
            visibility: 'visible',
            borderColor: colorsBlue[6]
        }
    },
    _b[BASE_CLASSNAME + "--gap-size-default" + themeClasses[themeNames.dark]] = {
        '&:after': {
            top: '-4px',
            bottom: '-4px',
            left: '-4px',
            right: '-4px'
        }
    },
    _b[BASE_CLASSNAME + "--gap-size-default" + themeClasses[themeNames.light]] = {
        '&:after': {
            top: '-5px',
            bottom: '-5px',
            left: '-5px',
            right: '-5px'
        }
    },
    _b[BASE_CLASSNAME + "--gap-size-small" + themeClasses[themeNames.dark]] = {
        '&:after': {
            top: '-4px',
            bottom: '-4px',
            left: '-4px',
            right: '-4px'
        }
    },
    _b[BASE_CLASSNAME + "--gap-size-small" + themeClasses[themeNames.light]] = {
        '&:after': {
            top: '-2px',
            bottom: '-2px',
            left: '-2px',
            right: '-2px'
        }
    },
    _b[BASE_CLASSNAME + "--shape-default"] = {
        '&:after': {
            borderRadius: '4px'
        }
    },
    _b[BASE_CLASSNAME + "--shape-pill"] = {
        '&:after': {
            borderRadius: '1000px' // use a big number to make a pill shape
        }
    },
    _b);
