// @pluralsight/ps-design-system-verticaltabs/dist/cjs/react/__stories__/fixtures.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FLOW_NAV = exports.DESIGN_SYSTEM_NAV = exports.ADMIN_TOOLS_NAV = void 0;
var Icon = __importStar(require("@pluralsight/ps-design-system-icon"));
var react_1 = __importDefault(require("react"));
exports.ADMIN_TOOLS_NAV = [
    {
        collapsible: false,
        header: {
            href: '#',
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'dashboard',
            title: 'Dashboard'
        },
        items: []
    },
    {
        collapsible: false,
        header: {
            href: '#',
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'account',
            title: 'Account'
        },
        items: []
    },
    {
        collapsible: true,
        header: {
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'people',
            title: 'People'
        },
        items: [
            { href: '#', id: 'org', title: 'Organization' },
            { href: '#', id: 'admins', title: 'Admins' },
            { href: '#', id: 'managers', title: 'Managers' },
            { href: '#', id: 'unassigned', title: 'Unassigned users' },
            { href: '#', id: 'pending', title: 'Pending invites' }
        ]
    },
    {
        collapsible: false,
        header: {
            href: '#',
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'analytics',
            title: 'Analytics'
        },
        items: []
    },
    {
        collapsible: false,
        header: {
            href: '#',
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'log',
            title: 'Log'
        },
        items: []
    }
];
exports.DESIGN_SYSTEM_NAV = [
    {
        collapsible: false,
        header: { title: 'Introduction' },
        sections: [
            { id: 'install', header: { href: '#', title: 'Install' } },
            { id: 'design-assets', header: { href: '#', title: 'Design assets' } },
            { id: 'contrib', header: { href: '#', title: 'Contribute' } },
            { id: 'roadmap', header: { href: '#', title: 'Roadmap' } }
        ]
    },
    {
        collapsible: false,
        header: { title: 'Foundations' },
        sections: [
            {
                id: 'color',
                header: {
                    href: '#',
                    icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
                    title: 'Color'
                }
            },
            {
                id: 'typography',
                header: {
                    href: '#',
                    icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
                    title: 'Typography'
                }
            },
            {
                id: 'spacing',
                header: {
                    href: '#',
                    icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
                    title: 'Spacing'
                }
            },
            {
                id: 'iconography',
                header: {
                    href: '#',
                    icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
                    title: 'Iconography'
                }
            },
            {
                id: 'voice-tone',
                header: {
                    href: '#',
                    icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
                    title: 'Voice & Tone'
                }
            }
        ]
    },
    {
        collapsible: false,
        header: { title: 'Components' },
        sections: [
            {
                id: 'avatar',
                header: {
                    href: '#',
                    title: 'Avatar'
                },
                items: [
                    {
                        href: '#',
                        id: 'avatar-prop-types',
                        title: 'PropTypes'
                    },
                    {
                        href: '#',
                        id: 'avatar-guidelines',
                        title: 'Guidelines'
                    }
                ]
            },
            {
                id: 'badge',
                header: {
                    href: '#',
                    title: 'Badge'
                },
                items: []
            },
            {
                id: 'breadcrumb',
                header: {
                    href: '#',
                    title: 'Breadcrumb'
                },
                items: []
            }
        ]
    }
];
exports.FLOW_NAV = [
    {
        collapsible: false,
        header: {
            href: '#',
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'reports-home',
            title: 'Reports Home'
        },
        items: []
    },
    {
        collapsible: true,
        header: {
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'operational-reports',
            title: 'Operational Reports'
        },
        items: [
            {
                href: '#',
                id: 'work-log',
                title: 'Work Log'
            },
            {
                href: '#',
                id: 'project-timeline',
                title: 'Project Timeline'
            },
            {
                href: '#',
                id: 'leaderboard',
                title: 'Leaderboard'
            },
            {
                href: '#',
                id: 'snapshot',
                title: 'Snapshot'
            },
            {
                href: '#',
                id: 'daily-updates',
                title: 'Daily Updates'
            },
            {
                href: '#',
                id: 'spot-check',
                title: 'Spot Check'
            }
        ]
    },
    {
        collapsible: true,
        header: {
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'review-collaborate',
            title: 'Review & Collaborate'
        },
        items: [
            {
                href: '#',
                id: 'review-workflow',
                title: 'Review Workflow'
            },
            {
                href: '#',
                id: 'review-collab',
                title: 'Review Collaboration'
            },
            {
                href: '#',
                id: 'pr-resolution',
                title: 'PR Resolution'
            },
            {
                href: '#',
                id: 'knowledge-sharing',
                title: 'Knowledge Sharing'
            },
            {
                href: '#',
                id: 'player-card',
                title: 'Player Card'
            }
        ]
    },
    {
        collapsible: true,
        header: {
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'delivery',
            title: 'Delivery'
        },
        items: [
            {
                href: '#',
                title: 'Retrospective'
            }
        ]
    },
    {
        collapsible: true,
        header: {
            icon: react_1.default.createElement(Icon.PlaceholderIcon, null),
            id: 'fundamentals',
            title: 'Fundamentals'
        },
        items: [
            {
                href: '#',
                id: 'fundamentals-code',
                title: 'Code'
            },
            {
                href: '#',
                id: 'fundamentals-submit',
                title: 'Submit'
            },
            {
                href: '#',
                id: 'fundamentals-review',
                title: 'Review'
            }
        ]
    }
];
