// @codecademy/gamut-styles/dist/variance/props.js
import { variance } from '@codecademy/variance';
import * as PROPERTIES from './config';
/**  General Prop Groups  */

export var typography = variance.create(PROPERTIES.typography);
export var grid = variance.create(PROPERTIES.grid);
export var flex = variance.create(PROPERTIES.flex);
export var layout = variance.create(PROPERTIES.layout);
export var positioning = variance.create(PROPERTIES.positioning);
export var background = variance.create(PROPERTIES.background);
export var color = variance.create(PROPERTIES.color);
export var shadow = variance.create(PROPERTIES.shadows);
export var space = variance.create(PROPERTIES.space);
export var border = variance.create(PROPERTIES.border);
/** Sub Groups */

export var padding = variance.create(PROPERTIES.padding);
export var margin = variance.create(PROPERTIES.margin);
/** CSS */

export var css = variance.createCss(PROPERTIES.all);
export var variant = variance.createVariant(PROPERTIES.all);
export var states = variance.createStates(PROPERTIES.all);
