// @pluralsight/ps-design-system-navuser/dist/cjs/css/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
exports.default = {
    '.psds-navuser': {
        display: 'inline-flex',
        alignItems: 'center',
        padding: ps_design_system_core_1.layout.spacingXXSmall + " " + ps_design_system_core_1.layout.spacingXSmall,
        textDecoration: 'none',
        border: 'none',
        background: 'none',
        borderRadius: '2px'
    },
    '.psds-navuser--clickable': {
        cursor: 'pointer',
        '&:hover, &:focus': {
            background: "rgba(" + ps_design_system_core_1.colorsBackgroundUtilityCsv + ",0.15)"
        },
        '&:focus': {
            outline: 'none'
        },
        '&:active': {
            background: "rgba(" + ps_design_system_core_1.colorsBackgroundUtilityCsv + ",0.25)"
        }
    },
    '.psds-navuser__plan-name': {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        fontSize: '11px',
        fontWeight: ps_design_system_core_1.type.fontWeightDefault,
        overflow: 'hidden',
        textOverflow: 'ellipsis'
    },
    '.psds-navuser__name': {
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        fontSize: '12px'
    },
    '.psds-navuser__words': {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        lineHeight: '16px',
        marginLeft: ps_design_system_core_1.layout.spacingXSmall,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        textAlign: 'left',
        whiteSpace: 'nowrap',
        maxWidth: '120px'
    }
};
