// @pluralsight/ps-design-system-textinput/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '.psds-text-input': {
            display: 'inline-block'
        },
        '.psds-text-input--disabled': {
            opacity: 0.5,
            cursor: 'not-allowed'
        },
        // __field
        '.psds-text-input__field': {
            alignItems: 'center',
            background: ps_design_system_core_1.colorsBackgroundLight[3],
            border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnLight,
            borderRadius: '2px',
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            display: 'flex',
            fontWeight: ps_design_system_core_1.type.fontWeightDefault,
            height: '40px',
            minWidth: '192px',
            padding: ps_design_system_core_1.layout.spacingXSmall + " " + ps_design_system_core_1.layout.spacingMedium,
            position: 'relative',
            width: '100%'
        }
    },
    _a[".psds-text-input__field.psds-theme--" + ps_design_system_theme_1.names.light] = {
        background: ps_design_system_core_1.colorsBackgroundLight[3],
        border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnLight
    },
    _a['.psds-text-input__field.psds-text-input--small'] = {
        height: '32px',
        padding: "6px " + ps_design_system_core_1.layout.spacingXSmall
    },
    _a['.psds-text-input__field--w-after'] = {
        paddingRight: ps_design_system_core_1.layout.spacingXSmall
    },
    _a['.psds-text-input__field-input'] = {
        width: '100%',
        background: 'none',
        border: 'none',
        color: 'inherit',
        fontSize: ps_design_system_core_1.type.fontSize200,
        lineHeight: ps_design_system_core_1.type.lineHeightStandard,
        outline: 'none',
        padding: 0,
        '&:focus': { outline: 'none' },
        '&::placeholder': { color: ps_design_system_core_1.colorsTextIcon.lowOnLight },
        '&:disabled': {
            cursor: 'not-allowed'
        }
    },
    _a[".psds-text-input__field-input--appearance-" + vars.appearances.subtle] = {
        '&::placeholder': { color: ps_design_system_core_1.colorsTextIcon.lowOnDark }
    },
    _a[".psds-text-input__field-input.psds-theme--" + ps_design_system_theme_1.names.light] = {
        '&::placeholder': { color: ps_design_system_core_1.colorsTextIcon.lowOnLight }
    },
    _a[".psds-text-input__field--error.psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: '1px solid transparent'
    },
    _a[".psds-text-input__field--appearance-" + vars.appearances.subtle] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        background: ps_design_system_core_1.colorsBackgroundDark[1],
        border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnDark
    },
    _a[".psds-text-input__field--appearance-" + vars.appearances.subtle + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-text-input__field--icon-align-" + vars.iconAligns.left] = {
        padding: "0 " + ps_design_system_core_1.layout.spacingMedium + " 0 calc(" + ps_design_system_core_1.layout.spacingXSmall + " + " + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + ")"
    },
    _a[".psds-text-input__field--icon-align-" + vars.iconAligns.right] = {
        padding: "0 calc(" + ps_design_system_core_1.layout.spacingXSmall + " + " + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + ") 0 " + ps_design_system_core_1.layout.spacingMedium
    },
    _a[".psds-text-input__field--icon-align-" + vars.iconAligns.left + ".psds-text-input--small"] = {
        padding: "0 " + ps_design_system_core_1.layout.spacingXSmall + " 0 " + (parseInt(ps_design_system_core_1.layout.spacingMedium, 10) + parseInt(ps_design_system_icon_1.widths.medium, 10)) + "px"
    },
    _a[".psds-text-input__field--icon-align-" + vars.iconAligns.right + ".psds-text-input--small"] = {
        padding: "0 " + (parseInt(ps_design_system_core_1.layout.spacingMedium, 10) + parseInt(ps_design_system_icon_1.widths.medium, 10)) + "px 0 " + ps_design_system_core_1.layout.spacingXSmall
    },
    // __field-container
    _a['.psds-text-input__field-container'] = {
        position: 'relative',
        display: 'flex',
        alignItems: 'center',
        minWidth: "calc(192px + " + ps_design_system_core_1.layout.spacingXSmall + ")"
    },
    // __icon
    _a['.psds-text-input__icon'] = {
        position: 'absolute',
        left: ps_design_system_core_1.layout.spacingXSmall,
        top: '50%',
        transform: 'translateY(-50%)',
        display: 'flex',
        alignItems: 'center',
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-text-input__icon.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-text-input__icon--icon-align-" + vars.iconAligns.right] = {
        left: 'auto',
        right: ps_design_system_core_1.layout.spacingXSmall
    },
    _a[".psds-text-input__icon--appearance-" + vars.appearances.subtle] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    // __label
    _a['.psds-text-input__label'] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        marginBottom: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-text-input__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    // __sub-label
    _a['.psds-text-input__sub-label'] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeightDefault,
        marginTop: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-text-input__sub-label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    // __error
    _a['.psds-text-input__error'] = {
        position: 'absolute',
        right: "calc(-1 * (" + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + "))",
        display: 'flex',
        alignItems: 'center',
        color: ps_design_system_core_1.colorsStatus.error,
        marginLeft: ps_design_system_core_1.layout.spacingXSmall
    },
    _a);
