// @pluralsight/ps-design-system-scrollable/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    outer: function () {
        return glamor.compose(glamor.css(index_1.default['.psds-scrollable__outer']), glamor.media('not print', index_1.default['.psds-scrollable__outer--screen']));
    },
    inner: function () { return glamor.css(index_1.default['.psds-scrollable__inner']); },
    content: function () { return glamor.css(index_1.default['.psds-scrollable__content']); },
    handle: function (grabbed) {
        return glamor.compose(glamor.css(index_1.default['.psds-scrollable__handle']), grabbed && glamor.css(index_1.default['.psds-scrollable__handle--grabbed']));
    }
};
var BLANK_IMAGE;
if (ps_design_system_util_1.canUseDOM()) {
    BLANK_IMAGE = new Image(0, 0);
    // prettier-ignore
    BLANK_IMAGE.src = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
}
function areEqualProps(prevProps, nextProps) {
    var changed = ps_design_system_util_1.shallowCompare(prevProps, nextProps);
    return !changed;
}
var Scrollable = react_1.default.forwardRef(function (props, forwardedRef) {
    var _a = props.renderContent, renderContent = _a === void 0 ? defaultRenderContent : _a, rest = __rest(props, ["renderContent"]);
    var ref = react_1.default.useRef();
    /* eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion */
    react_1.default.useImperativeHandle(forwardedRef, function () { return ref.current; });
    var dragPreview = react_1.default.useRef(BLANK_IMAGE);
    var _b = react_1.default.useState(true), hidden = _b[0], setHidden = _b[1];
    var _c = react_1.default.useState(0), pageY = _c[0], setPageY = _c[1];
    var _d = react_1.default.useState('0%'), offset = _d[0], setOffset = _d[1];
    var _e = react_1.default.useState(0), scrollRatio = _e[0], setScrollRatio = _e[1];
    var updateDimensions = react_1.default.useCallback(function () {
        if (!ref.current)
            return;
        var content = ref.current;
        var nextTotalHeight = content.scrollHeight;
        var nextOffset = (content.scrollTop / nextTotalHeight) * 100 + '%';
        var nextScrollRatio = content.clientHeight / nextTotalHeight;
        var shouldHide = nextScrollRatio >= 1;
        setHidden(shouldHide);
        setOffset(nextOffset);
        setScrollRatio(nextScrollRatio);
    }, [ref]);
    ps_design_system_util_1.useResizeObserver(ref, updateDimensions);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    react_1.default.useEffect(function () { return updateDimensions(); }, []);
    function onHandleDrag(evt) {
        var nextPageY = evt.pageY;
        if (nextPageY <= 0)
            return;
        var content = ref.current;
        if (!content)
            return;
        var delta = evt.pageY - pageY;
        setPageY(nextPageY);
        content.scrollTop += delta / scrollRatio;
    }
    function onHandleDragEnd(_evt) {
        setPageY(0);
    }
    function onHandleDragStart(evt) {
        if (evt &&
            evt.dataTransfer &&
            evt.dataTransfer.setDragImage &&
            dragPreview.current) {
            evt.dataTransfer.setDragImage(dragPreview.current, 0, 0);
        }
        setPageY(evt.pageY);
    }
    var styleOverride = react_1.default.useMemo(function () {
        var height = Math.max(scrollRatio * 100, 10) + '%';
        var visibility = hidden ? 'hidden' : 'initial';
        return glamor.css({ height: height, top: offset, visibility: visibility });
    }, [hidden, offset, scrollRatio]);
    var onScroll = ps_design_system_util_1.combineFns(updateDimensions, props.onScroll);
    return (react_1.default.createElement(Outer, __assign({ "data-scrollable": true }, rest),
        react_1.default.createElement(Inner, { key: "scrollable-wrapper" }, renderContent(__assign({ children: props.children, onScroll: onScroll }, styles.content()), function (node) { return (ref.current = node); })),
        react_1.default.createElement(Handle, __assign({ draggable: true, onDrag: onHandleDrag, onDragEnd: onHandleDragEnd, onDragStart: onHandleDragStart }, styleOverride))));
});
Scrollable.displayName = 'Scrollable';
var defaultRenderContent = function (contentProps, contentRef) { return react_1.default.createElement("div", __assign({}, contentProps, { ref: contentRef })); };
var Outer = react_1.default.memo(function OuterComp(props) {
    return react_1.default.createElement("div", __assign({}, styles.outer(), props));
}, areEqualProps);
Outer.displayName = 'Scrollable.Outer';
var Inner = react_1.default.memo(function InnerComp(props) {
    return react_1.default.createElement("div", __assign({}, styles.inner(), props));
}, areEqualProps);
Inner.displayName = 'Scrollable.Inner';
var Handle = function (props) {
    var _a = react_1.default.useState(false), grabbed = _a[0], setGrabbed = _a[1];
    var onDragEnd = ps_design_system_util_1.combineFns(function () {
        setGrabbed(false);
    }, props.onDragEnd);
    var onDragStart = ps_design_system_util_1.combineFns(function () {
        setGrabbed(true);
    }, props.onDragStart);
    return (react_1.default.createElement("div", __assign({}, styles.handle(grabbed), props, { onDragEnd: onDragEnd, onDragStart: onDragStart })));
};
Handle.displayName = 'Scrollable.Handle';
exports.default = Scrollable;
