// @pluralsight/ps-design-system-navbar/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_navitem_1 = __importDefault(require("@pluralsight/ps-design-system-navitem"));
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    brand: function () { return glamor.css(index_1.default['.psds-navbar__brand']); },
    mobileMenu: function () { return glamor.css(index_1.default['.psds-navbar__mobile-menu']); },
    navbar: function () { return glamor.css(index_1.default['.psds-navbar']); },
    items: function () { return glamor.css(index_1.default['.psds-navbar__items']); },
    user: function () { return glamor.css(index_1.default['.psds-navbar__user']); },
    utility: function () { return glamor.css(index_1.default['.psds-navbar__utility']); }
};
var NavBar = react_1.default.forwardRef(function (props, forwardedRef) {
    var brand = props.brand, items = props.items, onMobileMenuClick = props.onMobileMenuClick, utility = props.utility, user = props.user, rest = __rest(props, ["brand", "items", "onMobileMenuClick", "utility", "user"]);
    var ref = react_1.default.useRef(null);
    react_1.default.useImperativeHandle(forwardedRef, function () { return ref.current; });
    return (react_1.default.createElement("div", __assign({ ref: ref }, styles.navbar(), rest),
        onMobileMenuClick && (react_1.default.createElement("div", __assign({}, styles.mobileMenu()),
            react_1.default.createElement(ps_design_system_navitem_1.default, { icon: react_1.default.createElement(ps_design_system_icon_1.MenuIcon, null), onClick: onMobileMenuClick }))),
        react_1.default.createElement("div", __assign({}, styles.brand()), brand),
        react_1.default.createElement("div", __assign({}, styles.items()), items),
        react_1.default.createElement("div", __assign({}, styles.utility()), utility),
        react_1.default.createElement("div", __assign({}, styles.user()), user)));
});
NavBar.displayName = 'NavBar';
exports.default = NavBar;
