// @pluralsight/ps-design-system-appframe/dist/cjs/react/__stories__/shared.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkillsLogo = exports.SkillsBrand = exports.TopNav = exports.SectionDivider = exports.SectionHeader = exports.SideNav = exports.MockContent = void 0;
var react_1 = __importDefault(require("react"));
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_layout_1 = require("@pluralsight/ps-design-system-layout");
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var ps_design_system_navbar_1 = __importDefault(require("@pluralsight/ps-design-system-navbar"));
var ps_design_system_navbrand_1 = __importDefault(require("@pluralsight/ps-design-system-navbrand"));
var ps_design_system_navitem_1 = __importDefault(require("@pluralsight/ps-design-system-navitem"));
var ps_design_system_navuser_1 = __importDefault(require("@pluralsight/ps-design-system-navuser"));
var Text = __importStar(require("@pluralsight/ps-design-system-text"));
var ps_design_system_verticaltabs_1 = __importDefault(require("@pluralsight/ps-design-system-verticaltabs"));
exports.MockContent = function () {
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement("div", { style: { background: ps_design_system_core_1.colorsPink[6] } },
            react_1.default.createElement(ps_design_system_layout_1.PageWidthLayout, null,
                react_1.default.createElement(ps_design_system_layout_1.PageHeadingLayout, { heading: react_1.default.createElement("h3", null, "Page title") }))),
        react_1.default.createElement(ps_design_system_layout_1.PageWidthLayout, null,
            react_1.default.createElement(Text.P, null, "Cupcake ipsum dolor sit amet. Sweet gummi bears drag\u00E9e. Pie drag\u00E9e cotton candy candy canes bear claw apple pie. Dessert apple pie caramels carrot cake bear claw sweet donut. Croissant chocolate bear claw jelly chocolate candy pie pudding caramels. Cake muffin candy liquorice liquorice tiramisu powder pie drag\u00E9e. Danish souffl\u00E9 sweet apple pie cupcake tart. Danish cookie oat cake tiramisu lemon drops pie apple pie cake cotton candy. Pudding brownie jujubes. Chocolate bar biscuit liquorice candy jelly-o tiramisu icing muffin sweet roll. Jelly candy canes chocolate drag\u00E9e. Danish biscuit sugar plum cake. Cotton candy souffl\u00E9 muffin gummi bears."),
            react_1.default.createElement(Text.P, null, "Sweet roll sweet roll cotton candy marshmallow lemon drops caramels sweet roll toffee. Bear claw powder ice cream topping wafer cake tootsie roll. Lemon drops pudding dessert toffee fruitcake toffee. Icing marzipan cake pastry biscuit liquorice chupa chups sesame snaps. Biscuit muffin donut. Marshmallow carrot cake icing icing drag\u00E9e jujubes lemon drops. Danish cheesecake sesame snaps. Sweet sugar plum fruitcake tootsie roll. Fruitcake jelly-o souffl\u00E9 muffin cheesecake pie apple pie jelly powder. Cake drag\u00E9e danish muffin drag\u00E9e souffl\u00E9 marzipan croissant danish. Souffl\u00E9 chupa chups cake biscuit icing pastry cake danish danish. Icing cotton candy oat cake chocolate macaroon tiramisu sweet drag\u00E9e cake. Ice cream carrot cake lemon drops. Jelly sweet wafer liquorice brownie sesame snaps halvah apple pie jelly-o."),
            react_1.default.createElement(Text.P, null, "Tootsie roll drag\u00E9e apple pie. Pie cake toffee caramels gummies chupa chups carrot cake chupa chups. Jelly-o fruitcake macaroon chocolate. Brownie lollipop sugar plum gummies marshmallow muffin bonbon carrot cake. Pudding powder lemon drops halvah. Icing cupcake pastry chupa chups danish sweet bear claw sugar plum. Chupa chups wafer donut lollipop. Sweet roll croissant cake chocolate cake croissant pastry toffee. Chocolate cake liquorice halvah pastry danish candy canes. Apple pie pastry topping sesame snaps cupcake liquorice icing chocolate bar brownie. Chocolate cake sesame snaps cake gummies cupcake pastry jujubes fruitcake. Muffin fruitcake souffl\u00E9 macaroon danish. Dessert chocolate marshmallow sesame snaps donut toffee ice cream sugar plum drag\u00E9e."),
            react_1.default.createElement(Text.P, null, "Marzipan lollipop jelly-o souffl\u00E9 macaroon tart danish tart dessert. Chupa chups lemon drops lemon drops jujubes drag\u00E9e dessert. Wafer chupa chups gummi bears sugar plum gummies. Powder topping apple pie danish sweet roll. Pudding toffee brownie. Tootsie roll lemon drops sweet macaroon jelly-o oat cake muffin chocolate. Wafer chocolate danish cookie cookie danish. Chocolate jelly-o gingerbread fruitcake croissant wafer lollipop bonbon macaroon. Carrot cake candy dessert. Brownie gummies topping sweet roll jelly sesame snaps croissant marshmallow caramels. Candy canes jelly-o jelly beans chupa chups tiramisu cupcake cake tart. Biscuit brownie chocolate bar danish candy canes donut liquorice. Muffin cake jelly beans muffin powder gingerbread toffee. Gingerbread topping toffee chupa chups sweet roll."),
            react_1.default.createElement(Text.P, null, "Jelly beans gingerbread dessert macaroon candy canes. Candy liquorice carrot cake chocolate bar pastry dessert caramels danish. Donut apple pie tart drag\u00E9e sesame snaps chocolate tootsie roll jelly-o sugar plum. Chocolate cake sesame snaps jelly beans. Gummies pudding gummies. Jelly-o topping biscuit cookie halvah pudding gingerbread toffee cake. Sesame snaps topping icing jelly cake donut macaroon candy canes jelly beans. Tiramisu pie pie. Cake gingerbread pastry topping jujubes apple pie pie. Jelly beans marshmallow chocolate cake chocolate cake biscuit danish dessert. Halvah biscuit macaroon. Sugar plum pie gummi bears gummies pastry."),
            react_1.default.createElement(Text.P, null, "Marzipan lollipop jelly-o souffl\u00E9 macaroon tart danish tart dessert. Chupa chups lemon drops lemon drops jujubes drag\u00E9e dessert. Wafer chupa chups gummi bears sugar plum gummies. Powder topping apple pie danish sweet roll. Pudding toffee brownie. Tootsie roll lemon drops sweet macaroon jelly-o oat cake muffin chocolate. Wafer chocolate danish cookie cookie danish. Chocolate jelly-o gingerbread fruitcake croissant wafer lollipop bonbon macaroon. Carrot cake candy dessert. Brownie gummies topping sweet roll jelly sesame snaps croissant marshmallow caramels. Candy canes jelly-o jelly beans chupa chups tiramisu cupcake cake tart. Biscuit brownie chocolate bar danish candy canes donut liquorice. Muffin cake jelly beans muffin powder gingerbread toffee. Gingerbread topping toffee chupa chups sweet roll."),
            react_1.default.createElement(Text.P, null, "Jelly beans gingerbread dessert macaroon candy canes. Candy liquorice carrot cake chocolate bar pastry dessert caramels danish. Donut apple pie tart drag\u00E9e sesame snaps chocolate tootsie roll jelly-o sugar plum. Chocolate cake sesame snaps jelly beans. Gummies pudding gummies. Jelly-o topping biscuit cookie halvah pudding gingerbread toffee cake. Sesame snaps topping icing jelly cake donut macaroon candy canes jelly beans. Tiramisu pie pie. Cake gingerbread pastry topping jujubes apple pie pie. Jelly beans marshmallow chocolate cake chocolate cake biscuit danish dessert. Halvah biscuit macaroon. Sugar plum pie gummi bears gummies pastry."))));
};
var defaultSections = [
    {
        header: {
            href: '#',
            icon: react_1.default.createElement(ps_design_system_icon_1.PlaceholderIcon, null),
            id: 'section-header',
            title: 'Section Header'
        },
        items: [
            {
                href: '#',
                icon: react_1.default.createElement(ps_design_system_icon_1.PlaceholderIcon, null),
                id: 'section-item__first',
                title: 'First Item',
                subItems: [
                    {
                        href: '#',
                        icon: react_1.default.createElement(ps_design_system_icon_1.PlaceholderIcon, null),
                        id: 'section-subitem__first',
                        title: 'First SubItem'
                    },
                    {
                        href: '#',
                        icon: react_1.default.createElement(ps_design_system_icon_1.PlaceholderIcon, null),
                        id: 'section-subitem__second',
                        title: 'Second SubItem'
                    }
                ]
            },
            {
                href: '#',
                icon: react_1.default.createElement(ps_design_system_icon_1.PlaceholderIcon, null),
                id: 'section-item__second',
                title: 'Second Item'
            }
        ]
    }
];
exports.SideNav = function (props) {
    var children = props.children, _a = props.collapsed, collapsed = _a === void 0 ? false : _a, _b = props.sections, sections = _b === void 0 ? defaultSections : _b, rest = __rest(props, ["children", "collapsed", "sections"]);
    return (react_1.default.createElement("div", __assign({}, rest),
        sections.map(function (section, sectionKey) {
            var header = section.header, _a = section.items, items = _a === void 0 ? [] : _a;
            return (react_1.default.createElement(react_1.default.Fragment, { key: sectionKey },
                header && (react_1.default.createElement(exports.SectionHeader, { collapsed: collapsed, title: header.title })),
                react_1.default.createElement(ps_design_system_verticaltabs_1.default, { forceCollapsed: collapsed, hideLabels: collapsed }, items.map(function (item) {
                    var _a = item.subItems, subItems = _a === void 0 ? [] : _a;
                    return (react_1.default.createElement(ps_design_system_verticaltabs_1.default.Group, { key: item.id },
                        react_1.default.createElement(ps_design_system_verticaltabs_1.default.Tier1, { header: react_1.default.createElement(ps_design_system_verticaltabs_1.default.Tier1.Header, { href: item.href, icon: item.icon }, item.title) }),
                        subItems.map(function (subItem) { return (react_1.default.createElement(ps_design_system_verticaltabs_1.default.Tier2, { key: subItem.id, header: react_1.default.createElement(ps_design_system_verticaltabs_1.default.Tier2.Header, { href: subItem.href }, subItem.title) })); })));
                })),
                react_1.default.createElement(exports.SectionDivider, null)));
        }),
        children));
};
exports.SectionHeader = function (props) {
    var _a = props.collapsed, collapsed = _a === void 0 ? false : _a, title = props.title, rest = __rest(props, ["collapsed", "title"]);
    return (react_1.default.createElement("div", __assign({ style: {
            margin: ps_design_system_core_1.layout.spacingSmall + " 0 0",
            opacity: collapsed ? 0 : 1,
            padding: "0 " + ps_design_system_core_1.layout.spacingLarge
        } }, rest),
        react_1.default.createElement("h2", { style: { margin: 0 } },
            react_1.default.createElement(Text.Label, { size: Text.Label.sizes.xSmall, color: Text.Label.colors.secondary, caps: true }, title))));
};
exports.SectionDivider = function (props) {
    return (react_1.default.createElement("div", __assign({ style: { height: 1, borderBottom: "1px solid " + ps_design_system_core_1.colorsBorder.lowOnDark } }, props)));
};
exports.TopNav = function (props) {
    var _a = props.brand, brand = _a === void 0 ? react_1.default.createElement(exports.SkillsBrand, null) : _a, children = props.children, _b = props.items, items = _b === void 0 ? [] : _b, onMobileMenuClick = props.onMobileMenuClick, rest = __rest(props, ["brand", "children", "items", "onMobileMenuClick"]);
    return (react_1.default.createElement("div", __assign({}, rest),
        react_1.default.createElement(ps_design_system_navbar_1.default, { brand: brand, items: react_1.default.createElement(react_1.default.Fragment, null,
                items.map(function (item) {
                    return (react_1.default.createElement("div", { key: item.id, style: {
                            marginRight: ps_design_system_core_1.layout.spacingXXSmall,
                            display: 'inline-block'
                        } },
                        react_1.default.createElement(ps_design_system_navitem_1.default, { icon: item.icon }, item.title)));
                }),
                children), onMobileMenuClick: onMobileMenuClick, user: react_1.default.createElement(ps_design_system_navuser_1.default, { name: "Jake", meta: "Accenture" }) })));
};
exports.SkillsBrand = function (props) {
    return react_1.default.createElement(ps_design_system_navbrand_1.default, __assign({}, props, { logo: react_1.default.createElement(exports.SkillsLogo, null), wordmark: "SKILLS" }));
};
exports.SkillsLogo = function () {
    return (react_1.default.createElement("svg", { "aria-hidden": true, viewBox: "0 0 32 32" },
        react_1.default.createElement("defs", null,
            react_1.default.createElement("linearGradient", { id: "skills-gradient", x1: "45.6377", y1: "47.4727", x2: "-32.2436", y2: "-35.2537", gradientUnits: "userSpaceOnUse" },
                react_1.default.createElement("stop", { offset: "0.03", stopColor: "#F05A28" }),
                react_1.default.createElement("stop", { offset: "0.93", stopColor: "#EB008B" }))),
        react_1.default.createElement("path", { d: "M0 0V32H32V0H0ZM9.4053 12.7438L15.088 16L9.4053 19.287V12.7438ZM9.4053 24.8503V21.6468L19.1842 16L9.4053 10.3532V7.17166L24.6955 16L9.4053 24.8503Z", fill: "url(#skills-gradient)" })));
};
