// @pluralsight/ps-design-system-field/dist/esm/react/field.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import Halo from '@pluralsight/ps-design-system-halo';
import { WarningIcon } from '@pluralsight/ps-design-system-icon';
import { combineFns, forwardRefWithStatics } from '@pluralsight/ps-design-system-util';
import Theme, { names as themeNames } from '@pluralsight/ps-design-system-theme';
import glamorDefault, * as glamorExports from 'glamor';
import React from 'react';
import { FieldContext } from './context';
import stylesheet from '../css/field';
import Input from './input';
import Label from './label';
import SubLabel from './sub-label';
import TextArea from './text-area';
import { appearances, sizes } from '../vars/index';
var glamor = glamorDefault || glamorExports;
var styles = {
    container: function (opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-field__container']), opts.disabled &&
            glamor.css(stylesheet['.psds-field__container--disabled']), opts.error && glamor.css(stylesheet['.psds-field__container--error']));
    },
    field: function (opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-field']), glamor.css(stylesheet[".psds-field--" + opts.size]), opts.hasPrefix && glamor.css(stylesheet['.psds-field--prefix']), opts.hasSuffix && glamor.css(stylesheet['.psds-field--suffix']));
    },
    halo: function () { return glamor.css(stylesheet['.psds-field__halo']); },
    prefix: function () { return glamor.css(stylesheet['.psds-field__prefix']); },
    suffix: function () { return glamor.css(stylesheet['.psds-field__suffix']); },
    errorIcon: function () { return glamor.css(stylesheet['.psds-field__error-icon']); }
};
var Field = forwardRefWithStatics(function (props, forwardedRef) {
    var children = props.children, disabled = props.disabled, error = props.error, label = props.label, onClick = props.onClick, prefix = props.prefix, _a = props.renderContainer, renderContainer = _a === void 0 ? defaultRenderContainer : _a, _b = props.renderTag, renderTag = _b === void 0 ? defaultRenderTag : _b, _c = props.size, size = _c === void 0 ? sizes.medium : _c, subLabel = props.subLabel, suffix = props.suffix, rest = __rest(props, ["children", "disabled", "error", "label", "onClick", "prefix", "renderContainer", "renderTag", "size", "subLabel", "suffix"]);
    var containerRef = React.useRef(null);
    React.useImperativeHandle(forwardedRef, function () { return containerRef.current; });
    var Container = React.useMemo(function () { return renderContainer; }, [renderContainer]);
    var Tag = React.useMemo(function () { return renderTag; }, [renderTag]);
    var focusOnClick = React.useCallback(function (evt) {
        var focusableTags = ['input', 'select', 'textarea'];
        var el = containerRef.current;
        if (!el || el.contains(document.activeElement))
            return;
        if (focusableTags.includes(getTargetTag(evt)))
            return;
        var node = el.querySelector(focusableTags.join(','));
        if (node)
            node.focus();
    }, []);
    var handleClick = combineFns(onClick, focusOnClick);
    return (React.createElement(FieldContext.Provider, { value: {
            size: size
        } },
        React.createElement(Container, __assign({}, styles.container({ disabled: disabled, error: error }), { onClick: handleClick, ref: containerRef }),
            label && label,
            React.createElement(Theme, { name: themeNames.light },
                React.createElement(Halo, __assign({ error: error, gapSize: Halo.gapSizes.small }, styles.halo()),
                    React.createElement(Tag, __assign({}, styles.field({
                        hasPrefix: !!prefix,
                        hasSuffix: !!suffix,
                        size: size
                    }), rest),
                        prefix && React.createElement("div", __assign({}, styles.prefix()), prefix),
                        children,
                        suffix && React.createElement("div", __assign({}, styles.suffix()), suffix),
                        error && (React.createElement("div", __assign({}, styles.errorIcon()),
                            React.createElement(WarningIcon, null)))))),
            subLabel && subLabel)));
});
var defaultRenderContainer = React.forwardRef(function (props, ref) { return React.createElement("div", __assign({ ref: ref }, props)); });
var defaultRenderTag = function (props) { return React.createElement("div", __assign({}, props)); };
var getTargetTag = function (evt) {
    if (!(evt.target instanceof Element))
        return 'unknown';
    return evt.target.tagName.toLowerCase();
};
Field.displayName = 'Field';
Field.Input = Input;
Field.Label = Label;
Field.SubLabel = SubLabel;
Field.TextArea = TextArea;
Field.appearances = appearances;
Field.sizes = sizes;
export default Field;
