// @pluralsight/ps-design-system-steps/dist/cjs/react/__stories__/shared.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FixedWidthContainer = exports.useStoryData = void 0;
var core = __importStar(require("@pluralsight/ps-design-system-core"));
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = require("../../vars/index");
var glamor = glamor_1.default || glamorExports;
function useStoryData() {
    var description = 'An optional description to provide more detail about this step.';
    var _a = react_1.default.useState([
        {
            description: description,
            title: 'Select type',
            status: index_1.statuses.completed
        },
        {
            description: description,
            title: 'Select template',
            status: index_1.statuses.current
        },
        {
            description: description,
            title: 'Add details',
            status: index_1.statuses.incomplete
        },
        {
            description: description,
            title: 'Select goals',
            status: index_1.statuses.incomplete
        },
        {
            description: description,
            title: 'Review',
            status: index_1.statuses.incomplete
        }
    ]), steps = _a[0], setSteps = _a[1];
    var selectStep = react_1.default.useCallback(function (nextStep) {
        var curIndex = steps.indexOf(nextStep);
        if (curIndex < 0)
            return;
        setSteps(function (prev) {
            return prev.map(function (s, i) {
                var status = i === curIndex
                    ? 'current'
                    : i < curIndex
                        ? 'completed'
                        : 'incomplete';
                return __assign(__assign({}, s), { status: status });
            });
        });
    }, [steps]);
    return { steps: steps, selectStep: selectStep };
}
exports.useStoryData = useStoryData;
exports.FixedWidthContainer = function (props) {
    var _a = props.width, width = _a === void 0 ? 300 : _a, rest = __rest(props, ["width"]);
    return (react_1.default.createElement("div", __assign({}, glamor.css({ padding: core.layout.spacingMedium, width: width }), rest)));
};
