// @pluralsight/ps-design-system-navbrand/dist/cjs/css/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
exports.default = {
    '.psds-navbrand': {
        display: 'flex',
        alignItems: 'center',
        padding: ps_design_system_core_1.layout.spacingXSmall,
        borderRadius: '2px',
        height: '40px'
    },
    // TODO: handle hover and focus
    '.psds-navbrand--clickable': {
        cursor: 'pointer',
        background: 'none',
        border: 'none',
        textDecoration: 'none',
        '&:hover, &:focus': {
            background: "rgba(" + ps_design_system_core_1.colorsBackgroundUtilityCsv + ",0.15)"
        },
        '&:focus': {
            outline: 'none'
        },
        '&:active': {
            background: "rgba(" + ps_design_system_core_1.colorsBackgroundUtilityCsv + ",0.25)"
        }
    },
    '.psds-navbrand__logo': {
        height: '24px',
        width: '24px',
        '& svg': {
            height: '100%',
            width: '100%'
        }
    },
    '.psds-navbrand__wordmark': {
        display: 'none',
        lineHeight: 0,
        color: ps_design_system_core_1.colorsWhite,
        fontWeight: 900,
        fontSize: '20px',
        letterSpacing: '-.03em',
        whiteSpace: 'nowrap'
    },
    '@media (min-width: 1200px)': {
        '.psds-navbrand__logo--wordmark': {
            marginRight: ps_design_system_core_1.layout.spacingXSmall
        },
        '.psds-navbrand__wordmark': {
            display: 'block'
        }
    }
};
