// @pluralsight/ps-design-system-switch/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.labelAligns = exports.sizes = exports.colors = void 0;
var ps_design_system_halo_1 = __importDefault(require("@pluralsight/ps-design-system-halo"));
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    switch: function (disabled, labelAlign) {
        return glamor.compose(glamor.css(index_1.default['.psds-switch']), glamor.css(index_1.default[".psds-switch--labelAlign-" + labelAlign]), disabled && glamor.css(index_1.default['.psds-switch--disabled']));
    },
    track: function (props) {
        return glamor.compose(glamor.css(index_1.default['.psds-switch__track']), glamor.css(index_1.default[".psds-switch__track.psds-theme--" + props.themeName]), props.checked &&
            glamor.css(index_1.default[".psds-switch__track--checked.psds-switch__track--color-" + props.color]), glamor.css(index_1.default[".psds-switch__track.psds-switch__track--size-" + props.size]));
    },
    thumb: function (checked, size) {
        return glamor.compose(glamor.css(index_1.default['.psds-switch__thumb']), glamor.css(index_1.default[".psds-switch__thumb--size-" + size]), checked &&
            glamor.css(index_1.default[".psds-switch__thumb--checked.psds-switch__thumb--size-" + size]));
    },
    label: function (props) {
        return glamor.compose(glamor.css(index_1.default['.psds-switch__label']), glamor.css(index_1.default[".psds-switch__label--size-" + props.size]), glamor.css(index_1.default[".psds-switch__label--size-" + props.size + ".psds-switch__label--labelAlign-" + props.labelAlign]), glamor.css(index_1.default[".psds-switch__label.psds-theme--" + props.themeName]));
    },
    checkbox: function () { return glamor.css(index_1.default['.psds-switch__checkbox']); }
};
var Switch = react_1.default.forwardRef(function (_a, forwardedRef) {
    var ariaLabel = _a["aria-label"], _b = _a.checked, checked = _b === void 0 ? false : _b, _c = _a.color, color = _c === void 0 ? vars.colors.orange : _c, _d = _a.disabled, disabled = _d === void 0 ? false : _d, _e = _a.error, error = _e === void 0 ? false : _e, _f = _a.labelAlign, labelAlign = _f === void 0 ? vars.labelAligns.right : _f, name = _a.name, _g = _a.size, size = _g === void 0 ? vars.sizes.large : _g, _h = _a.tabIndex, tabIndex = _h === void 0 ? 0 : _h, onClick = _a.onClick, onBlur = _a.onBlur, onFocus = _a.onFocus, children = _a.children, style = _a.style, className = _a.className, rest = __rest(_a, ['aria-label', "checked", "color", "disabled", "error", "labelAlign", "name", "size", "tabIndex", "onClick", "onBlur", "onFocus", "children", "style", "className"]);
    var ref = react_1.default.useRef();
    react_1.default.useImperativeHandle(forwardedRef, 
    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
    function () { return ref.current; });
    var themeName = ps_design_system_theme_1.useTheme();
    var _j = react_1.default.useState(false), isFocused = _j[0], setIsFocused = _j[1];
    var handleBlur = function (evt) {
        setIsFocused(false);
        onBlur && onBlur(evt);
    };
    function handleClick(evt) {
        if (evt.type === 'click' ||
            ('key' in evt && evt.type === 'keydown' && evt.key === ' ')) {
            evt.preventDefault();
            onClick && onClick(!checked);
        }
    }
    var handleFocus = function (evt) {
        setIsFocused(true);
        onFocus && onFocus(evt);
    };
    return (react_1.default.createElement("label", __assign({ className: className, style: style }, styles.switch(disabled, labelAlign), rest),
        react_1.default.createElement(ps_design_system_halo_1.default, { error: error, shape: ps_design_system_halo_1.default.shapes.pill, inline: true, visible: isFocused },
            react_1.default.createElement("div", __assign({}, styles.track({ themeName: themeName, checked: checked, color: color, size: size })),
                react_1.default.createElement("div", __assign({}, styles.thumb(checked, size))))),
        react_1.default.createElement("input", __assign({ "aria-label": ariaLabel, checked: checked, disabled: disabled, readOnly: true, ref: ref, name: name, onKeyDown: handleClick, onClick: disabled ? undefined : handleClick, onBlur: handleBlur, onFocus: handleFocus, tabIndex: disabled ? -1 : tabIndex, type: "checkbox" }, styles.checkbox())),
        children && (react_1.default.createElement("span", __assign({}, styles.label({ themeName: themeName, labelAlign: labelAlign, size: size })), children))));
});
Switch.displayName = 'Switch';
Switch.colors = vars.colors;
Switch.sizes = vars.sizes;
Switch.labelAligns = vars.labelAligns;
exports.colors = vars.colors;
exports.sizes = vars.sizes;
exports.labelAligns = vars.labelAligns;
exports.default = Switch;
