// @pluralsight/ps-design-system-avatar/dist/cjs/vars/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.widths = exports.fallbackPixel = exports.colorByLetter = exports.colors = exports.sizes = void 0;
var core = __importStar(require("@pluralsight/ps-design-system-core"));
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
exports.sizes = ps_design_system_util_1.keyMirror('xSmall', 'small', 'medium', 'large', 'xLarge');
exports.colors = [
    core.colorsBlue[7],
    core.colorsPurple[7],
    core.colorsGreen[7],
    core.colorsRed[7],
    core.colorsTeal[8],
    core.colorsPink[7],
    core.colorsLime[8],
    core.colorsOrange[8],
    core.colorsYellow[9]
];
var letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
exports.colorByLetter = letters.reduce(function (map, letter, i) {
    map[letter] = exports.colors[i % exports.colors.length];
    return map;
}, {});
exports.fallbackPixel = 'https://s3-us-west-2.amazonaws.com/ps-cdn/design-system/assets/transparent.gif';
exports.widths = {
    xSmall: '32px',
    small: '48px',
    medium: '72px',
    large: '96px',
    xLarge: '160px'
};
