// @pluralsight/ps-design-system-radio/dist/cjs/react/button.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_halo_1 = __importDefault(require("@pluralsight/ps-design-system-halo"));
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var context_1 = require("./context");
var index_1 = __importDefault(require("../css/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    button: function (disabled) {
        return glamor.compose(glamor.css(index_1.default['.psds-radio-button']), disabled && glamor.css(index_1.default['.psds-radio-button--disabled']));
    },
    circle: function (themeName, checked) {
        return glamor.compose(glamor.css(index_1.default['.psds-radio-button__circle']), glamor.css(index_1.default[".psds-radio-button__circle.psds-theme--" + themeName]), checked && glamor.css(index_1.default['.psds-radio-button__circle--checked']));
    },
    circleOuter: function () { return glamor.css(index_1.default['.psds-radio-button__circle-outer']); },
    circleInner: function () { return glamor.css(index_1.default['.psds-radio-button__circle-inner']); },
    halo: function () { return glamor.css(index_1.default['.psds-radio-button__halo']); },
    input: function () { return glamor.css(index_1.default['.psds-radio-button__input']); },
    label: function (themeName) {
        return glamor.compose(glamor.css(index_1.default['.psds-radio-button__label']), glamor.css(index_1.default[".psds-radio-button__label.psds-theme--" + themeName]));
    }
};
var isChecked = function (a, b) { return a === b; };
var Button = react_1.default.forwardRef(function (_a, forwardedRef) {
    var value = _a.value, label = _a.label, props = __rest(_a, ["value", "label"]);
    var themeName = ps_design_system_theme_1.useTheme();
    var _b = react_1.default.useContext(context_1.RadioContext), checkedValue = _b.checkedValue, onChange = _b.onChange, disabled = _b.disabled, error = _b.error, name = _b.name;
    var ref = react_1.default.useRef(null);
    react_1.default.useImperativeHandle(forwardedRef, function () { return ref.current; });
    var _c = react_1.default.useState(false), isFocused = _c[0], setFocus = _c[1];
    var handleFocus = function (evt) {
        if (disabled)
            return;
        ps_design_system_util_1.combineFns(function (_evt) { return setFocus(true); }, props.onFocus)(evt);
    };
    var handleBlur = function (evt) {
        if (disabled)
            return;
        ps_design_system_util_1.combineFns(props.onBlur, function () { return setFocus(false); })(evt);
    };
    var handleClick = function (evt) {
        var value = evt.target.value;
        ps_design_system_util_1.combineFns(onChange, props.onClick)(evt, value);
        ref.current.focus();
    };
    var checked = isChecked(value, checkedValue);
    return (react_1.default.createElement("label", __assign({}, styles.button(disabled)),
        react_1.default.createElement("div", __assign({}, styles.circleOuter()),
            react_1.default.createElement(ps_design_system_halo_1.default, __assign({ error: error, inline: true, shape: ps_design_system_halo_1.default.shapes.pill, visibleOnFocus: !disabled, visible: isFocused }, styles.halo()),
                react_1.default.createElement("div", __assign({}, styles.circle(themeName, checked)), checked && react_1.default.createElement("div", __assign({}, styles.circleInner()))))),
        react_1.default.createElement("input", __assign({}, props, { checked: checked, onClick: disabled ? undefined : handleClick, onFocus: handleFocus, onBlur: handleBlur, type: "radio", readOnly: true, name: name, ref: ref, value: value }, styles.input())),
        react_1.default.createElement("div", __assign({}, styles.label(themeName)), label)));
});
Button.displayName = 'Radio.Button';
exports.default = Button;
