// @pluralsight/ps-design-system-radio/dist/cjs/react/group.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var context_1 = require("./context");
var index_1 = __importDefault(require("../css/index"));
var glamor = glamor_1.default || glamorExports;
var styles = {
    buttonContainer: function () {
        return glamor.css(index_1.default['.psds-radio-group__button-container']);
    },
    group: function (disabled) {
        return glamor.css(index_1.default['.psds-radio-group'], disabled && index_1.default['.psds-radio-group--disabled']);
    },
    label: function (themeName) {
        return glamor.css(index_1.default['.psds-radio-group__label'], index_1.default[".psds-radio-group__label.psds-theme--" + themeName]);
    },
    subLabel: function (themeName) {
        return glamor.css(index_1.default['.psds-radio-group__sub-label'], index_1.default[".psds-radio-group__sub-label.psds-theme--" + themeName]);
    }
};
var useValue = function (_a) {
    var value = _a.value, onChange = _a.onChange;
    var _b = react_1.default.useState(value), _value = _b[0], setValue = _b[1];
    return __assign({}, (value !== undefined && onChange !== undefined
        ? { checkedValue: value, onChange: onChange }
        : {
            checkedValue: _value,
            onChange: function (evt, val) {
                return setValue(val);
            }
        }));
};
var Group = react_1.default.forwardRef(function (_a, ref) {
    var children = _a.children, _b = _a.disabled, disabled = _b === void 0 ? false : _b, _c = _a.error, error = _c === void 0 ? false : _c, label = _a.label, name = _a.name, onChange = _a.onChange, subLabel = _a.subLabel, value = _a.value, rest = __rest(_a, ["children", "disabled", "error", "label", "name", "onChange", "subLabel", "value"]);
    var themeName = ps_design_system_theme_1.useTheme();
    return (react_1.default.createElement("div", __assign({}, styles.group(disabled), rest, { ref: ref, role: "radiogroup" }),
        label && react_1.default.createElement("div", __assign({}, styles.label(themeName)), label),
        react_1.default.createElement("div", __assign({}, styles.buttonContainer()),
            react_1.default.createElement(context_1.RadioContext.Provider, { value: __assign({ disabled: disabled,
                    error: error,
                    name: name }, useValue({ value: value, onChange: onChange })) }, children)),
        subLabel && react_1.default.createElement("div", __assign({}, styles.subLabel(themeName)), subLabel)));
});
Group.displayName = 'Radio.Group';
exports.default = Group;
