// @pluralsight/ps-design-system-datepicker/dist/cjs/css/index.js
"use strict";
var _a, _b, _c;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
exports.default = {
    '@keyframes psds-calendar__keyframes__forward': {
        from: {
            transform: "translateX(calc(-1 * " + ps_design_system_core_1.layout.spacingXSmall + "))"
        },
        to: {
            transform: 'translateX(0)'
        }
    },
    '@keyframes psds-calendar__keyframes__backward': {
        from: {
            transform: "translateX(" + ps_design_system_core_1.layout.spacingXSmall + ")"
        },
        to: {
            transform: 'translateX(0)'
        }
    },
    '.psds-calendar': {
        padding: '24px',
        maxHeight: 360,
        backgroundColor: ps_design_system_core_1.colorsWhite,
        color: ps_design_system_core_1.colorsTextIcon.highOnLight,
        display: 'inline-flex',
        flexDirection: 'column'
    },
    '.psds-calendar__grid-wrapper': {
        transition: "height " + ps_design_system_core_1.motion.speedNormal
    },
    '.psds-calendar__grid-slide': {
        display: 'flex',
        flex: 1,
        alignItems: 'flex-start'
    },
    '.psds-calendar__grid-slide--forward': function (forward) { return ({
        animation: (forward || 'psds-calendar__keyframes__forward') + " " + ps_design_system_core_1.motion.speedNormal + " forwards"
    }); },
    '.psds-calendar__grid-slide--backward': function (backward) { return ({
        animation: (backward || 'psds-calendar__keyframes__backward') + " " + ps_design_system_core_1.motion.speedNormal + " forwards"
    }); },
    '.psds-calendar__month': {
        display: 'flex',
        padding: '8px',
        alignItems: 'center',
        width: '100%'
    },
    '.psds-calendar__header-wrapper': {
        display: 'flex'
    },
    '.psds-calendar__header': {
        width: 280,
        '&:not(:last-child)': {
            marginRight: ps_design_system_core_1.layout.spacingLarge
        }
    },
    '.psds-calendar__header-button': {},
    '.psds-calendar__header-month': {
        display: 'flex',
        flex: 1,
        flexDirection: 'column',
        fontSize: '16px',
        lineHeight: '20px',
        textAlign: 'center'
    },
    '.psds-calendar__weekday-header': {
        display: 'inline-flex',
        justifyContent: 'center',
        alignItems: 'center',
        width: 40,
        height: '32px',
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        fontWeight: ps_design_system_core_1.type.fontWeightBook
    },
    '.psds-calendar__date-grid': {
        display: 'flex',
        flexWrap: 'wrap',
        zIndex: 1,
        position: 'relative',
        width: 280,
        '&:not(:last-child)': {
            marginRight: ps_design_system_core_1.layout.spacingLarge
        }
    },
    '.psds-calendar__date': (_a = {
            width: 40,
            height: '40px',
            padding: 0,
            boxSizing: 'border-box',
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            fontWeight: 600,
            fontSize: ps_design_system_core_1.type.fontSizeXSmall,
            lineHeight: '16px',
            textAlign: 'center',
            alignItems: 'center',
            justifyItems: 'center',
            backgroundColor: 'transparent',
            border: 'none'
        },
        _a["&:focus"] = {
            outline: 'none'
        },
        _a["&:hover"] = {
            borderRadius: '50%',
            border: "2px solid " + ps_design_system_core_1.colorsBlue[6],
            background: ps_design_system_core_1.colorsBlue[2],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        },
        _a["&:disabled"] = {
            color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
            textDecoration: 'line-through'
        },
        _a),
    '.psds-calendar__date--in-range': (_b = {
            transition: "background-color " + ps_design_system_core_1.motion.speedNormal,
            background: ps_design_system_core_1.colorsBlue[2]
        },
        _b["&:hover"] = {
            borderRadius: '0',
            border: "none",
            background: ps_design_system_core_1.colorsBlue[3],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        },
        _b),
    '.psds-calendar__date--selected': (_c = {
            borderRadius: '50%',
            border: "2px solid " + ps_design_system_core_1.colorsBlue[6],
            background: ps_design_system_core_1.colorsBlue[6],
            color: ps_design_system_core_1.colorsTextIcon.highOnDark,
            position: 'relative'
        },
        _c["&:hover"] = {
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            background: ps_design_system_core_1.colorsBlue[2]
        },
        _c),
    '.psds-calendar__date--selected-start': {
        position: 'relative',
        '&:after': {
            content: ' ',
            display: 'block',
            position: 'absolute',
            zIndex: -1,
            top: '-2px',
            transform: 'translateX(100%)',
            width: 'calc(1px + 50%)',
            height: '40px',
            background: ps_design_system_core_1.colorsBlue[2]
        }
    },
    '.psds-calendar__date--selected-end': {
        position: 'relative',
        '&:after': {
            content: ' ',
            display: 'block',
            position: 'absolute',
            top: '-2px',
            zIndex: -1,
            transform: 'translateX(-2px)',
            width: 'calc(2px + 50%)',
            height: '40px',
            background: ps_design_system_core_1.colorsBlue[2]
        }
    },
    '.psds-calendar__date--today': {},
    '.psds-calendar__filler': {
        width: 40,
        height: '40px'
    }
};
