// @pluralsight/ps-design-system-dialog/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.tailPositions = void 0;
var ps_design_system_focusmanager_1 = __importDefault(require("@pluralsight/ps-design-system-focusmanager"));
var ps_design_system_theme_1 = __importDefault(require("@pluralsight/ps-design-system-theme"));
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
/* eslint-disable-next-line camelcase */
var MODAL_OVERLAY_ID = 'psds-dialog__overlay';
var fade = glamor.keyframes(index_1.default['@keyframes psds-dialog__keyframes__fade']);
var styles = {
    dialog: function (modal, tailPosition) {
        return glamor.compose(glamor.css(index_1.default['.psds-dialog']({ fade: fade })), modal && glamor.css(index_1.default['.psds-dialog--modal']), Boolean(tailPosition) &&
            glamor.compose(index_1.default['.psds-dialog--w-tail'], index_1.default[".psds-dialog--tailPosition-" + tailPosition]));
    },
    content: function (props) {
        return glamor.css(index_1.default['.psds-dialog__content'], ps_design_system_util_1.stylesFor('dialog__content', props));
    },
    close: function () { return glamor.css(index_1.default['.psds-dialog__close']); },
    overlay: function () { return glamor.css(index_1.default['.psds-dialog__overlay']); }
};
var CloseButton = function (props) { return (react_1.default.createElement("button", __assign({}, styles.close(), props, { "aria-label": "Close dialog" }),
    react_1.default.createElement("svg", { "aria-label": "close icon", role: "img", viewBox: "0 0 24 24", xmlns: "http://www.w3.org/2000/svg" },
        react_1.default.createElement("path", { d: "M18 7.41L16.59 6 12 10.59 7.41 6 6 7.41 10.59 12 6 16.59 7.41 18 12 13.41 16.59 18 18 16.59 13.41 12" })))); };
var Overlay = function (_a) {
    var disableCloseOnOverlayClick = _a.disableCloseOnOverlayClick, onClose = _a.onClose, props = __rest(_a, ["disableCloseOnOverlayClick", "onClose"]);
    function handleOverlayClick(evt) {
        if (disableCloseOnOverlayClick)
            return;
        if (evt.target.id === MODAL_OVERLAY_ID)
            onClose && onClose(evt);
    }
    return (react_1.default.createElement("div", __assign({}, styles.overlay(), props, { id: MODAL_OVERLAY_ID, onClick: handleOverlayClick, role: "region" })));
};
var Dialog = react_1.default.forwardRef(function (props, ref) {
    var children = props.children, _a = props.disableCloseButton, disableCloseButton = _a === void 0 ? false : _a, _b = props.disableCloseOnEscape, disableCloseOnEscape = _b === void 0 ? false : _b, _c = props.disableCloseOnOverlayClick, disableCloseOnOverlayClick = _c === void 0 ? false : _c, _d = props.disableFocusOnMount, disableFocusOnMount = _d === void 0 ? false : _d, onClose = props.onClose, _e = props.modal, modal = _e === void 0 ? false : _e, _f = props.returnFocus, returnFocus = _f === void 0 ? true : _f, tailPosition = props.tailPosition, 
    /* eslint-disable-next-line camelcase */
    UNSAFE_stylesFor = props.UNSAFE_stylesFor, rest = __rest(props, ["children", "disableCloseButton", "disableCloseOnEscape", "disableCloseOnOverlayClick", "disableFocusOnMount", "onClose", "modal", "returnFocus", "tailPosition", "UNSAFE_stylesFor"]);
    var autofocus = !disableFocusOnMount;
    var trapped = !!modal || !!onClose;
    var closeOnEscape = ps_design_system_util_1.isFunction(onClose) && !disableCloseOnEscape;
    var portal = ps_design_system_util_1.usePortal();
    // TODO: combine fns
    function handleKeyUp(evt) {
        if (!isEscape(evt))
            return;
        onClose && onClose(evt);
    }
    var content = (react_1.default.createElement(ps_design_system_focusmanager_1.default, __assign({}, styles.dialog(modal, tailPosition), rest, (closeOnEscape && { onKeyUp: handleKeyUp }), (modal && { 'aria-label': undefined }), { autofocus: autofocus, trapped: trapped, returnFocus: returnFocus, ref: ref }),
        react_1.default.createElement(ps_design_system_theme_1.default, { name: ps_design_system_theme_1.default.names.light },
            react_1.default.createElement("div", __assign({}, styles.content(props)),
                !disableCloseButton && ps_design_system_util_1.isFunction(onClose) && (
                // eslint-disable-next-line react/jsx-handler-names
                react_1.default.createElement(CloseButton, { onClick: onClose })),
                children))));
    return (react_1.default.createElement(react_1.default.Fragment, null, modal
        ? ps_design_system_util_1.createUniversalPortal(react_1.default.createElement(Overlay, { "aria-label": rest['aria-label'], disableCloseOnOverlayClick: disableCloseOnOverlayClick, onClose: onClose }, content), portal.current)
        : content));
});
Dialog.displayName = 'Dialog';
Dialog.tailPositions = vars.tailPositions;
exports.tailPositions = Dialog.tailPositions;
exports.default = Dialog;
function isEscape(evt) {
    return evt.key === 'Escape';
}
