// @pluralsight/ps-design-system-util/dist/esm/use-close-on-document-events.js
import React from 'react';
import { canUseDOM } from './can-use-dom';
export var onClickOutside = function (el, callback) {
    if (!canUseDOM())
        return noop;
    var handleClick = function (evt) {
        if (evt.target instanceof HTMLElement ||
            evt.target instanceof HTMLDocument) {
            if (el.contains(evt.target))
                return;
            callback(evt);
        }
    };
    document.addEventListener('click', handleClick, { capture: true });
    return function () {
        document.removeEventListener('click', handleClick, {
            capture: true
        });
    };
};
export var useEvent = function (eventHandler) { return function (ref, cb) {
    if (cb === void 0) { cb = noop; }
    /* eslint-disable-next-line react-hooks/rules-of-hooks */
    React.useEffect(function () {
        if (!canUseDOM())
            return noop;
        var el = ref.current;
        if (el) {
            return eventHandler(el, cb);
        }
        return noop;
    }, [ref, cb]);
}; };
export var useClickOutside = function (ref, cb) {
    if (cb === void 0) { cb = noop; }
    return useEvent(onClickOutside)(ref, cb);
};
export var onResize = function (_el, callback) {
    if (!canUseDOM())
        return noop;
    var currentAnimationFrame;
    var handleResize = function (evt) {
        window.cancelAnimationFrame(currentAnimationFrame);
        currentAnimationFrame = window.requestAnimationFrame(function () { return callback(evt); });
        return currentAnimationFrame;
    };
    window.addEventListener('resize', handleResize, {
        passive: true,
        capture: true
    });
    return function () {
        window.removeEventListener('resize', handleResize, {
            capture: true
        });
    };
};
export var useResize = function (ref, cb) {
    if (cb === void 0) { cb = noop; }
    return useEvent(onResize)(ref, cb);
};
export var onScrollOutside = function (el, callback) {
    if (!canUseDOM())
        return noop;
    var currentAnimationFrame;
    var handleScroll = function (evt) {
        window.cancelAnimationFrame(currentAnimationFrame);
        if (evt.target instanceof HTMLElement ||
            evt.target instanceof HTMLDocument) {
            if (el.contains(evt.target))
                return;
            currentAnimationFrame = window.requestAnimationFrame(function () { return callback(evt); });
        }
        return currentAnimationFrame;
    };
    document.addEventListener('scroll', handleScroll, {
        passive: true,
        capture: true
    });
    return function () {
        window.removeEventListener('scroll', handleScroll, {
            capture: true
        });
    };
};
export var useScrollOutside = function (ref, cb) {
    if (cb === void 0) { cb = noop; }
    return useEvent(onScrollOutside)(ref, cb);
};
export var onGlobalEventsClose = function (el, callback) {
    if (!canUseDOM())
        return noop;
    var handleClickOutsideMenu = function (evt) {
        if (evt.target instanceof HTMLElement) {
            if (el.contains(evt.target))
                return;
            callback(evt);
        }
    };
    var currentAnimationFrame;
    var requestAnimationFrame = function (evt) {
        window.cancelAnimationFrame(currentAnimationFrame);
        currentAnimationFrame = window.requestAnimationFrame(function () { return callback(evt); });
        return currentAnimationFrame;
    };
    document.addEventListener('click', handleClickOutsideMenu, { capture: true });
    window.addEventListener('resize', requestAnimationFrame, {
        passive: true,
        capture: true
    });
    document.addEventListener('scroll', requestAnimationFrame, {
        passive: true,
        capture: true
    });
    return function () {
        document.removeEventListener('click', handleClickOutsideMenu, {
            capture: true
        });
        window.removeEventListener('resize', requestAnimationFrame, {
            capture: true
        });
        window.removeEventListener('scroll', requestAnimationFrame, {
            capture: true
        });
    };
};
export var useCloseOnDocumentEvents = function (ref, cb) {
    if (cb === void 0) { cb = noop; }
    return React.useEffect(function () {
        if (!canUseDOM())
            return noop;
        var el = ref.current;
        if (el) {
            return onGlobalEventsClose(el, cb);
        }
        return noop;
    }, [ref, cb]);
};
// eslint-disable-next-line @typescript-eslint/no-empty-function
var noop = function () { };
