// @pluralsight/ps-design-system-banner/dist/esm/css/index.js
var _a;
import { colorsBlack, colorsWhite, colorsTextIcon, layout, motion, colorsBlue, colorsGreen, colorsYellow, colorsRed } from '@pluralsight/ps-design-system-core';
import { transparentize } from '@pluralsight/ps-design-system-util';
import * as vars from '../vars/index';
export default (_a = {
        '.psds-banner': {
            display: 'flex',
            alignItems: 'flex-start',
            width: '100%',
            overflow: 'hidden',
            padding: layout.spacingXSmall + " " + layout.spacingSmall
        }
    },
    _a[".psds-banner--color-" + vars.colors.blue] = {
        color: colorsTextIcon.highOnDark,
        backgroundColor: colorsBlue[7]
    },
    _a[".psds-banner--color-" + vars.colors.green] = {
        color: colorsTextIcon.highOnDark,
        backgroundColor: colorsGreen[8]
    },
    _a[".psds-banner--color-" + vars.colors.yellow] = {
        color: colorsTextIcon.highOnLight,
        backgroundColor: colorsYellow[6]
    },
    _a[".psds-banner--color-" + vars.colors.red] = {
        color: colorsTextIcon.highOnDark,
        backgroundColor: colorsRed[7]
    },
    _a['.psds-banner__button'] = {
        ':not(#fake-id-to-force-higher-specificity) &': {
            background: 'none',
            borderColor: 'currentColor',
            color: 'currentColor',
            '&:hover, &:active, &:focus': {
                backgroundColor: transparentize(0.9, colorsBlack)
            },
            '&:focus': { boxShadow: "0 0 0 3px " + transparentize(0.5, colorsWhite) }
        }
    },
    _a[".psds-banner__button--color-" + vars.colors.blue] = {},
    _a[".psds-banner__button--color-" + vars.colors.green] = {},
    _a[".psds-banner__button--color-" + vars.colors.yellow] = {
        ':not(#fake-id-to-force-higher-specificity)': {
            '&:focus': { boxShadow: "0 0 0 3px " + transparentize(0.5, colorsBlack) }
        }
    },
    _a[".psds-banner__button--color-" + vars.colors.red] = {},
    _a['.psds-banner__text'] = {
        margin: '0 !important',
        flex: '1',
        textAlign: 'center',
        color: 'currentColor !important',
        '& a': {
            color: 'inherit',
            cursor: 'pointer',
            textDecoration: 'underline',
            '&:hover, &:active, &:focus': {
                color: 'inherit',
                transition: "all " + motion.speedNormal,
                opacity: '0.85'
            }
        }
    },
    _a['.psds-banner__dismiss'] = {
        position: 'relative',
        display: 'flex',
        alignItems: 'flex-start',
        background: 'none',
        border: 'none',
        marginLeft: layout.spacingSmall,
        padding: '0',
        color: 'currentColor',
        cursor: 'pointer',
        '&:hover, &:focus, &:active': {
            outline: 'none',
            borderRadius: '2px',
            boxShadow: '0 0 0 3px currentColor'
        }
    },
    _a);
