// @pluralsight/ps-design-system-field/dist/cjs/css/field.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_icon_1 = require("@pluralsight/ps-design-system-icon");
var index_1 = require("../vars/index");
exports.default = (_a = {
        '.psds-field__container': {
            label: 'field__container',
            display: 'inline-block'
        },
        '.psds-field__container--disabled': {
            opacity: 0.5,
            cursor: 'not-allowed'
        },
        '.psds-field__container--error': {},
        '.psds-field': {
            label: 'field',
            alignItems: 'center',
            background: ps_design_system_core_1.colorsBackgroundLight[3],
            border: "1px solid " + ps_design_system_core_1.colorsBorder.highOnLight,
            borderRadius: '2px',
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            display: 'flex',
            fontWeight: ps_design_system_core_1.type.fontWeightDefault,
            padding: "0 " + ps_design_system_core_1.layout.spacingMedium,
            lineHeight: '20px',
            position: 'relative',
            width: '100%'
        }
    },
    _a[".psds-field--" + index_1.sizes.small] = {
        padding: "0 " + ps_design_system_core_1.layout.spacingXSmall
    },
    _a['.psds-field--prefix'] = {
        paddingLeft: '11px'
    },
    _a['.psds-field--suffix'] = {
        paddingRight: '11px'
    },
    _a['.psds-field__halo'] = { width: '100%' },
    _a['.psds-field__prefix'] = {
        label: 'field__prefix',
        alignSelf: 'stretch',
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        display: 'flex',
        alignItems: 'center',
        flex: '0 0 auto',
        height: 'auto',
        marginRight: ps_design_system_core_1.layout.spacingXSmall
    },
    _a['.psds-field__suffix'] = {
        label: 'field__suffix',
        alignSelf: 'stretch',
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        display: 'flex',
        alignItems: 'center',
        flex: '0 0 auto',
        height: 'auto',
        marginLeft: ps_design_system_core_1.layout.spacingXSmall,
        overflow: 'visible'
    },
    _a['.psds-field__error-icon'] = {
        label: 'field__error-icon',
        alignItems: 'center',
        color: ps_design_system_core_1.colorsStatus.error,
        display: 'flex',
        position: 'absolute',
        right: "calc(-1 * (" + ps_design_system_icon_1.widths.medium + " + " + ps_design_system_core_1.layout.spacingXSmall + "))"
    },
    _a);
