// @pluralsight/ps-design-system-scrollable/dist/esm/react/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { canUseDOM, combineFns, shallowCompare, useResizeObserver } from '@pluralsight/ps-design-system-util';
import glamorDefault, * as glamorExports from 'glamor';
import React from 'react';
import stylesheet from '../css/index';
var glamor = glamorDefault || glamorExports;
var styles = {
    outer: function () {
        return glamor.compose(glamor.css(stylesheet['.psds-scrollable__outer']), glamor.media('not print', stylesheet['.psds-scrollable__outer--screen']));
    },
    inner: function () { return glamor.css(stylesheet['.psds-scrollable__inner']); },
    content: function () { return glamor.css(stylesheet['.psds-scrollable__content']); },
    handle: function (grabbed) {
        return glamor.compose(glamor.css(stylesheet['.psds-scrollable__handle']), grabbed && glamor.css(stylesheet['.psds-scrollable__handle--grabbed']));
    }
};
var BLANK_IMAGE;
if (canUseDOM()) {
    BLANK_IMAGE = new Image(0, 0);
    // prettier-ignore
    BLANK_IMAGE.src = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
}
function areEqualProps(prevProps, nextProps) {
    var changed = shallowCompare(prevProps, nextProps);
    return !changed;
}
var Scrollable = React.forwardRef(function (props, forwardedRef) {
    var _a = props.renderContent, renderContent = _a === void 0 ? defaultRenderContent : _a, rest = __rest(props, ["renderContent"]);
    var ref = React.useRef();
    /* eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion */
    React.useImperativeHandle(forwardedRef, function () { return ref.current; });
    var dragPreview = React.useRef(BLANK_IMAGE);
    var _b = React.useState(true), hidden = _b[0], setHidden = _b[1];
    var _c = React.useState(0), pageY = _c[0], setPageY = _c[1];
    var _d = React.useState('0%'), offset = _d[0], setOffset = _d[1];
    var _e = React.useState(0), scrollRatio = _e[0], setScrollRatio = _e[1];
    var updateDimensions = React.useCallback(function () {
        if (!ref.current)
            return;
        var content = ref.current;
        var nextTotalHeight = content.scrollHeight;
        var nextOffset = (content.scrollTop / nextTotalHeight) * 100 + '%';
        var nextScrollRatio = content.clientHeight / nextTotalHeight;
        var shouldHide = nextScrollRatio >= 1;
        setHidden(shouldHide);
        setOffset(nextOffset);
        setScrollRatio(nextScrollRatio);
    }, [ref]);
    useResizeObserver(ref, updateDimensions);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    React.useEffect(function () { return updateDimensions(); }, []);
    function onHandleDrag(evt) {
        var nextPageY = evt.pageY;
        if (nextPageY <= 0)
            return;
        var content = ref.current;
        if (!content)
            return;
        var delta = evt.pageY - pageY;
        setPageY(nextPageY);
        content.scrollTop += delta / scrollRatio;
    }
    function onHandleDragEnd(_evt) {
        setPageY(0);
    }
    function onHandleDragStart(evt) {
        if (evt &&
            evt.dataTransfer &&
            evt.dataTransfer.setDragImage &&
            dragPreview.current) {
            evt.dataTransfer.setDragImage(dragPreview.current, 0, 0);
        }
        setPageY(evt.pageY);
    }
    var styleOverride = React.useMemo(function () {
        var height = Math.max(scrollRatio * 100, 10) + '%';
        var visibility = hidden ? 'hidden' : 'initial';
        return glamor.css({ height: height, top: offset, visibility: visibility });
    }, [hidden, offset, scrollRatio]);
    var onScroll = combineFns(updateDimensions, props.onScroll);
    return (React.createElement(Outer, __assign({ "data-scrollable": true }, rest),
        React.createElement(Inner, { key: "scrollable-wrapper" }, renderContent(__assign({ children: props.children, onScroll: onScroll }, styles.content()), function (node) { return (ref.current = node); })),
        React.createElement(Handle, __assign({ draggable: true, onDrag: onHandleDrag, onDragEnd: onHandleDragEnd, onDragStart: onHandleDragStart }, styleOverride))));
});
Scrollable.displayName = 'Scrollable';
var defaultRenderContent = function (contentProps, contentRef) { return React.createElement("div", __assign({}, contentProps, { ref: contentRef })); };
var Outer = React.memo(function OuterComp(props) {
    return React.createElement("div", __assign({}, styles.outer(), props));
}, areEqualProps);
Outer.displayName = 'Scrollable.Outer';
var Inner = React.memo(function InnerComp(props) {
    return React.createElement("div", __assign({}, styles.inner(), props));
}, areEqualProps);
Inner.displayName = 'Scrollable.Inner';
var Handle = function (props) {
    var _a = React.useState(false), grabbed = _a[0], setGrabbed = _a[1];
    var onDragEnd = combineFns(function () {
        setGrabbed(false);
    }, props.onDragEnd);
    var onDragStart = combineFns(function () {
        setGrabbed(true);
    }, props.onDragStart);
    return (React.createElement("div", __assign({}, styles.handle(grabbed), props, { onDragEnd: onDragEnd, onDragStart: onDragStart })));
};
Handle.displayName = 'Scrollable.Handle';
export default Scrollable;
