// @pluralsight/ps-design-system-button/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sizes = exports.iconAligns = exports.appearances = void 0;
var ps_design_system_icon_1 = __importStar(require("@pluralsight/ps-design-system-icon"));
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
var spin = glamor.keyframes(index_1.default['@keyframes psds-button__keyframes__spin']);
var styles = {
    button: function (_a) {
        var appearance = _a.appearance, disabled = _a.disabled, icon = _a.icon, iconAlign = _a.iconAlign, iconOnly = _a.iconOnly, layout = _a.layout, size = _a.size, themeName = _a.themeName;
        return glamor.css(index_1.default['.psds-button'], index_1.default[".psds-button--layout-" + layout], index_1.default[".psds-button--size-" + size], index_1.default[".psds-button--appearance-" + appearance], index_1.default[".psds-button.psds-theme--" + themeName], index_1.default[".psds-button--appearance-" + appearance + ".psds-theme--" + themeName], disabled && __assign(__assign(__assign({}, index_1.default[".psds-button--disabled"]), index_1.default[".psds-button--disabled.psds-theme--" + themeName]), index_1.default[".psds-button--disabled.psds-button--appearance-" + appearance]), icon &&
            !iconOnly && __assign(__assign({}, index_1.default[".psds-button--iconAlign-" + iconAlign + ".psds-button--not-iconOnly"]), index_1.default[".psds-button--iconAlign-" + iconAlign + ".psds-button--not-iconOnly.psds-button--size-" + size]), iconAlign === vars.iconAligns.right &&
            index_1.default[".psds-button--iconAlign-" + iconAlign], iconOnly && __assign(__assign({}, index_1.default[".psds-button--iconOnly"]), index_1.default[".psds-button--iconOnly.psds-button--size-" + size]));
    },
    loading: function (_a) {
        var appearance = _a.appearance, themeName = _a.themeName, size = _a.size;
        return glamor.css(index_1.default[".psds-button__loading"]({ spin: spin }), index_1.default[".psds-button__loading--size-" + size], index_1.default[".psds-button__loading--appearance-" + appearance], index_1.default[".psds-button__loading--appearance-" + appearance + ".psds-button__loading--theme-" + themeName]);
    },
    icon: function (_a) {
        var iconAlign = _a.iconAlign, iconOnly = _a.iconOnly, labelOnly = _a.labelOnly, loading = _a.loading, size = _a.size;
        return glamor.css(index_1.default['.psds-button__icon'], index_1.default[".psds-button__icon--iconAlign-" + iconAlign], index_1.default[".psds-button__icon--iconAlign-" + iconAlign + ".psds-button--size-" + size], (iconOnly || (loading && labelOnly)) &&
            index_1.default['.psds-button__icon--iconOnly'], loading && labelOnly && index_1.default['.psds-button__icon--loadingLabelOnly']);
    },
    text: function (invisible) {
        return glamor.compose(glamor.css(index_1.default[".psds-button__text"]), invisible && glamor.css(index_1.default[".psds-button__text--invisible"]));
    }
};
var mapIconSize = function (size) {
    var _a;
    var btnToIconSizes = (_a = {},
        _a[vars.sizes.xSmall] = ps_design_system_icon_1.sizes.xSmall,
        _a[vars.sizes.small] = ps_design_system_icon_1.sizes.small,
        _a[vars.sizes.medium] = ps_design_system_icon_1.sizes.medium,
        _a[vars.sizes.large] = ps_design_system_icon_1.sizes.medium,
        _a);
    return btnToIconSizes[size] ? btnToIconSizes[size] : ps_design_system_icon_1.sizes.medium;
};
var IconContainer = function (props) {
    return props.loading ? (react_1.default.createElement("div", __assign({}, styles.icon({
        iconAlign: props.iconAlign,
        iconOnly: props.iconOnly,
        labelOnly: props.labelOnly,
        loading: props.loading,
        size: props.size
    })),
        react_1.default.createElement(ps_design_system_icon_1.default, { size: mapIconSize(props.size) },
            react_1.default.createElement("span", __assign({}, styles.loading({
                appearance: props.appearance,
                size: props.size,
                themeName: props.themeName
            })))))) : props.icon ? (react_1.default.createElement("div", __assign({}, styles.icon({
        iconAlign: props.iconAlign,
        iconOnly: props.iconOnly,
        labelOnly: props.labelOnly,
        loading: props.loading,
        size: props.size
    })), react_1.default.cloneElement(props.icon, {
        size: mapIconSize(props.size)
    }))) : null;
};
var Button = react_1.default.forwardRef(function (props, forwardedRef) {
    var _a = props.appearance, appearance = _a === void 0 ? vars.appearances.primary : _a, children = props.children, _b = props.disabled, disabled = _b === void 0 ? false : _b, icon = props.icon, _c = props.iconAlign, iconAlign = _c === void 0 ? vars.iconAligns.left : _c, _d = props.layout, layout = _d === void 0 ? vars.layouts.inline : _d, _e = props.loading, loading = _e === void 0 ? false : _e, _f = props.size, size = _f === void 0 ? vars.sizes.medium : _f, rest = __rest(props, ["appearance", "children", "disabled", "icon", "iconAlign", "layout", "loading", "size"]);
    var themeName = ps_design_system_theme_1.useTheme();
    var ref = react_1.default.useRef();
    react_1.default.useImperativeHandle(forwardedRef, function () { return ref.current; });
    var hasLabel = react_1.default.Children.count(children) > 0;
    var iconOnly = !hasLabel;
    var glamorStyle = styles.button({
        appearance: appearance,
        disabled: disabled,
        icon: Boolean(icon),
        iconAlign: iconAlign,
        iconOnly: iconOnly,
        layout: layout,
        size: size,
        themeName: themeName
    });
    var iconEl = (react_1.default.createElement(IconContainer, { appearance: appearance, icon: icon, iconAlign: iconAlign, iconOnly: iconOnly, labelOnly: hasLabel && !icon, loading: loading, size: size, themeName: themeName }));
    var isLoadingAndLabelOnly = hasLabel && loading && !icon;
    var labelEl = (react_1.default.createElement("span", __assign({}, styles.text(isLoadingAndLabelOnly), { "aria-hidden": isLoadingAndLabelOnly }), children));
    var isAnchor = Boolean(rest.href);
    var Component = isAnchor ? 'a' : 'button';
    !isAnchor && delete rest.download;
    var isDisabled = isAnchor ? undefined : disabled || loading;
    var handleClick = disabled && isAnchor
        ? undefined
        : rest.onClick;
    return (react_1.default.createElement(Component, __assign({}, rest, { disabled: isDisabled }, glamorStyle, { onClick: handleClick, ref: ref }),
        iconEl,
        labelEl));
});
Button.appearances = vars.appearances;
Button.iconAligns = vars.iconAligns;
Button.layouts = vars.layouts;
Button.sizes = vars.sizes;
exports.appearances = vars.appearances;
exports.iconAligns = vars.iconAligns;
exports.sizes = vars.sizes;
exports.default = Button;
