// @pluralsight/ps-design-system-circularprogress/dist/esm/css/index.js
var _a;
import { colorsTextIcon } from '@pluralsight/ps-design-system-core';
import { defaultName as themeDefaultName, names as themeNames } from '@pluralsight/ps-design-system-theme';
import * as vars from '../vars/index';
export default (_a = {
        '@keyframes psds-circularprogress__keyframes__spin': {
            to: {
                transform: 'rotate(270deg)'
            }
        }
    },
    // --size
    _a[".psds-circularprogress--size-" + vars.sizes.small] = {
        height: '24px',
        width: '24px'
    },
    _a[".psds-circularprogress--size-" + vars.sizes.medium] = {
        height: '48px',
        width: '48px'
    },
    // __svg
    _a[".psds-circularprogress__svg"] = {
        transform: 'rotate(-90deg)'
    },
    // __svg--no-value
    _a[".psds-circularprogress__svg--no-value"] = function (_a) {
        var spin = _a.spin;
        return ({
            animation: (spin || 'psds-circularprogress__keyframes__spin') + " 1s linear infinite"
        });
    },
    // __fg
    _a[".psds-circularprogress__fg"] = {
        fill: 'transparent',
        strokeWidth: vars.style.strokeWidth + "px",
        transition: 'stroke-dashoffset 1s'
    },
    _a[".psds-circularprogress__fg.psds-theme--" + themeNames.light] = {
        stroke: colorsTextIcon.highOnLight
    },
    _a[".psds-circularprogress__fg.psds-theme--" + themeDefaultName] = {
        stroke: colorsTextIcon.highOnDark
    },
    // __bg
    _a[".psds-circularprogress__bg"] = {
        fill: 'transparent',
        strokeWidth: vars.style.strokeWidth + "px"
    },
    _a[".psds-circularprogress__bg.psds-theme--" + themeNames.light] = {
        stroke: 'rgba(0, 0, 0, 0.15)'
    },
    _a[".psds-circularprogress__bg.psds-theme--" + themeDefaultName] = {
        stroke: 'rgba(255, 255, 255, 0.15)'
    },
    _a);
