// @pluralsight/ps-design-system-radio/dist/cjs/css/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
exports.default = (_a = {
        '.psds-radio-group': {
            '&:focus': { outline: 'none' }
        },
        '.psds-radio-group--disabled': {
            opacity: '0.5'
        },
        '.psds-radio-group__button-container': {
            padding: ps_design_system_core_1.layout.spacingXXSmall + " 0"
        },
        '.psds-radio-group__label': {
            color: ps_design_system_core_1.colorsTextIcon.highOnDark,
            fontSize: ps_design_system_core_1.type.fontSize100,
            fontWeight: ps_design_system_core_1.type.fontWeight500,
            marginBottom: ps_design_system_core_1.layout.spacingXXSmall
        }
    },
    _a[".psds-radio-group__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a['.psds-radio-group__sub-label'] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        fontSize: ps_design_system_core_1.type.fontSize100,
        fontWeight: ps_design_system_core_1.type.fontWeightDefault,
        marginTop: ps_design_system_core_1.layout.spacingXXSmall
    },
    _a[".psds-radio-group__sub-label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a['.psds-radio-button'] = {
        display: 'flex',
        alignItems: 'center',
        padding: ps_design_system_core_1.layout.spacingXXSmall + " 0",
        cursor: 'pointer'
    },
    _a['.psds-radio-button--disabled'] = {
        cursor: 'not-allowed'
    },
    _a['.psds-radio-button__halo'] = {
        transform: 'translateY(3px)'
    },
    _a['.psds-radio-button__circle-outer'] = {
        width: '16px',
        marginRight: ps_design_system_core_1.layout.spacingSmall,
        pointerEvents: 'none'
    },
    _a['.psds-radio-button__circle'] = {
        position: 'relative',
        display: 'inline-block',
        height: '16px',
        width: '16px',
        borderRadius: '50%',
        border: "2px solid " + ps_design_system_core_1.colorsTextIcon.lowOnDark,
        background: ps_design_system_core_1.colorsBackgroundDark[1],
        '&:focus': { outline: 'none' }
    },
    _a[".psds-radio-button__circle.psds-theme--" + ps_design_system_theme_1.names.light] = {
        background: ps_design_system_core_1.colorsBackgroundLight[3],
        borderColor: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a['.psds-radio-button__circle--checked'] = {
        borderColor: ps_design_system_core_1.colorsBlue[6],
        background: ps_design_system_core_1.colorsBlue[6]
    },
    _a['.psds-radio-button__circle-inner'] = {
        display: 'block',
        height: 'calc(100% - 6px)',
        width: 'calc(100% - 6px)',
        margin: '3px',
        borderRadius: '50%',
        background: ps_design_system_core_1.colorsWhite
    },
    _a['.psds-radio-button__input'] = ps_design_system_core_1.accessibility.screenReaderOnly,
    _a['.psds-radio-button__label'] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        fontSize: ps_design_system_core_1.type.fontSize200,
        lineHeight: ps_design_system_core_1.type.lineHeightStandard,
        fontWeight: ps_design_system_core_1.type.fontWeightDefault
    },
    _a[".psds-radio-button__label.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a);
