// @pluralsight/ps-design-system-circularprogress/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '@keyframes psds-circularprogress__keyframes__spin': {
            to: {
                transform: 'rotate(270deg)'
            }
        }
    },
    // --size
    _a[".psds-circularprogress--size-" + vars.sizes.small] = {
        height: '24px',
        width: '24px'
    },
    _a[".psds-circularprogress--size-" + vars.sizes.medium] = {
        height: '48px',
        width: '48px'
    },
    // __svg
    _a[".psds-circularprogress__svg"] = {
        transform: 'rotate(-90deg)'
    },
    // __svg--no-value
    _a[".psds-circularprogress__svg--no-value"] = function (_a) {
        var spin = _a.spin;
        return ({
            animation: (spin || 'psds-circularprogress__keyframes__spin') + " 1s linear infinite"
        });
    },
    // __fg
    _a[".psds-circularprogress__fg"] = {
        fill: 'transparent',
        strokeWidth: vars.style.strokeWidth + "px",
        transition: 'stroke-dashoffset 1s'
    },
    _a[".psds-circularprogress__fg.psds-theme--" + ps_design_system_theme_1.names.light] = {
        stroke: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-circularprogress__fg.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        stroke: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    // __bg
    _a[".psds-circularprogress__bg"] = {
        fill: 'transparent',
        strokeWidth: vars.style.strokeWidth + "px"
    },
    _a[".psds-circularprogress__bg.psds-theme--" + ps_design_system_theme_1.names.light] = {
        stroke: 'rgba(0, 0, 0, 0.15)'
    },
    _a[".psds-circularprogress__bg.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        stroke: 'rgba(255, 255, 255, 0.15)'
    },
    _a);
