// @pluralsight/ps-design-system-layout/dist/cjs/css/aside-layout.js
"use strict";
var _a, _b;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var index_1 = require("../vars/index");
exports.default = (_a = {
        '.psds-aside-layout': {
            display: 'flex'
        }
    },
    _a[".psds-aside-layout--aside-position-" + index_1.asideLayout.asidePositions.first] = {
        flexDirection: 'column'
    },
    _a[".psds-aside-layout--aside-position-" + index_1.asideLayout.asidePositions.last] = {
        flexDirection: 'column-reverse'
    },
    _a[".psds-aside-layout__aside--aside-position-" + index_1.asideLayout.asidePositions.first] = {
        marginBottom: ps_design_system_core_1.layout.spacingLarge
    },
    _a[".psds-aside-layout__aside--aside-position-" + index_1.asideLayout.asidePositions.last] = {
        marginTop: ps_design_system_core_1.layout.spacingLarge
    },
    _a['@media (min-width: 769px)'] = (_b = {},
        _b[".psds-aside-layout--aside-position-" + index_1.asideLayout.asidePositions.first] = {
            flexDirection: 'row'
        },
        _b[".psds-aside-layout--aside-position-" + index_1.asideLayout.asidePositions.last] = {
            flexDirection: 'row-reverse'
        },
        _b['.psds-aside-layout__aside'] = {
            width: 'calc(25%)'
        },
        _b[".psds-aside-layout__aside--aside-position-" + index_1.asideLayout.asidePositions.first] = {
            marginBottom: 0,
            marginRight: ps_design_system_core_1.layout.spacingLarge
        },
        _b[".psds-aside-layout__aside--aside-position-" + index_1.asideLayout.asidePositions.last] = {
            marginTop: 0,
            marginLeft: ps_design_system_core_1.layout.spacingLarge
        },
        _b['.psds-aside-layout__main'] = {
            width: 'calc(75%)'
        },
        _b),
    _a);
