// @pluralsight/ps-design-system-layout/dist/cjs/css/page-heading-layout.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var glamor = glamor_1.default || glamorExports;
var selectors = {
    '.psds-page-heading-layout': {
        padding: ps_design_system_core_1.layout.spacingLarge + " 0"
    },
    '.psds-page-heading-layout__actions': {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'left',
        flexWrap: 'wrap',
        '& > *': {
            marginRight: ps_design_system_core_1.layout.spacingSmall,
            marginBottom: ps_design_system_core_1.layout.spacingMedium
        }
    },
    '@media (min-width: 769px)': {
        '.psds-page-heading-layout__heading': {
            display: 'flex',
            alignItems: 'center'
        },
        '.psds-page-heading-layout__actions': {
            marginLeft: 'auto',
            paddingLeft: ps_design_system_core_1.layout.spacingLarge,
            justifyContent: 'flex-end',
            alignItems: 'center',
            flexWrap: 'nowrap',
            '& > *': {
                marginRight: 0,
                marginBottom: 0
            },
            '& > * + *': {
                marginLeft: ps_design_system_core_1.layout.spacingSmall
            }
        }
    }
};
exports.default = selectors;
