// @pluralsight/ps-design-system-halo/dist/cjs/react/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var glamor_1 = __importStar(require("glamor")), glamorExports = glamor_1;
var focus_within_1 = __importDefault(require("focus-within"));
var react_1 = __importDefault(require("react"));
var index_1 = __importStar(require("../css/index"));
var vars = __importStar(require("../vars/index"));
var glamor = glamor_1.default || glamorExports;
if (ps_design_system_util_1.canUseDOM())
    focus_within_1.default(document);
var styles = {
    halo: function (themeName, props) {
        var base = index_1.BASE_CLASSNAME;
        var theme = base + index_1.themeClasses[themeName];
        var shape = index_1.BASE_CLASSNAME + "--shape-" + props.shape;
        var gapSize = index_1.BASE_CLASSNAME + "--gap-size-" + props.gapSize;
        var gapTheme = gapSize + index_1.themeClasses[themeName];
        var visible = index_1.BASE_CLASSNAME + "--visible";
        var visibleOnFocus = index_1.BASE_CLASSNAME + "--visible-on-focus";
        return glamor.compose(glamor.css(index_1.default[base]), glamor.css(index_1.default[theme]), glamor.css(index_1.default[shape]), glamor.css(index_1.default[gapSize]), glamor.css(index_1.default[gapTheme]), props.inline && glamor.css(index_1.default[index_1.BASE_CLASSNAME + "--inline"]), props.error && glamor.css(index_1.default[index_1.BASE_CLASSNAME + "--error"]), props.visible && glamor.css(index_1.default[visible]), props.visibleOnFocus && glamor.css(index_1.default[visibleOnFocus]));
    }
};
var Halo = react_1.default.forwardRef(function (props, ref) {
    var themeName = ps_design_system_theme_1.useTheme();
    var _a = props.error, error = _a === void 0 ? false : _a, _b = props.gapSize, gapSize = _b === void 0 ? vars.gapSizes.default : _b, _c = props.inline, inline = _c === void 0 ? false : _c, _d = props.shape, shape = _d === void 0 ? vars.shapes.default : _d, _e = props.visible, visible = _e === void 0 ? false : _e, _f = props.visibleOnFocus, visibleOnFocus = _f === void 0 ? true : _f, rest = __rest(props, ["error", "gapSize", "inline", "shape", "visible", "visibleOnFocus"]);
    var style = styles.halo(themeName, {
        error: error,
        gapSize: gapSize,
        inline: inline,
        shape: shape,
        visible: visible,
        visibleOnFocus: visibleOnFocus
    });
    return react_1.default.createElement("div", __assign({ ref: ref }, style, rest));
});
Halo.gapSizes = vars.gapSizes;
Halo.shapes = vars.shapes;
exports.default = Halo;
