// @codecademy/gamut-labs/dist/LayoutMenu/LayoutMenu.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { Box, Logo, StrokeButton } from '@codecademy/gamut';
import React, { useRef } from 'react';
import { Flyout } from '../Flyout';
import { AccordionMenu } from './AccordionMenu';
export var LayoutMenu = function LayoutMenu(_ref) {
  var sections = _ref.sections,
      onSectionToggle = _ref.onSectionToggle,
      selectedItem = _ref.selectedItem,
      mobileButtonText = _ref.mobileButtonText,
      _ref$breakpoint = _ref.breakpoint,
      breakpoint = _ref$breakpoint === void 0 ? 'lg' : _ref$breakpoint,
      children = _ref.children;
  var closeFlyoutRef = useRef(function () {});

  var closeFlyout = function closeFlyout() {
    return closeFlyoutRef.current();
  };

  var accordionMenuSections = sections.map(function (section) {
    return /*#__PURE__*/React.createElement(AccordionMenu, {
      key: section.slug,
      section: section,
      onSectionToggle: onSectionToggle,
      onItemClick: closeFlyout,
      selectedItem: selectedItem
    });
  });

  var renderButton = function renderButton(onClick) {
    return /*#__PURE__*/React.createElement(StrokeButton, {
      variant: "secondary",
      width: 1,
      onClick: onClick
    }, mobileButtonText);
  };

  return /*#__PURE__*/React.createElement("nav", null, /*#__PURE__*/React.createElement(Box, {
    display: _defineProperty({
      _: 'block'
    }, breakpoint, 'none')
  }, /*#__PURE__*/React.createElement(Flyout, {
    renderButton: renderButton,
    closeFlyoutRef: closeFlyoutRef,
    overflowY: "auto",
    overflowX: "hidden"
  }, /*#__PURE__*/React.createElement(Box, {
    bg: "white",
    minHeight: 1,
    p: 16
  }, /*#__PURE__*/React.createElement(Logo, {
    mb: 32
  }), accordionMenuSections, children))), /*#__PURE__*/React.createElement(Box, {
    display: _defineProperty({
      _: 'none'
    }, breakpoint, 'block')
  }, accordionMenuSections, children));
};
