// @pluralsight/ps-design-system-button/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.loading = void 0;
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var vars = __importStar(require("../vars/index"));
exports.loading = {
    trackColorOnLight: ps_design_system_core_1.colorsBorder.lowOnLight,
    trackColorOnDark: ps_design_system_core_1.colorsBorder.lowOnDark,
    borderTopColorOnLight: ps_design_system_core_1.colorsTextIcon.highOnLight,
    borderTopColorOnDark: ps_design_system_core_1.colorsTextIcon.highOnDark,
    borderTopColorOnStroke: ps_design_system_core_1.colorsPrimaryAction.background
};
exports.default = (_a = {
        '@keyframes psds-button__keyframes__spin': {
            '100%': {
                transform: 'rotate(360deg)'
            }
        },
        '.psds-button': {
            position: 'relative',
            display: 'inline-flex',
            justifyContent: 'center',
            alignItems: 'stretch',
            border: 0,
            borderRadius: '3px',
            fontSize: ps_design_system_core_1.type.fontSize300,
            fontWeight: ps_design_system_core_1.type.fontWeight500,
            textAlign: 'center',
            whiteSpace: 'nowrap',
            textDecoration: 'none',
            transition: "background " + ps_design_system_core_1.motion.speedFast + ", transform " + ps_design_system_core_1.motion.speedXFast,
            verticalAlign: 'middle',
            '&:not([disabled]):hover': {
                cursor: 'pointer'
            },
            '&:focus:not(:focus-visible)': {
                outline: 'none',
                boxShadow: 'none'
            },
            '&:not([disabled]):active': {
                transform: 'scale(0.98)'
            }
        }
    },
    _a[".psds-button.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        '&:focus': {
            outline: 'none',
            boxShadow: "0 0 0 2px " + ps_design_system_core_1.colorsBlue[4]
        },
        '&:focus-visible': {
            outline: 'none',
            boxShadow: "0 0 0 2px " + ps_design_system_core_1.colorsBlue[4]
        }
    },
    _a[".psds-button.psds-theme--" + ps_design_system_theme_1.names.light] = {
        '&:focus': {
            outline: 'none',
            boxShadow: "0 0 0 2px " + ps_design_system_core_1.colorsBlue[8]
        },
        '&:focus-visible': {
            outline: 'none',
            boxShadow: "0 0 0 2px " + ps_design_system_core_1.colorsBlue[8]
        }
    },
    _a[".psds-button--layout-" + vars.layouts.fullWidth] = {
        display: 'flex',
        width: '100%'
    },
    _a[".psds-button--size-" + vars.sizes.xSmall] = {
        fontSize: ps_design_system_core_1.type.fontSize100,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLooser,
        padding: "4px 8px",
        height: '24px'
    },
    _a[".psds-button--size-" + vars.sizes.small] = {
        fontSize: ps_design_system_core_1.type.fontSize200,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLooser,
        padding: "6px 12px",
        height: '32px'
    },
    _a[".psds-button--size-" + vars.sizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize300,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLoose,
        padding: "10px 16px",
        height: '40px'
    },
    _a[".psds-button--size-" + vars.sizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize400,
        letterSpacing: ps_design_system_core_1.type.letterSpacingNone,
        padding: "12px 20px",
        height: '48px'
    },
    _a[".psds-button--appearance-" + vars.appearances.primary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        background: ps_design_system_core_1.colorsPrimaryAction.background,
        '&:not([disabled]):hover': {
            background: ps_design_system_core_1.colorsBlue[8],
            cursor: 'pointer'
        },
        '&:not([disabled]):focus': {
            background: ps_design_system_core_1.colorsBlue[8],
            cursor: 'pointer'
        },
        '&:not([disabled]):active': {
            background: ps_design_system_core_1.colorsBlue[9]
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.primary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        background: ps_design_system_core_1.colorsPrimaryAction.background,
        '&:not([disabled]):hover': {
            background: ps_design_system_core_1.colorsBlue[8],
            cursor: 'pointer'
        },
        '&:not([disabled]):focus': {
            background: ps_design_system_core_1.colorsBlue[8],
            cursor: 'pointer',
            boxShadow: "0 0 0 2px " + ps_design_system_core_1.colorsBlue[5]
        },
        '&:not([disabled]):focus-visible': {
            boxShadow: "0 0 0 2px " + ps_design_system_core_1.colorsBlue[5]
        },
        '&:not([disabled]):active': {
            background: ps_design_system_core_1.colorsBlue[9]
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.secondary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        background: ps_design_system_core_1.colorsBackgroundUtility[25],
        '&:not([disabled]):hover': {
            background: ps_design_system_core_1.colorsBackgroundUtility[30]
        },
        '&:not([disabled]):focus': {
            background: ps_design_system_core_1.colorsBackgroundUtility[30]
        },
        '&:not([disabled]):active': {
            background: ps_design_system_core_1.colorsBackgroundUtility[20]
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.secondary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight,
        background: ps_design_system_core_1.colorsBackgroundUtility[20],
        '&:not([disabled]):hover': {
            background: ps_design_system_core_1.colorsBackgroundUtility[25]
        },
        '&:not([disabled]):focus': {
            background: ps_design_system_core_1.colorsBackgroundUtility[25]
        },
        '&:not([disabled]):active': {
            background: ps_design_system_core_1.colorsBackgroundUtility[30]
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.stroke + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        border: "2px solid " + ps_design_system_core_1.colorsPrimaryAction.background,
        color: ps_design_system_core_1.colorsPrimaryAction.textDarkTheme,
        background: 'none',
        '&:not([disabled]):hover': {
            border: "2px solid " + ps_design_system_core_1.colorsBlue[5],
            background: ps_design_system_util_1.transparentize(0.85, ps_design_system_core_1.colorsPrimaryAction.textDarkTheme),
            color: ps_design_system_core_1.colorsBlue[4]
        },
        '&:not([disabled]):focus': {
            border: "2px solid " + ps_design_system_core_1.colorsBlue[5],
            background: ps_design_system_util_1.transparentize(0.85, ps_design_system_core_1.colorsPrimaryAction.textDarkTheme),
            color: ps_design_system_core_1.colorsBlue[4]
        },
        '&:not([disabled]):active': {
            border: "2px solid " + ps_design_system_core_1.colorsBlue[7],
            background: ps_design_system_util_1.transparentize(0.8, ps_design_system_core_1.colorsPrimaryAction.textDarkTheme)
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.stroke + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        border: "2px solid " + ps_design_system_core_1.colorsPrimaryAction.background,
        color: ps_design_system_core_1.colorsPrimaryAction.textLightTheme,
        background: 'none',
        '&:not([disabled]):hover': {
            border: "2px solid " + ps_design_system_core_1.colorsBlue[7],
            background: ps_design_system_util_1.transparentize(0.95, ps_design_system_core_1.colorsPrimaryAction.textLightTheme),
            color: ps_design_system_core_1.colorsBlue[8]
        },
        '&:not([disabled]):focus': {
            border: "2px solid " + ps_design_system_core_1.colorsBlue[7],
            background: ps_design_system_util_1.transparentize(0.95, ps_design_system_core_1.colorsPrimaryAction.textLightTheme),
            color: ps_design_system_core_1.colorsBlue[8]
        },
        '&:not([disabled]):active': {
            border: "2px solid " + ps_design_system_core_1.colorsBlue[8],
            background: ps_design_system_util_1.transparentize(0.9, ps_design_system_core_1.colorsPrimaryAction.textLightTheme)
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.flat] = {
        border: 'none',
        background: 'none'
    },
    _a[".psds-button--appearance-" + vars.appearances.flat + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark,
        '&:not([disabled]):hover': {
            background: ps_design_system_core_1.colorsBackgroundUtility[25],
            color: ps_design_system_core_1.colorsTextIcon.highOnDark
        },
        '&:not([disabled]):focus': {
            background: ps_design_system_core_1.colorsBackgroundUtility[25],
            color: ps_design_system_core_1.colorsTextIcon.highOnDark
        },
        '&:not([disabled]):active': {
            background: ps_design_system_core_1.colorsBackgroundUtility[20],
            color: ps_design_system_core_1.colorsTextIcon.highOnDark
        }
    },
    _a[".psds-button--appearance-" + vars.appearances.flat + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        '&:not([disabled]):hover': {
            background: ps_design_system_core_1.colorsBackgroundUtility[20],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        },
        '&:not([disabled]):focus': {
            background: ps_design_system_core_1.colorsBackgroundUtility[20],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        },
        '&:not([disabled]):active': {
            background: ps_design_system_core_1.colorsBackgroundUtility[15],
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        }
    },
    _a[".psds-button--disabled"] = {
        opacity: '50%',
        cursor: 'not-allowed',
        '&:not([disabled]):hover': {
            color: 'inherit',
            background: 'inherit',
            border: 'inherit'
        }
    },
    _a[".psds-button--disabled.psds-theme--" + ps_design_system_theme_1.names.light] = {
        opacity: 0.5
    },
    _a[".psds-button--disabled.psds-button--appearance-" + vars.appearances.primary] = {
        opacity: 0.5
    },
    _a[".psds-button--disabled.psds-button--appearance-" + vars.appearances.secondary] = {
        opacity: 0.5
    },
    _a[".psds-button--disabled.psds-button--appearance-" + vars.appearances.stroke] = {
        opacity: 0.5
    },
    _a[".psds-button--disabled.psds-button--appearance-" + vars.appearances.flat] = {
        opacity: 0.5
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.right] = {
        flexDirection: 'row-reverse'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.right + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.xSmall] = {
        paddingRight: '4px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.left + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.xSmall] = {
        paddingLeft: '4px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.right + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.small] = {
        paddingRight: '8px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.left + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.small] = {
        paddingLeft: '8px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.right + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.medium] = {
        paddingRight: '12px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.left + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.medium] = {
        paddingLeft: '12px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.right + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.large] = {
        paddingRight: '16px'
    },
    _a[".psds-button--iconAlign-" + vars.iconAligns.left + ".psds-button--not-iconOnly.psds-button--size-" + vars.sizes.large] = {
        paddingLeft: '16px'
    },
    _a[".psds-button--iconOnly"] = {
        padding: 0
    },
    _a[".psds-button--iconOnly.psds-button--size-" + vars.sizes.xSmall] = {
        width: '24px'
    },
    _a[".psds-button--iconOnly.psds-button--size-" + vars.sizes.small] = {
        width: '32px'
    },
    _a[".psds-button--iconOnly.psds-button--size-" + vars.sizes.medium] = {
        width: '40px'
    },
    _a[".psds-button--iconOnly.psds-button--size-" + vars.sizes.large] = {
        width: '48px'
    },
    _a[".psds-button__icon"] = {
        display: 'flex',
        alignItems: 'center',
        minHeight: '100%'
    },
    _a[".psds-button__icon--iconAlign-" + vars.iconAligns.right] = {
        marginLeft: '8px',
        marginRight: 0
    },
    _a[".psds-button__icon--iconAlign-" + vars.iconAligns.left] = {
        marginLeft: 0,
        marginRight: '8px'
    },
    _a[".psds-button__icon--iconAlign-" + vars.iconAligns.right + ".psds-button--size-" + vars.sizes.xSmall] = {
        marginLeft: '4px',
        marginRight: 0
    },
    _a[".psds-button__icon--iconAlign-" + vars.iconAligns.left + ".psds-button--size-" + vars.sizes.xSmall] = {
        marginLeft: 0,
        marginRight: '4px'
    },
    _a[".psds-button__icon--iconAlign-" + vars.iconAligns.right + ".psds-button--size-" + vars.sizes.small] = {
        marginLeft: '4px',
        marginRight: 0
    },
    _a[".psds-button__icon--iconAlign-" + vars.iconAligns.left + ".psds-button--size-" + vars.sizes.small] = {
        marginLeft: 0,
        marginRight: '4px'
    },
    _a[".psds-button__icon--iconOnly"] = {
        justifyContent: 'center',
        width: '100%',
        margin: 0
    },
    _a['.psds-button__icon--loadingLabelOnly'] = {
        position: 'absolute',
        top: 0,
        justifyContent: 'center',
        width: '100%',
        margin: 0
    },
    _a[".psds-button__loading"] = function (_a) {
        var spin = _a.spin;
        return ({
            display: 'inline-block',
            height: 'calc(100% - 4px)',
            width: 'calc(100% - 4px)',
            margin: '2px',
            borderStyle: 'solid',
            borderWidth: '2px',
            borderRadius: '100%',
            animation: (spin || 'psds-button__keyframes__spin') + " 1s linear infinite"
        });
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.primary] = {
        borderColor: exports.loading.trackColorOnDark,
        borderTopColor: exports.loading.borderTopColorOnDark
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.secondary + ".psds-button__loading--theme-" + ps_design_system_theme_1.names.light] = {
        borderColor: exports.loading.trackColorOnLight,
        borderTopColor: exports.loading.borderTopColorOnLight
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.secondary + ".psds-button__loading--theme-" + ps_design_system_theme_1.defaultName] = {
        borderColor: exports.loading.trackColorOnDark,
        borderTopColor: exports.loading.borderTopColorOnDark
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.stroke + ".psds-button__loading--theme-" + ps_design_system_theme_1.names.light] = {
        borderColor: exports.loading.trackColorOnLight,
        borderTopColor: exports.loading.borderTopColorOnStroke
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.stroke + ".psds-button__loading--theme-" + ps_design_system_theme_1.defaultName] = {
        borderColor: exports.loading.trackColorOnDark,
        borderTopColor: exports.loading.borderTopColorOnStroke
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.flat + ".psds-button__loading--theme-" + ps_design_system_theme_1.names.light] = {
        borderColor: exports.loading.trackColorOnLight,
        borderTopColor: exports.loading.borderTopColorOnLight
    },
    _a[".psds-button__loading--appearance-" + vars.appearances.flat + ".psds-button__loading--theme-" + ps_design_system_theme_1.defaultName] = {
        borderColor: exports.loading.trackColorOnDark,
        borderTopColor: exports.loading.borderTopColorOnDark
    },
    _a[".psds-button__text"] = {
        alignItems: 'center',
        display: 'inline-flex',
        pointerEvents: 'none'
    },
    _a[".psds-button__text--invisible"] = {
        visibility: 'hidden'
    },
    _a);
