// @codecademy/gamut/dist/Button/shared.js
import _styled from "@emotion/styled/base";

var _system$css;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { fontSmoothPixel, modeColorProps, styledOptions, system, transitionConcat } from '@codecademy/gamut-styles';
import { variance } from '@codecademy/variance';
import { ButtonBase, ButtonSelectors } from '../ButtonBase/ButtonBase';
export var config = styledOptions(['size']);
export var buttonProps = variance.compose(system.layout, system.positioning, system.space, system.border);
export var templateVariants = function templateVariants(variants, template) {
  var variantConfig = {};
  variants.forEach(function (key) {
    variantConfig[key] = template(key);
  });
  return system.variant({
    defaultVariant: variants[0],
    variants: variantConfig
  });
};
export var buttonVariants = ['primary', 'secondary', 'danger'];
export var buttonStyles = system.css((_system$css = {
  position: 'relative',
  fontWeight: 'title',
  whiteSpace: 'nowrap',
  display: 'inline-flex',
  justifyContent: 'center',
  alignItems: 'center',
  border: 2,
  borderRadius: '4px',
  borderColor: 'transparent',
  transition: transitionConcat(['border-color', 'color', 'background-color', 'box-shadow'], 'fast', 'ease-in')
}, _defineProperty(_system$css, ButtonSelectors.DISABLED, {
  cursor: 'not-allowed',
  userSelect: 'none'
}), _defineProperty(_system$css, ButtonSelectors.OUTLINE, {
  content: '""',
  transition: transitionConcat(['opacity'], 'fast'),
  position: 'absolute',
  borderRadius: '6px',
  border: 2,
  inset: -5,
  opacity: 0,
  zIndex: 0
}), _defineProperty(_system$css, ButtonSelectors.OUTLINE_FOCUS_VISIBLE, {
  opacity: 1
}), _system$css));
export var createButtonComponent = function createButtonComponent() {
  for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  return /*#__PURE__*/_styled(ButtonBase, {
    target: "e1w6mdco0",
    label: "createButtonComponent"
  }).apply(void 0, [fontSmoothPixel, modeColorProps, buttonStyles].concat(args, [buttonProps]));
};
