// @pluralsight/ps-design-system-steps/dist/esm/react/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { isString } from '@pluralsight/ps-design-system-util';
import { useTheme } from '@pluralsight/ps-design-system-theme';
import glamorDefault, * as glamorExports from 'glamor';
import React from 'react';
import stylesheet from '../css/index';
import * as vars from '../vars/index';
import Context, { initialContext } from './context';
var glamor = glamorDefault || glamorExports;
var styles = {
    steps: function (themeName, opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-steps']), glamor.css(stylesheet[".psds-steps.psds-theme--" + themeName]), glamor.css(stylesheet[".psds-steps--" + opts.orientation]));
    },
    step: function (themeName, opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-steps__step']), glamor.css(stylesheet[".psds-steps__step.psds-theme--" + themeName]), glamor.css(stylesheet[".psds-steps__step--" + opts.orientation]), glamor.css(stylesheet[".psds-steps__step--" + opts.size]), glamor.css(stylesheet[".psds-steps__step--" + opts.status]), opts.interactive &&
            glamor.css(stylesheet['.psds-steps__step--interactive']));
    },
    title: function (themeName, opts) {
        var theme = ".psds-theme--" + themeName;
        var status = ".psds-steps__title--" + opts.status;
        return glamor.compose(glamor.css(stylesheet['.psds-steps__title']), glamor.css(stylesheet['.psds-steps__title' + theme]), glamor.css(stylesheet[".psds-steps__title--" + opts.size]), glamor.css(stylesheet[status]), glamor.css(stylesheet[status + theme]));
    },
    description: function (opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-steps__description']), glamor.css(stylesheet[".psds-steps__description--" + opts.size]), glamor.css(stylesheet[".psds-steps__description--" + opts.status]));
    },
    markerContainer: function (opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-steps__marker-container']), glamor.css(stylesheet[".psds-steps__marker-container--" + opts.size]));
    },
    marker: function (themeName, opts) {
        var theme = ".psds-theme--" + themeName;
        var status = ".psds-steps__marker--" + opts.status;
        return glamor.compose(glamor.css(stylesheet['.psds-steps__marker']), glamor.css(stylesheet['.psds-steps__marker' + theme]), glamor.css(stylesheet[".psds-steps__marker--" + opts.size]), glamor.css(stylesheet[status]), glamor.css(stylesheet[status + theme]), !opts.counter &&
            glamor.css(stylesheet['.psds-steps__marker--hide-counter']));
    },
    markerCircle: function () { return glamor.css(stylesheet['.psds-steps__marker__circle']); },
    markerCheck: function (opts) {
        return glamor.compose(glamor.css(stylesheet['.psds-steps__marker__check']), glamor.css(stylesheet[".psds-steps__marker__check--" + opts.status]));
    }
};
var Steps = React.forwardRef(function (props, ref) {
    var _a = props.counter, counter = _a === void 0 ? initialContext.counter : _a, _b = props.orientation, orientation = _b === void 0 ? initialContext.orientation : _b, _c = props.size, size = _c === void 0 ? initialContext.size : _c, rest = __rest(props, ["counter", "orientation", "size"]);
    var themeName = useTheme();
    var contextValue = { counter: counter, orientation: orientation, size: size };
    return (React.createElement(Context.Provider, { value: contextValue },
        React.createElement("div", __assign({ ref: ref }, styles.steps(themeName, { orientation: orientation }), rest))));
});
Steps.displayName = 'Steps';
var defaultMarkerContainer = function (props) { return React.createElement("div", __assign({}, props)); };
var Step = React.forwardRef(function (props, ref) {
    var children = props.children, description = props.description, _a = props.marker, StepMarker = _a === void 0 ? Marker : _a, _b = props.renderMarkerContainer, renderMarkerContainer = _b === void 0 ? defaultMarkerContainer : _b, status = props.status, rest = __rest(props, ["children", "description", "marker", "renderMarkerContainer", "status"]);
    var MarkerContainer = React.useMemo(function () { return renderMarkerContainer; }, [
        renderMarkerContainer
    ]);
    var interactive = 'onClick' in props;
    var themeName = useTheme();
    var _c = React.useContext(Context), orientation = _c.orientation, size = _c.size;
    return (React.createElement("div", __assign({ ref: ref }, styles.step(themeName, { interactive: interactive, orientation: orientation, size: size, status: status }), rest),
        React.createElement(MarkerContainer, __assign({}, styles.markerContainer({ size: size })),
            React.createElement(StepMarker, { status: status })),
        React.createElement("div", null,
            React.createElement("div", __assign({}, styles.title(themeName, { size: size, status: status })), isString(children) ? React.createElement("p", null, children) : children),
            description && (React.createElement("div", __assign({}, styles.description({ size: size, status: status })), isString(description) ? React.createElement("p", null, description) : description)))));
});
Step.displayName = 'Steps.Step';
var Marker = React.forwardRef(function (props, ref) {
    var status = props.status, rest = __rest(props, ["status"]);
    var themeName = useTheme();
    var _a = React.useContext(Context), counter = _a.counter, size = _a.size;
    var _b = React.useMemo(function () {
        var large = size === 'large';
        var _diameter = large ? 36 : 24;
        var _stroke = large ? 2 : 2;
        var _radius = _diameter / 2 - _stroke / 2;
        return [_diameter, _stroke, _radius];
    }, [size]), diameter = _b[0], stroke = _b[1], radius = _b[2];
    return (React.createElement("div", __assign({ ref: ref }, styles.marker(themeName, { counter: counter, size: size, status: status }), glamor.css({ width: diameter, height: diameter }), rest),
        React.createElement("svg", { viewBox: "0 0 " + diameter + " " + diameter },
            React.createElement("circle", __assign({ cx: diameter / 2, cy: diameter / 2, r: radius, strokeWidth: stroke }, styles.markerCircle())),
            React.createElement("polyline", __assign({ strokeWidth: stroke, points: "\n            " + diameter / 3.8 + " " + diameter / 2 + ",\n            " + diameter / 2.3 + " " + diameter / 1.5 + ",\n            " + diameter / 1.4 + " " + diameter / 3 + "\n          " }, styles.markerCheck({ status: status }))))));
});
Marker.displayName = 'Steps.Marker';
Steps.Marker = Marker;
Steps.Step = Step;
Steps.orientations = vars.orientations;
Steps.sizes = vars.sizes;
Steps.statuses = vars.statuses;
export default Steps;
