// @codecademy/gamut-labs/dist/GlobalHeader/GlobalHeaderVariants.js
import { communityDropdown, courseCatalog, forBusiness, freeProfile, login, logo, myHome, notifications, pricingDropdown, proLogo, proProfile, resourcesDropdown, search, signUp, tryProForFree, unpausePro, upgradeToPro } from './GlobalHeaderItems';

var anonHeaderItems = function anonHeaderItems(renderLogin, renderSignUp, renderSearch) {
  var leftItems = [logo, courseCatalog, resourcesDropdown, communityDropdown, pricingDropdown, forBusiness];
  var rightItems = [];
  renderSearch && rightItems.push(search(renderSearch));
  renderLogin && rightItems.push(login);
  renderSignUp && rightItems.push(signUp);
  return {
    left: leftItems,
    right: rightItems
  };
};

var anonMobileHeaderItems = function anonMobileHeaderItems(renderLogin, renderSignUp) {
  var leftItems = [logo];
  var rightItems = [];
  renderLogin && rightItems.push(login);
  renderSignUp && rightItems.push(signUp);
  var mainMenuItems = [courseCatalog, resourcesDropdown, communityDropdown, pricingDropdown, forBusiness, signUp, login];
  return {
    left: leftItems,
    right: rightItems,
    mainMenu: mainMenuItems
  };
};

export var anonDefaultHeaderItems = function anonDefaultHeaderItems(renderSearch) {
  return anonHeaderItems(true, true, renderSearch);
};
export var anonDefaultMobileHeaderItems = function anonDefaultMobileHeaderItems() {
  return anonMobileHeaderItems(true, true);
};
export var anonLandingHeaderItems = function anonLandingHeaderItems() {
  return anonHeaderItems(true, false);
};
export var anonLandingMobileHeaderItems = function anonLandingMobileHeaderItems() {
  return anonMobileHeaderItems(true, false);
};
export var anonLoginHeaderItems = function anonLoginHeaderItems(renderSearch) {
  return anonHeaderItems(false, true, renderSearch);
};
export var anonLoginMobileHeaderItems = function anonLoginMobileHeaderItems() {
  return anonMobileHeaderItems(false, true);
};
export var anonSignupHeaderItems = function anonSignupHeaderItems(renderSearch) {
  return anonHeaderItems(true, false, renderSearch);
};
export var anonSignupMobileHeaderItems = function anonSignupMobileHeaderItems() {
  return anonMobileHeaderItems(true, false);
};
export var freeHeaderItems = function freeHeaderItems(user, renderSearch, renderNotifications) {
  var leftItems = [logo, myHome, courseCatalog, resourcesDropdown, communityDropdown, pricingDropdown, forBusiness];
  var rightItems = [];
  renderSearch && rightItems.push(search(renderSearch));
  renderNotifications && rightItems.push(notifications(renderNotifications));
  rightItems.push(freeProfile(user));
  rightItems.push(user.showProUpgrade ? upgradeToPro(user.proCheckoutUrl) : tryProForFree(user.proCheckoutUrl));
  return {
    left: leftItems,
    right: rightItems
  };
};
export var freeMobileHeaderItems = function freeMobileHeaderItems(user, renderNotifications) {
  var leftItems = [logo];
  var rightItems = [];
  renderNotifications && rightItems.push(notifications(renderNotifications));
  var mainMenuItems = [myHome, courseCatalog, resourcesDropdown, communityDropdown, pricingDropdown, forBusiness, freeProfile(user, true)];
  mainMenuItems.push(user.showProUpgrade ? upgradeToPro(user.proCheckoutUrl) : tryProForFree(user.proCheckoutUrl));
  return {
    left: leftItems,
    right: rightItems,
    mainMenu: mainMenuItems
  };
};
export var proHeaderItems = function proHeaderItems(user, renderSearch, renderNotifications) {
  var leftItems = [proLogo, myHome, courseCatalog, resourcesDropdown, communityDropdown];
  var rightItems = [];
  renderSearch && rightItems.push(search(renderSearch));
  renderNotifications && rightItems.push(notifications(renderNotifications));
  rightItems.push(proProfile(user));
  user.isPaused && rightItems.push(unpausePro);
  return {
    left: leftItems,
    right: rightItems
  };
};
export var proMobileHeaderItems = function proMobileHeaderItems(user, renderNotifications) {
  var leftItems = [proLogo];
  var rightItems = [];
  renderNotifications && rightItems.push(notifications(renderNotifications));
  var mainMenuItems = [myHome, courseCatalog, resourcesDropdown, communityDropdown, proProfile(user, true)];
  user.isPaused && mainMenuItems.push(unpausePro);
  return {
    left: leftItems,
    right: rightItems,
    mainMenu: mainMenuItems
  };
};
export var loadingHeaderItems = {
  left: [logo],
  right: []
};
export var loadingMobileHeaderItems = {
  left: [logo],
  right: [],
  mainMenu: []
};
