// @codecademy/gamut/dist/Overlay/index.js
import React, { useCallback } from 'react';
import { BodyPortal } from '../BodyPortal';
import { FlexBox } from '../Box';
import { FocusTrap } from '../FocusTrap';
export var Overlay = function Overlay(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$inline = _ref.inline,
      inline = _ref$inline === void 0 ? false : _ref$inline,
      _ref$clickOutsideClos = _ref.clickOutsideCloses,
      clickOutsideCloses = _ref$clickOutsideClos === void 0 ? true : _ref$clickOutsideClos,
      _ref$escapeCloses = _ref.escapeCloses,
      escapeCloses = _ref$escapeCloses === void 0 ? true : _ref$escapeCloses,
      onRequestClose = _ref.onRequestClose,
      isOpen = _ref.isOpen;
  var handleOutsideClick = useCallback(function () {
    clickOutsideCloses && onRequestClose();
  }, [clickOutsideCloses, onRequestClose]);
  var handleEscapeKey = useCallback(function () {
    escapeCloses && onRequestClose();
  }, [escapeCloses, onRequestClose]);
  if (!isOpen) return null;
  var content = /*#__PURE__*/React.createElement(FlexBox, {
    "data-testid": "overlay-content-container",
    position: inline ? 'absolute' : 'fixed',
    justifyContent: "center",
    alignItems: "center",
    inset: 0,
    className: className
  }, /*#__PURE__*/React.createElement(FocusTrap, {
    active: !inline,
    onClickOutside: handleOutsideClick,
    onEscapeKey: handleEscapeKey
  }, children));
  if (inline) return content;
  return /*#__PURE__*/React.createElement(BodyPortal, null, content);
};
