// @pluralsight/ps-design-system-text/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '.psds-text__code': {
            padding: "0 " + ps_design_system_core_1.layout.spacingXXSmall,
            borderRadius: '2px',
            lineHeight: ps_design_system_core_1.type.lineHeightStandard,
            whiteSpace: 'pre-wrap',
            fontFamily: ps_design_system_core_1.type.fontFamilyCode,
            fontWeight: ps_design_system_core_1.type.fontWeight500,
            fontSize: ps_design_system_core_1.type.fontSize200,
            wordBreak: 'break-word'
        }
    },
    _a[".psds-text__code.psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsBlue[8],
        background: ps_design_system_core_1.colorsBackgroundUtility[25],
        border: "1px solid " + ps_design_system_core_1.colorsBorder.lowOnLight
    },
    _a[".psds-text__code.psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsBlue[3],
        background: ps_design_system_core_1.colorsBackgroundUtility[25],
        border: "1px solid " + ps_design_system_core_1.colorsBorder.lowOnDark
    },
    _a['.psds-text__heading'] = {
        margin: ps_design_system_core_1.layout.spacingMedium + " 0",
        fontWeight: ps_design_system_core_1.type.fontWeightStrong
    },
    _a[".psds-text__heading--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-text__heading--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-text__heading--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    _a[".psds-text__heading--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.xXXSmall] = {
        fontSize: ps_design_system_core_1.type.fontSize200,
        lineHeight: '16px',
        textTransform: 'uppercase',
        letterSpacing: ps_design_system_core_1.type.letterSpacingAllCaps
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.xXSmall] = {
        fontSize: ps_design_system_core_1.type.fontSize300,
        lineHeight: '20px',
        textTransform: 'uppercase',
        letterSpacing: ps_design_system_core_1.type.letterSpacingAllCaps
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.xSmall] = {
        fontSize: ps_design_system_core_1.type.fontSize500,
        lineHeight: '28px'
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.small] = {
        fontSize: ps_design_system_core_1.type.fontSize600,
        lineHeight: '32px'
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize700,
        lineHeight: '36px'
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize800,
        lineHeight: '40px'
    },
    _a[".psds-text__heading--size-" + vars.headingSizes.xLarge] = {
        fontSize: ps_design_system_core_1.type.fontSize900,
        lineHeight: '52px'
    },
    _a[".psds-text__label"] = {
        fontWeight: ps_design_system_core_1.type.fontWeightDefault
    },
    _a[".psds-text__label--strong"] = {
        fontWeight: ps_design_system_core_1.type.fontWeightStrong
    },
    _a[".psds-text__label--caps"] = {
        textTransform: 'uppercase',
        letterSpacing: ps_design_system_core_1.type.letterSpacingAllCaps
    },
    _a[".psds-text__label--mono"] = {
        fontFamily: ps_design_system_core_1.type.fontFamilyCode,
        fontFeatureSettings: '"ss05" on'
    },
    _a[".psds-text__label--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-text__label--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-text__label--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    _a[".psds-text__label--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a[".psds-text__label--size-" + vars.labelSizes.xSmall] = {
        fontSize: ps_design_system_core_1.type.fontSize100,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLooser,
        lineHeight: '16px'
    },
    _a[".psds-text__label--size-" + vars.labelSizes.small] = {
        fontSize: ps_design_system_core_1.type.fontSize200,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLoose,
        lineHeight: '20px'
    },
    _a[".psds-text__label--size-" + vars.labelSizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize300,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLoose,
        lineHeight: '20px'
    },
    _a[".psds-text__label--size-" + vars.labelSizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize400,
        letterSpacing: ps_design_system_core_1.type.letterSpacingNone,
        lineHeight: '24px'
    },
    _a[".psds-text__list"] = {
        listStyle: 'none',
        marginLeft: 0,
        lineHeight: ps_design_system_core_1.type.lineHeightExtra,
        fontWeight: ps_design_system_core_1.type.fontWeightDefault
    },
    _a[".psds-text__list--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-text__list--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-text__list--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    _a[".psds-text__list--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a[".psds-text__list--size-" + vars.listSizes.small] = {
        fontSize: ps_design_system_core_1.type.fontSize200,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLooser,
        lineHeight: '24px'
    },
    _a[".psds-text__list--size-" + vars.listSizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize300,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLoose,
        lineHeight: '24px'
    },
    _a[".psds-text__list--size-" + vars.listSizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize400,
        letterSpacing: ps_design_system_core_1.type.letterSpacingNone,
        lineHeight: '28px'
    },
    _a[".psds-text__list--type-" + vars.listTypes.bulleted] = {
        listStyle: 'initial',
        listStyleType: 'disc'
    },
    _a[".psds-text__list--type-" + vars.listTypes.numbered] = {
        listStyle: 'initial',
        listStyleType: 'decimal'
    },
    _a['.psds-text__p'] = {
        margin: ps_design_system_core_1.layout.spacingMedium + " 0",
        fontWeight: ps_design_system_core_1.type.fontWeightDefault
    },
    _a[".psds-text__p--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight
    },
    _a[".psds-text__p--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.names.light] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight
    },
    _a[".psds-text__p--color-" + vars.textColors.primary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark
    },
    _a[".psds-text__p--color-" + vars.textColors.secondary + ".psds-theme--" + ps_design_system_theme_1.defaultName] = {
        color: ps_design_system_core_1.colorsTextIcon.lowOnDark
    },
    _a[".psds-text__p--size-" + vars.pSizes.small] = {
        fontSize: ps_design_system_core_1.type.fontSize200,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLooser,
        lineHeight: '24px'
    },
    _a[".psds-text__p--size-" + vars.pSizes.medium] = {
        fontSize: ps_design_system_core_1.type.fontSize300,
        letterSpacing: ps_design_system_core_1.type.letterSpacingLoose,
        lineHeight: '24px'
    },
    _a[".psds-text__p--size-" + vars.pSizes.large] = {
        fontSize: ps_design_system_core_1.type.fontSize400,
        letterSpacing: ps_design_system_core_1.type.letterSpacingNone,
        lineHeight: '28px'
    },
    _a);
