// @pluralsight/ps-design-system-verticaltabs/dist/esm/react/group.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { useTheme } from '@pluralsight/ps-design-system-theme';
import Collapsible from '@pluralsight/ps-design-system-collapsible';
import { CaretDownIcon } from '@pluralsight/ps-design-system-icon';
import { omit } from '@pluralsight/ps-design-system-util';
import glamorDefault, * as glamorExports from 'glamor';
import React from 'react';
import stylesheet from '../css/index';
import { List } from './list';
var glamor = glamorDefault || glamorExports;
var styles = {
    group: function () { return glamor.css({ label: 'verticaltabs__group' }); },
    groupHeader: function (themeName) {
        var label = 'verticaltabs__group__header';
        return glamor.compose(glamor.css({ label: label }), glamor.css(stylesheet[".psds-" + label]), glamor.css(stylesheet[".psds-" + label + ".psds-theme--" + themeName]));
    },
    groupButton: function () {
        return glamor.css(stylesheet['.psds-verticaltabs__group__button']);
    },
    groupCollapsibleList: function () {
        return glamor.compose(glamor.css(stylesheet['.psds-verticaltabs__list']), glamor.css(stylesheet['.psds-verticaltabs__group__collapsible-list']));
    },
    rotatable: function (open) {
        return glamor.css(stylesheet['.psds-verticaltabs__rotatable'], open && stylesheet['.psds-verticaltabs__rotatable--isOpen']);
    },
    headerLabel: function () { return glamor.css(stylesheet['.psds-verticaltabs__header__label']); }
};
var Group = React.forwardRef(function (props, ref) {
    var rest = omit(props, ['header', 'startOpen']);
    return (React.createElement("li", __assign({ ref: ref }, styles.group(), rest),
        props.header,
        React.createElement(List, null, props.children)));
});
var GroupHeader = React.forwardRef(function (props, ref) {
    var children = props.children, _a = props.tagName, Tag = _a === void 0 ? 'h2' : _a, rest = __rest(props, ["children", "tagName"]);
    var themeName = useTheme();
    return (React.createElement(Tag, __assign({}, rest, styles.groupHeader(themeName), { ref: ref }),
        React.createElement("div", null,
            React.createElement("span", __assign({}, styles.headerLabel()), children))));
});
Group.Header = GroupHeader;
Group.displayName = 'VerticalTabs.Group';
Group.Header.displayName = 'VerticalTabs.Group.Header';
var CollapsibleGroup = React.forwardRef(function (props, ref) {
    var children = props.children, header = props.header, startOpen = props.startOpen, rest = __rest(props, ["children", "header", "startOpen"]);
    var _a = React.useState(startOpen), open = _a[0], setOpenState = _a[1];
    var toggle = function () {
        setOpenState(!open);
    };
    return (React.createElement("div", __assign({ ref: ref }, rest),
        React.isValidElement(header) &&
            React.cloneElement(header, {
                toggle: toggle,
                open: open
            }),
        React.createElement(Collapsible, __assign({ isOpen: open, tagName: "ul" }, styles.groupCollapsibleList()), children)));
});
var CollapsibleGroupHeader = React.forwardRef(function (props, ref) {
    var children = props.children, open = props.open, _a = props.tagName, Tag = _a === void 0 ? 'h2' : _a, toggle = props.toggle, rest = __rest(props, ["children", "open", "tagName", "toggle"]);
    var themeName = useTheme();
    return (React.createElement(Tag, __assign({}, rest, styles.groupHeader(themeName), { ref: ref }),
        React.createElement("button", __assign({}, styles.groupButton(), { onClick: toggle, "aria-expanded": open ? 'true' : 'false' }),
            React.createElement("span", __assign({}, styles.headerLabel()), children),
            React.createElement(CaretDownIcon, __assign({ "aria-hidden": "true", size: CaretDownIcon.sizes.small }, styles.rotatable(Boolean(open)))))));
});
CollapsibleGroup.Header = CollapsibleGroupHeader;
CollapsibleGroup.displayName = 'VerticalTabs.CollapsibleGroup';
CollapsibleGroup.Header.displayName = 'VerticalTabs.CollapsibleGroup.Header';
export { Group, CollapsibleGroup };
