// @pluralsight/ps-design-system-halo/dist/cjs/css/index.js
"use strict";
var _a, _b;
Object.defineProperty(exports, "__esModule", { value: true });
exports.themeClasses = exports.BASE_CLASSNAME = void 0;
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_theme_1 = require("@pluralsight/ps-design-system-theme");
exports.BASE_CLASSNAME = '.psds-halo';
exports.themeClasses = (_a = {},
    _a[ps_design_system_theme_1.names.dark] = ".psds-theme--dark",
    _a[ps_design_system_theme_1.names.light] = ".psds-theme--light",
    _a);
exports.default = (_b = {},
    _b[exports.BASE_CLASSNAME] = {
        position: 'relative',
        display: 'inline-block',
        lineHeight: 0,
        flex: '1 1 100%',
        '&:after': {
            content: ' ',
            position: 'absolute',
            borderWidth: '3px',
            borderStyle: 'solid',
            borderColor: ps_design_system_core_1.colorsBlue[6],
            pointerEvents: 'none',
            visibility: 'hidden'
        }
    },
    _b[exports.BASE_CLASSNAME + "--inline"] = {
        flexBasis: 0,
        flexGrow: 0
    },
    _b[exports.BASE_CLASSNAME + "--error"] = {
        '&:after': {
            visibility: 'visible',
            borderColor: ps_design_system_core_1.colorsStatus.error
        }
    },
    _b[exports.BASE_CLASSNAME + "--visible"] = {
        '&:after': {
            visibility: 'visible',
            borderColor: ps_design_system_core_1.colorsBlue[6]
        }
    },
    _b[exports.BASE_CLASSNAME + "--visible-on-focus"] = {
        '&:focus-within:after, &[focus-within]:after': {
            visibility: 'visible',
            borderColor: ps_design_system_core_1.colorsBlue[6]
        }
    },
    _b[exports.BASE_CLASSNAME + "--gap-size-default" + exports.themeClasses[ps_design_system_theme_1.names.dark]] = {
        '&:after': {
            top: '-4px',
            bottom: '-4px',
            left: '-4px',
            right: '-4px'
        }
    },
    _b[exports.BASE_CLASSNAME + "--gap-size-default" + exports.themeClasses[ps_design_system_theme_1.names.light]] = {
        '&:after': {
            top: '-5px',
            bottom: '-5px',
            left: '-5px',
            right: '-5px'
        }
    },
    _b[exports.BASE_CLASSNAME + "--gap-size-small" + exports.themeClasses[ps_design_system_theme_1.names.dark]] = {
        '&:after': {
            top: '-4px',
            bottom: '-4px',
            left: '-4px',
            right: '-4px'
        }
    },
    _b[exports.BASE_CLASSNAME + "--gap-size-small" + exports.themeClasses[ps_design_system_theme_1.names.light]] = {
        '&:after': {
            top: '-2px',
            bottom: '-2px',
            left: '-2px',
            right: '-2px'
        }
    },
    _b[exports.BASE_CLASSNAME + "--shape-default"] = {
        '&:after': {
            borderRadius: '4px'
        }
    },
    _b[exports.BASE_CLASSNAME + "--shape-pill"] = {
        '&:after': {
            borderRadius: '1000px' // use a big number to make a pill shape
        }
    },
    _b);
