// @pluralsight/ps-design-system-dialog/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var vars = __importStar(require("../vars/index"));
var tailDimension = '14px';
exports.default = (_a = {
        '@keyframes psds-dialog__keyframes__fade': {
            '100%': {
                transform: 'translateY(0)',
                opacity: 1
            }
        },
        '.psds-dialog': function (_a) {
            var fade = _a.fade;
            return ({
                animation: (fade || 'psds-dialog__keyframes__fade') + " " + ps_design_system_core_1.motion.speedFast + " ease-out forwards",
                background: ps_design_system_core_1.colorsWhite,
                borderRadius: '2px',
                boxShadow: '0 1px 2px rgba(0, 0, 0, 0.5)',
                display: 'inline-flex',
                opacity: 0,
                position: 'relative',
                transform: "translateY(" + ps_design_system_core_1.layout.spacingXSmall + ")",
                '&:focus': { outline: 'none' }
            });
        },
        '.psds-dialog__content': {
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            fontSize: ps_design_system_core_1.type.fontSize200,
            fontWeight: ps_design_system_core_1.type.fontWeight500,
            lineHeight: ps_design_system_core_1.type.lineHeightTight,
            padding: ps_design_system_core_1.layout.spacingLarge,
            flex: '1 1 auto',
            overflowY: 'auto'
        },
        // --w-tail
        '.psds-dialog--w-tail': {
            '&:after': {
                backgroundColor: ps_design_system_core_1.colorsWhite,
                content: ' ',
                display: 'block',
                height: tailDimension,
                lineHeight: '0',
                position: 'absolute',
                transform: 'rotate(45deg)',
                whiteSpace: 'pre',
                width: tailDimension
            }
        }
    },
    _a[".psds-dialog--tailPosition-" + vars.tailPositions.topLeft] = {
        '&:after': {
            left: tailDimension,
            top: "calc(" + tailDimension + " / -2)"
        }
    },
    _a[".psds-dialog--tailPosition-" + vars.tailPositions.topCenter] = {
        '&:after': {
            left: '50%',
            top: "calc(" + tailDimension + " / -2)",
            transform: 'translateX(-50%) rotate(45deg)'
        }
    },
    _a[".psds-dialog--tailPosition-" + vars.tailPositions.topRight] = {
        '&:after': {
            right: tailDimension,
            top: "calc(" + tailDimension + " / -2)"
        }
    },
    _a[".psds-dialog--tailPosition-" + vars.tailPositions.bottomLeft] = {
        '&:after': {
            bottom: "calc(" + tailDimension + " / -2)",
            boxShadow: "1px 1px 1px rgba(0, 0, 0, 0.5)",
            left: tailDimension
        }
    },
    _a[".psds-dialog--tailPosition-" + vars.tailPositions.bottomCenter] = {
        '&:after': {
            bottom: "calc(" + tailDimension + " / -2)",
            boxShadow: "1px 1px 1px rgba(0, 0, 0, 0.5)",
            left: '50%',
            transform: 'translateX(-50%) rotate(45deg)'
        }
    },
    _a[".psds-dialog--tailPosition-" + vars.tailPositions.bottomRight] = {
        '&:after': {
            bottom: "calc(" + tailDimension + " / -2)",
            boxShadow: "1px 1px 1px rgba(0, 0, 0, 0.5)",
            right: tailDimension
        }
    },
    // --modal
    _a['.psds-dialog--modal'] = {
        maxHeight: "calc(100% - (2 * " + ps_design_system_core_1.layout.spacingLarge + "))",
        maxWidth: "calc(100% - (2 * " + ps_design_system_core_1.layout.spacingLarge + "))"
    },
    // __close
    _a['.psds-dialog__close'] = {
        position: 'absolute',
        right: ps_design_system_core_1.layout.spacingMedium,
        top: ps_design_system_core_1.layout.spacingMedium,
        // NOTE: copy-paste-modify from psds-button, etc
        display: 'inline-flex',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: '2px',
        fontWeight: ps_design_system_core_1.type.fontWeight500,
        textAlign: 'center',
        whiteSpace: 'nowrap',
        textDecoration: 'none',
        transition: "all " + ps_design_system_core_1.motion.speedNormal,
        verticalAlign: 'middle',
        fontSize: ps_design_system_core_1.type.fontSize200,
        lineHeight: 0,
        padding: 0,
        height: '32px',
        width: '32px',
        border: 'none',
        background: 'none',
        color: ps_design_system_core_1.colorsTextIcon.lowOnLight,
        '&:not([disabled]):hover, &:not([disabled]):focus': {
            outline: 'none',
            boxShadow: "0 0 0 3px " + ps_design_system_util_1.transparentize(0.5, ps_design_system_core_1.colorsPrimaryAction.background),
            cursor: 'pointer',
            background: "rgba(" + ps_design_system_core_1.colorsBackgroundUtilityCsv + ",0.45)",
            color: ps_design_system_core_1.colorsTextIcon.highOnLight
        },
        '&:not([disabled]):active': {
            background: "rgba(" + ps_design_system_core_1.colorsBackgroundUtilityCsv + ",0.65)",
            color: ps_design_system_core_1.colorsTextIcon.highOnLight,
            transform: 'scale(0.98)'
        },
        '& svg': {
            fill: ps_design_system_core_1.colorsTextIcon.lowOnLight,
            height: '24px',
            width: '24px'
        }
    },
    // __overlay
    _a['.psds-dialog__overlay'] = {
        alignItems: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
        display: 'flex',
        height: '100%',
        justifyContent: 'center',
        left: 0,
        position: 'fixed',
        top: 0,
        width: '100%'
    },
    _a);
