// @pluralsight/ps-design-system-banner/dist/cjs/css/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ps_design_system_core_1 = require("@pluralsight/ps-design-system-core");
var ps_design_system_util_1 = require("@pluralsight/ps-design-system-util");
var vars = __importStar(require("../vars/index"));
exports.default = (_a = {
        '.psds-banner': {
            display: 'flex',
            alignItems: 'flex-start',
            width: '100%',
            overflow: 'hidden',
            padding: ps_design_system_core_1.layout.spacingXSmall + " " + ps_design_system_core_1.layout.spacingSmall
        }
    },
    _a[".psds-banner--color-" + vars.colors.blue] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        backgroundColor: ps_design_system_core_1.colorsBlue[7]
    },
    _a[".psds-banner--color-" + vars.colors.green] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        backgroundColor: ps_design_system_core_1.colorsGreen[8]
    },
    _a[".psds-banner--color-" + vars.colors.yellow] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnLight,
        backgroundColor: ps_design_system_core_1.colorsYellow[6]
    },
    _a[".psds-banner--color-" + vars.colors.red] = {
        color: ps_design_system_core_1.colorsTextIcon.highOnDark,
        backgroundColor: ps_design_system_core_1.colorsRed[7]
    },
    _a['.psds-banner__button'] = {
        ':not(#fake-id-to-force-higher-specificity) &': {
            background: 'none',
            borderColor: 'currentColor',
            color: 'currentColor',
            '&:hover, &:active, &:focus': {
                backgroundColor: ps_design_system_util_1.transparentize(0.9, ps_design_system_core_1.colorsBlack)
            },
            '&:focus': { boxShadow: "0 0 0 3px " + ps_design_system_util_1.transparentize(0.5, ps_design_system_core_1.colorsWhite) }
        }
    },
    _a[".psds-banner__button--color-" + vars.colors.blue] = {},
    _a[".psds-banner__button--color-" + vars.colors.green] = {},
    _a[".psds-banner__button--color-" + vars.colors.yellow] = {
        ':not(#fake-id-to-force-higher-specificity)': {
            '&:focus': { boxShadow: "0 0 0 3px " + ps_design_system_util_1.transparentize(0.5, ps_design_system_core_1.colorsBlack) }
        }
    },
    _a[".psds-banner__button--color-" + vars.colors.red] = {},
    _a['.psds-banner__text'] = {
        margin: '0 !important',
        flex: '1',
        textAlign: 'center',
        color: 'currentColor !important',
        '& a': {
            color: 'inherit',
            cursor: 'pointer',
            textDecoration: 'underline',
            '&:hover, &:active, &:focus': {
                color: 'inherit',
                transition: "all " + ps_design_system_core_1.motion.speedNormal,
                opacity: '0.85'
            }
        }
    },
    _a['.psds-banner__dismiss'] = {
        position: 'relative',
        display: 'flex',
        alignItems: 'flex-start',
        background: 'none',
        border: 'none',
        marginLeft: ps_design_system_core_1.layout.spacingSmall,
        padding: '0',
        color: 'currentColor',
        cursor: 'pointer',
        '&:hover, &:focus, &:active': {
            outline: 'none',
            borderRadius: '2px',
            boxShadow: '0 0 0 3px currentColor'
        }
    },
    _a);
